---
title: "Spillover"
output:
  rmdformats::readthedown:
    highlight: kate
author: 
  name: Gregor Sturm
  email: sturm@pieris.com
  affilation: Pieris Pharmaceuticals GmbH, Lise-Meitner-StraÃŸe 30, 85354 Freising, Germany
---


```{r setup}
if (!require("pacman")) install.packages("pacman")
pacman::p_load(Biobase, 
               readxl,
               MASS,
               tidyverse,
               testit,
               BioQC,
               pheatmap,
               xCell,
               MCPcounter,
               EPIC,
               withr)

```

Part 1 of the immune deconvolution benchmark: The Spillover analysis. 

* take pure samples of the given cell types (ideally not from the training data)
* negative control (something completly different)
* positive control = blood (should contain everything)
* NGS vs. microarray? 
-> NGS is more important 


# Data
## Single Cell
* Human cell atlas (https://www.humancellatlas.org/): News item claims that single cell data of 1e6 immune cells is already available
* 10X genomics PBMC https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.1.0/pbmc8k


## Bulk tissue 
* Fantom5 


# Spillover analysis with fantom 5 samples
```{r}
# load and rename fantom5 data
load("../results/f5_eset.rda")

f5_eset = f5_eset[,!is.na(pData(f5_eset)$cell_type)]
```

## BioQC with Angelova Signatures
```{r bioqc, fig.width=7, fig.height=10}
gmt = readGmt("../data/angelova_immune_cells.gmt")
bioqc_res = wmwTest(f5_eset, gmt, col="hgnc_symbol")
colnames(bioqc_res) = pData(f5_eset)$name

pheatmap(absLog10p(bioqc_res), cluster_rows = FALSE, cluster_cols = FALSE)
```


## xCell
```{r xcell, fig.width=7, fig.height=15}
eset_mat = exprs(f5_eset)
rownames(eset_mat) = fData(f5_eset)$hgnc_symbol

xcell_res = xCellAnalysis(eset_mat)
colnames(xcell_res) = pData(f5_eset)$name

pheatmap(xcell_res, cluster_rows = FALSE, cluster_cols = FALSE)

```


## MCPcounter
```{r mcp_counter, , fig.width=7, fig.height=10}
mcp_res = MCPcounter.estimate(eset_mat, "HUGO_symbols")

colnames(mcp_res) = pData(f5_eset)$name

pheatmap(mcp_res, cluster_rows = FALSE, cluster_cols = FALSE)

```


## EPIC
```{r epic, fig.width=7, fig.height=10}
epic_res_raw = EPIC(bulk=eset_mat)
epic_res = t(epic_res_raw$cellFractions)

colnames(epic_res) = pData(f5_eset)$name

pheatmap(epic_res, cluster_rows = FALSE, cluster_cols = FALSE)

```



## TIMER
```{r TIMER, fig.width=7, fig.height=10}
# export data for timer
write_tsv(as_tibble(eset_mat, rownames="gene_symbol"), path="../results/timer_input.tsv")

preprocess_timer = function(df) {
  df = as.data.frame(df)
  rownames(df) = df$sampleID
  df$sampleID = NULL
  df = t(df)
  colnames(df) = pData(f5_eset)$name
  df
}

timer_acc = read_csv("../data/timer/score_matrix_ACC.csv") %>% preprocess_timer()
timer_gbm = read_csv("../data/timer/score_matrix_GBM.csv") %>% preprocess_timer()


pheatmap(timer_acc, cluster_rows = FALSE, cluster_cols = FALSE)
pheatmap(timer_gbm, cluster_rows = FALSE, cluster_cols = FALSE)


```


## CIBERSORT
```{r CIBERSORT, fig.width=7, fig.height=10}
preprocess_cibersort = function(df) {
  df = as.data.frame(df)
  rownames(df) = df$`Input Sample`
  df$`Input Sample` = NULL
  df = t(df)
  colnames(df) = pData(f5_eset)$name
  df
}

cibersort_abs = read_tsv("../data/cibersort/CIBERSORT.Output_Abs_Job1.txt") %>% select(-`Pearson Correlation`, -`Absolute score`, -`P-value`) %>% preprocess_cibersort()
cibersort_rel = read_tsv("../data/cibersort/CIBERSORT.Output_Job1.txt") %>% select(-`P-value`, -`Pearson Correlation`, -`RMSE`) %>% preprocess_cibersort()


pheatmap(cibersort_abs, cluster_rows = FALSE, cluster_cols = FALSE)
pheatmap(cibersort_rel, cluster_rows = FALSE, cluster_cols = FALSE)


```

## quanTIseq
```{r quanTIseq, fig.width=7, fig.height=10}
with_dir("../lib/quantiseq_decon", {
  source("quanTIseq_decon.R")
  qseq_res = deconvolute_quantiseq(eset_mat) 
  qseq_res %<>% as_tibble() %>% select(-Sample) %>% as.matrix() %>% t()
})

colnames(qseq_res) = pData(f5_eset)$name
pheatmap(qseq_res, cluster_rows = FALSE, cluster_cols = FALSE)


```


# Comparison
```{r, fig.width=12, fig.height=5}
range01 <- function(x){(x-min(x))/(max(x)-min(x))}

mapping = read_excel("../tables/cell_type_mapping.xlsx", sheet = 1)

results = list(
  cibersort = cibersort_abs,
  timer = timer_acc,
  xcell = xcell_res,
  mcpcounter = mcp_res,
  quantiseq = qseq_res,
  epic = epic_res
)

results2 = lapply(names(results), function(method) {
  tmp_map = mapping %>% select(cell_type, !!method)
  tmp_res = results[[method]] %>% range01() %>% as_tibble(rownames = "tmp_cell_type")
  tmp_res %>% inner_join(tmp_map, by=c("tmp_cell_type"=method)) %>% select(-tmp_cell_type) %>%
    mutate(method=method)
})

res_table = bind_rows(results2) %>% gather(sample, predicted_fraction, -method, -cell_type) %>% 
  separate(sample, into=c("sample_cell_type", "sample_other_info"), sep=",")

res_table %>% ggplot(aes(x=method, y=cell_type, fill=predicted_fraction)) +
  geom_tile(colour="white") +
  scale_fill_gradientn(colours=rev(brewer.pal(n = 7, name="RdYlBu"))) +
  facet_wrap(~sample_cell_type, nrow=1) +
  theme(axis.text.x=element_text(angle = -90, vjust = 0.5, hjust=0)) 


```