<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta property="og:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
<meta property="og:type" content="book" />


<meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />


<meta name="author" content="manuscript in preparation" />
<meta name="author" content="Gregor Sturm^{1,2}" />
<meta name="author" content="Tatsiana Aneichyk^1" />


<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-AMS_CHTML-full" type="text/javascript"></script>

<meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">

<title>Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</title>

<script src="libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
</style>
</head>

<body>

<div class="container-fluid main-container">


<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul>
<li class="has-sub"><a href="index.html#introduction"><span class="toc-section-number">1</span> Introduction</a><ul>
<li class="has-sub"><a href="1-1-conceptual-differences-between-methods.html#conceptual-differences-between-methods"><span class="toc-section-number">1.1</span> Conceptual differences between methods</a><ul>
<li><a href="1-1-conceptual-differences-between-methods.html#marker-gene-based-methods-allow-comparison-between-samples-but-not-between-cell-types"><span class="toc-section-number">1.1.1</span> Marker gene based methods (allow comparison between samples, but not between cell types)</a></li>
<li><a href="1-1-conceptual-differences-between-methods.html#deconvolution-based-methods-relative-fractions-allow-comparisons-between-cell-types-but-not-samples"><span class="toc-section-number">1.1.2</span> Deconvolution-based methods (relative fractions, allow comparisons between cell types, but not samples)</a></li>
<li><a href="1-1-conceptual-differences-between-methods.html#deconvolution-based-methods-absolute-fractions-allow-comparisons-between-cell-types-and-samples"><span class="toc-section-number">1.1.3</span> Deconvolution-based methods (absolute fractions, allow comparisons between cell types and samples)</a></li>
</ul></li>
</ul></li>
<li class="has-sub"><a href="2-input-data.html#input-data"><span class="toc-section-number">2</span> Input data</a><ul>
<li><a href="2-1-cell-type-hierarchy.html#cell-type-hierarchy"><span class="toc-section-number">2.1</span> cell type hierarchy</a></li>
<li><a href="2-2-single-cell-data-for-simulated-mixtures.html#single-cell-data-for-simulated-mixtures"><span class="toc-section-number">2.2</span> Single cell data for simulated mixtures</a></li>
<li><a href="2-3-immune-cell-bulk-reference-samples-quantiseq-training-data.html#immune-cell-bulk-reference-samples-quantiseq-training-data"><span class="toc-section-number">2.3</span> Immune cell bulk reference samples (quanTIseq training data)</a></li>
<li class="has-sub"><a href="2-4-validation-data-rnaseq-matched-gold-standard.html#validation-data-rnaseq-matched-gold-standard"><span class="toc-section-number">2.4</span> Validation data (RNAseq + matched gold standard)</a><ul>
<li><a href="2-4-validation-data-rnaseq-matched-gold-standard.html#hoeck-data"><span class="toc-section-number">2.4.1</span> Hoeck data</a></li>
<li><a href="2-4-validation-data-rnaseq-matched-gold-standard.html#schelker-data"><span class="toc-section-number">2.4.2</span> Schelker data</a></li>
<li><a href="2-4-validation-data-rnaseq-matched-gold-standard.html#racle-data"><span class="toc-section-number">2.4.3</span> Racle data</a></li>
</ul></li>
</ul></li>
<li class="has-sub"><a href="3-validitiy-of-single-cell-rna-seq-as-reference.html#validitiy-of-single-cell-rna-seq-as-reference"><span class="toc-section-number">3</span> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="has-sub"><a href="3-1-validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data.html#validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data"><span class="toc-section-number">3.1</span> Validity of simulated bulk tissues from single cell sequencing data</a><ul>
<li><a href="3-1-validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data.html#compare-simululated-samples-to-bulk-samples"><span class="toc-section-number">3.1.1</span> Compare simululated samples to bulk samples</a></li>
<li><a href="3-1-validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data.html#compare-the-predicted-fractions"><span class="toc-section-number">3.1.2</span> Compare the predicted fractions</a></li>
</ul></li>
<li><a href="3-2-correlated-simulated-samples-with-bulk-of-the-same-tissue-.html#correlated-simulated-samples-with-bulk-of-the-same-tissue."><span class="toc-section-number">3.2</span> Correlated simulated samples with bulk of the same tissue.</a></li>
</ul></li>
<li class="has-sub"><a href="4-mixing-study.html#mixing-study"><span class="toc-section-number">4</span> Mixing study</a><ul>
<li><a href="4-1-average-fraction-of-tumour-cells.html#average-fraction-of-tumour-cells"><span class="toc-section-number">4.1</span> Average fraction of tumour cells</a></li>
<li><a href="4-2-create-simulated-bulk-tissues.html#create-simulated-bulk-tissues"><span class="toc-section-number">4.2</span> Create simulated bulk tissues</a></li>
<li><a href="4-3-run-the-deconvolution.html#run-the-deconvolution"><span class="toc-section-number">4.3</span> Run the deconvolution</a></li>
<li class="has-sub"><a href="4-4-results.html#results"><span class="toc-section-number">4.4</span> Results</a><ul>
<li><a href="4-4-results.html#correlation-plots"><span class="toc-section-number">4.4.1</span> Correlation plots</a></li>
<li><a href="4-4-results.html#calculate-correlations-for-each-method-and-cell-type"><span class="toc-section-number">4.4.2</span> Calculate correlations for each method and cell type</a></li>
<li><a href="4-4-results.html#deviation-from-slope-xy-for-all-absolute-methods"><span class="toc-section-number">4.4.3</span> Deviation from slope (x=y) for all absolute methods</a></li>
</ul></li>
</ul></li>
<li><a href="5-validation-with-real-data.html#validation-with-real-data"><span class="toc-section-number">5</span> Validation with real data</a></li>
<li class="has-sub"><a href="6-correlation-of-methods-across-tcga-samples-.html#correlation-of-methods-across-tcga-samples."><span class="toc-section-number">6</span> Correlation of methods across TCGA samples.</a><ul>
<li><a href="6-1-heatmap-clustered-by-correlation.html#heatmap-clustered-by-correlation"><span class="toc-section-number">6.1</span> Heatmap clustered by correlation</a></li>
<li><a href="6-2-heatmap-unclustered-sorted-by-cell-type.html#heatmap-unclustered-sorted-by-cell-type"><span class="toc-section-number">6.2</span> Heatmap unclustered, sorted by cell type</a></li>
</ul></li>
<li class="has-sub"><a href="7-sensitivity-analysis.html#sensitivity-analysis"><span class="toc-section-number">7</span> Sensitivity analysis</a><ul>
<li class="has-sub"><a href="7-1-predictions-with-increasing-immune-cell-content.html#predictions-with-increasing-immune-cell-content"><span class="toc-section-number">7.1</span> Predictions with increasing immune cell content</a><ul>
<li><a href="7-1-predictions-with-increasing-immune-cell-content.html#summary-figure-at-which-percentage-the-method-captures-the-cell-type-above-noise-level"><span class="toc-section-number">7.1.1</span> Summary figure: At which percentage the method captures the cell type above noise level?</a></li>
<li><a href="7-1-predictions-with-increasing-immune-cell-content.html#summary-figure-predicted-amount-of-cells-when-there-are-none"><span class="toc-section-number">7.1.2</span> Summary figure: predicted amount of cells when there are none</a></li>
</ul></li>
</ul></li>
<li class="has-sub"><a href="8-spillover-analysis.html#spillover-analysis"><span class="toc-section-number">8</span> Spillover Analysis</a><ul>
<li><a href="8-1-complete-spillover-matrix.html#complete-spillover-matrix"><span class="toc-section-number">8.1</span> Complete Spillover Matrix</a></li>
<li><a href="8-2-summary-figure-noise-ratio.html#summary-figure-noise-ratio"><span class="toc-section-number">8.2</span> Summary figure: Noise ratio</a></li>
</ul></li>
<li class="has-sub"><a href="9-robustness-against-noise.html#robustness-against-noise"><span class="toc-section-number">9</span> Robustness against noise</a><ul>
<li class="has-sub"><a href="9-1-result-figures.html#result-figures"><span class="toc-section-number">9.1</span> Result figures</a><ul>
<li><a href="9-1-result-figures.html#correlations"><span class="toc-section-number">9.1.1</span> Correlations</a></li>
</ul></li>
</ul></li>
<li><a href="10-conclusions.html#conclusions"><span class="toc-section-number">10</span> Conclusions</a></li>
<li class="has-sub"><a href="11-materials-and-methods.html#materials-and-methods"><span class="toc-section-number">11</span> Materials and Methods</a><ul>
<li><a href="11-1-data-acquisition.html#data-acquisition"><span class="toc-section-number">11.1</span> Data acquisition</a></li>
<li><a href="11-2-validation-datasets-flow-cytometry-scrnaseq-imageing-matched-rna-seq.html#validation-datasets-flow-cytometry-scrnaseq-imageing-matched-rna-seq"><span class="toc-section-number">11.2</span> Validation datasets (Flow cytometry, scRNAseq, Imageing + matched RNA seq)</a></li>
</ul></li>
<li><a href="12-creating-publication-ready-figures.html#creating-publication-ready-figures"><span class="toc-section-number">12</span> Creating publication ready figures</a></li>
<li><a href="13-references.html#references"><span class="toc-section-number">13</span> References</a></li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="validation-data-rnaseq-matched-gold-standard" class="section level2">
<h2><span class="header-section-number">2.4</span> Validation data (RNAseq + matched gold standard)</h2>
<p>e.g. IHC, Flow cytometry, single cell</p>
<div id="htmlwidget-d4fd829cca2c3dfc94be" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d4fd829cca2c3dfc94be">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8"],["Finotello 2017","Finotello 2017","Hoeck 2015","Johnson 2015","Racle 2017","Zimmermann 2016","Schelker 2017","Chung 2017"],[9,9,8,31,4,12,3,11],["CRC","PBMC/PMN","PBMC/PMN","metastatic melanoma","metastatic melanoma","PBMC before influenza vacc","ovarian cancer ascites","BRCA"],["RNAseq","RNAseq","RNAseq","RNAseq","RNAseq","RNAseq","RNAseq","RNAseq"],["IHC","FlowCytometry","FlowCytometry","IHC","FlowCytometry","FlowCytometry","FlowCytometry + Single Cell RNAseq","Single Cell RNAseq + IHC for B and T"],["CD3, CD8, CD45RO, FOXP3, PD1","B, Mono, Neu, NK, CD4, CD8, Treg, DC","T, B, NK, MDC, Mono, Neu","CD8, CD4, FOXP3+","B, CD4, CD8, NK, Melanoma/other","B, CD4, CD8, NK, Mono, Other","CD8, CD4, B, NK, DC, Mono, ...",null],[null,null,"Finotello 2017","Finotello 2017",null,"Racle 2017",null,null],["Available upon publication","Available upon publication","Available through personal communicaion","How to get?",null,"obtained through Racle 2017",null,null],[null,null,"x",null,"x",null,"x",null],[null,null,"x",null,"x","x","x",null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Study<\/th>\n      <th># Samples<\/th>\n      <th>Sample Type<\/th>\n      <th>Gene Expression by<\/th>\n      <th>Validation by<\/th>\n      <th>Cell types<\/th>\n      <th>Cited by<\/th>\n      <th>Comments<\/th>\n      <th>Got TPM<\/th>\n      <th>Got Validation<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="hoeck-data" class="section level3">
<h3><span class="header-section-number">2.4.1</span> Hoeck data</h3>
<p>8 PBMC, RNAseq+flow cytometry</p>
<p>Measured cell types: T cell, Macrophage/Monocyte, B cell, Dendritic cell, NK cell.</p>
</div>
<div id="schelker-data" class="section level3">
<h3><span class="header-section-number">2.4.2</span> Schelker data</h3>
<p>Load the data and merge replicates.
3 ovarian cancer ascites samplese which have matched flow cytometry/single cell/bulk data.</p>
<p>Single cell count for the 3 samples</p>
<table>
<thead>
<tr class="header">
<th align="left">donor</th>
<th align="right">sum(cell_count)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">7873M</td>
<td align="right">864</td>
</tr>
<tr class="even">
<td align="left">7882M</td>
<td align="right">902</td>
</tr>
<tr class="odd">
<td align="left">7892M</td>
<td align="right">773</td>
</tr>
</tbody>
</table>
<div id="htmlwidget-87d41fe4285d780a2c5d" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-87d41fe4285d780a2c5d">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7873M","7882M","7892M","7873M","7882M","7892M","7882M"],["B cell","B cell","B cell","Cancer associated fibroblast","Cancer associated fibroblast","Cancer associated fibroblast","Dendritic cell","Dendritic cell","Dendritic cell","Macrophage/Monocyte","Macrophage/Monocyte","Macrophage/Monocyte","NK cell","NK cell","NK cell","Ovarian carcinoma cell","Ovarian carcinoma cell","Ovarian carcinoma cell","PBMC","T cell CD4+ (non-regulatory)","T cell CD4+ (non-regulatory)","T cell CD4+ (non-regulatory)","T cell CD8+","T cell CD8+","T cell CD8+","T cell regulatory (Tregs)"],[4,4,5,4,11,16,22,46,39,579,599,520,8,51,42,17,34,46,200,26,137,100,4,19,5,1],[0.00462962962962963,0.00443458980044346,0.00646830530401035,0.00462962962962963,0.0121951219512195,0.0206985769728331,0.025462962962963,0.0509977827050998,0.0504527813712807,0.670138888888889,0.664079822616408,0.672703751617076,0.00925925925925926,0.0565410199556541,0.0543337645536869,0.0196759259259259,0.0376940133037694,0.0595084087968952,0.231481481481481,0.0300925925925926,0.151884700665188,0.129366106080207,0.00462962962962963,0.0210643015521064,0.00646830530401035,0.00110864745011086]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>donor<\/th>\n      <th>cell_type<\/th>\n      <th>cell_count<\/th>\n      <th>true_fraction<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="racle-data" class="section level3">
<h3><span class="header-section-number">2.4.3</span> Racle data</h3>
<p>4 metastatic melanoma, matched bulkRNAseq with FlowCytometry.
<div id="htmlwidget-f3bd09db4466b807453e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f3bd09db4466b807453e">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314"],["B cell","B cell","B cell","B cell","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD8+","T cell CD8+","T cell CD8+","T cell CD8+","NK cell","NK cell","NK cell","NK cell"],[0.1812,0.3248,0.0579,0.4667,0.0082,0.2315,0.0276,0.1815,0.0035,0.0582,0.0376,0.0454,0.005,0.0017,0.0017,0.0025]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>cell_type<\/th>\n      <th>true_fraction<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script></p>

</div>
</div>
<!-- </div> -->
<p style="text-align: center;">
<a href="2-3-immune-cell-bulk-reference-samples-quantiseq-training-data.html"><button class="btn btn-default">Previous</button></a>
<a href="3-validitiy-of-single-cell-rna-seq-as-reference.html"><button class="btn btn-default">Next</button></a>
</p>
</div>
</div>


</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

</body>
</html>
