<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="manuscript in preparation">
<meta name="author" content="Gregor Sturm^{1,2}">
<meta name="author" content="Tatsiana Aneichyk^1">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="spillover-analysis.html">
<link rel="next" href="references.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-reference-samples"><i class="fa fa-check"></i><b>2.3</b> Immune cell reference samples</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#pbmc-samples-from-hoek2015"><i class="fa fa-check"></i><b>2.4</b> 8 PBMC samples from <span class="citation">Hoek et al. (<span>2015</span>)</span></a></li>
<li class="chapter" data-level="2.5" data-path="input-data.html"><a href="input-data.html#ovarian-cancer-ascites-samples-from-schelker2017"><i class="fa fa-check"></i><b>2.5</b> 3 ovarian cancer ascites samples from <span class="citation">Schelker et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.6" data-path="input-data.html"><a href="input-data.html#metastatic-melanoma-samples-from-epic2017"><i class="fa fa-check"></i><b>2.6</b> 4 metastatic melanoma samples from <span class="citation">Racle et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.7" data-path="input-data.html"><a href="input-data.html#data-sanity-checks"><i class="fa fa-check"></i><b>2.7</b> Data sanity checks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-to-genuine-bulk-samples"><i class="fa fa-check"></i><b>3.1</b> Compare simululated to genuine bulk samples</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#enrichment-analysis-test-for-systematic-bias"><i class="fa fa-check"></i><b>3.2</b> Enrichment analysis: Test for systematic Bias</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#immune-relevant-genes-are-consistent-between-sc-and-bulk-data."><i class="fa fa-check"></i><b>3.2.1</b> Immune-relevant genes are consistent between sc and bulk data.</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-predicted-fractions"><i class="fa fa-check"></i><b>3.3</b> Compare predicted fractions</a></li>
<li class="chapter" data-level="3.4" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlation-with-immune-reference-samples."><i class="fa fa-check"></i><b>3.4</b> Correlation with immune reference samples.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html"><i class="fa fa-check"></i><b>4</b> Simulation benchmark</a><ul>
<li class="chapter" data-level="4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>4.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>4.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.3</b> Run the deconvolution</a></li>
<li class="chapter" data-level="4.4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#results"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#correlation-plots"><i class="fa fa-check"></i><b>4.4.1</b> Correlation plots</a></li>
<li class="chapter" data-level="4.4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>4.4.2</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="4.4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#absolute-deviation"><i class="fa fa-check"></i><b>4.4.3</b> Absolute deviation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>5</b> Validation with real data</a><ul>
<li class="chapter" data-level="5.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between--and-within-sample-comparisons"><i class="fa fa-check"></i><b>5.1</b> Between- and within-sample comparisons</a></li>
<li class="chapter" data-level="5.2" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#within-sample-comparison"><i class="fa fa-check"></i><b>5.2</b> Within-sample comparison</a></li>
<li class="chapter" data-level="5.3" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between-sample-comparisons"><i class="fa fa-check"></i><b>5.3</b> Between-sample comparisons</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html"><i class="fa fa-check"></i><b>6</b> Detection limit and false positive predictions</a><ul>
<li class="chapter" data-level="6.1" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#predictions-with-increasing-immune-cell-content"><i class="fa fa-check"></i><b>6.1</b> Predictions with increasing immune cell content</a></li>
<li class="chapter" data-level="6.2" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#detection-limit"><i class="fa fa-check"></i><b>6.2</b> Detection Limit</a></li>
<li class="chapter" data-level="6.3" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#false-positive-predictions"><i class="fa fa-check"></i><b>6.3</b> False-positive predictions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>7</b> Spillover Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#complete-spillover-matrix"><i class="fa fa-check"></i><b>7.1</b> Complete Spillover Matrix</a></li>
<li class="chapter" data-level="7.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-signal-to-noise-ratio"><i class="fa fa-check"></i><b>7.2</b> Summary figure: Signal to noise ratio</a></li>
<li class="chapter" data-level="7.3" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-migration-charts"><i class="fa fa-check"></i><b>7.3</b> Summary figure: migration charts</a></li>
<li class="chapter" data-level="7.4" data-path="spillover-analysis.html"><a href="spillover-analysis.html#investigate-marker-genes"><i class="fa fa-check"></i><b>7.4</b> Investigate marker genes</a></li>
<li class="chapter" data-level="7.5" data-path="spillover-analysis.html"><a href="spillover-analysis.html#check-expression-of-markers-in-detection-limit-simulation-dataset."><i class="fa fa-check"></i><b>7.5</b> Check expression of markers in “detection limit” simulation dataset.</a><ul>
<li class="chapter" data-level="7.5.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#expression-of-marker-genes-in-the-cell-populations"><i class="fa fa-check"></i><b>7.5.1</b> Expression of marker genes in the cell populations</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="spillover-analysis.html"><a href="spillover-analysis.html#deconvolution-results-before-and-after"><i class="fa fa-check"></i><b>7.6</b> Deconvolution results before and after</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html"><i class="fa fa-check"></i><b>8</b> Creating publication ready figures</a><ul>
<li class="chapter" data-level="8.1" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#benchmark-results"><i class="fa fa-check"></i><b>8.1</b> Benchmark results</a></li>
<li class="chapter" data-level="8.2" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#detection-limit-false-positive-figure"><i class="fa fa-check"></i><b>8.2</b> detection limit / false positive figure</a></li>
<li class="chapter" data-level="8.3" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#migration-charts-for-spillover-analysis"><i class="fa fa-check"></i><b>8.3</b> Migration charts for Spillover analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="creating-publication-ready-figures" class="section level1">
<h1><span class="header-section-number">8</span> Creating publication ready figures</h1>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1">fun_breaks3 =<span class="st"> </span><span class="cf">function</span>(limits) {</a>
<a class="sourceLine" id="cb35-2" data-line-number="2">  breaks =<span class="st"> </span><span class="kw">signif</span>(<span class="kw">max</span>(limits) <span class="op">*</span><span class="st"> </span><span class="kw">c</span>(<span class="fl">0.25</span>, <span class="fl">0.75</span>),<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb35-3" data-line-number="3">  <span class="kw">names</span>(breaks) =<span class="st"> </span><span class="kw">attr</span>(breaks, <span class="st">&quot;labels&quot;</span>)</a>
<a class="sourceLine" id="cb35-4" data-line-number="4">  breaks</a>
<a class="sourceLine" id="cb35-5" data-line-number="5">}</a>
<a class="sourceLine" id="cb35-6" data-line-number="6"></a>
<a class="sourceLine" id="cb35-7" data-line-number="7">tmp_cor =<span class="st"> </span>res_methods_validity<span class="op">$</span>all_results_mapped <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-8" data-line-number="8"><span class="st">  </span><span class="kw">group_by</span>(method) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-9" data-line-number="9"><span class="st">  </span><span class="kw">do</span>(<span class="kw">make_cor</span>(.<span class="op">$</span>bulk, .<span class="op">$</span>simulated)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-10" data-line-number="10"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">pearson=</span><span class="kw">round</span>(pearson, <span class="dv">2</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb35-11" data-line-number="11"><span class="st">  </span><span class="kw">inner_join</span>(method_names)</a>
<a class="sourceLine" id="cb35-12" data-line-number="12"></a>
<a class="sourceLine" id="cb35-13" data-line-number="13">res_methods_validity<span class="op">$</span>all_results_mapped <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-14" data-line-number="14"><span class="st">  </span><span class="kw">inner_join</span>(cell_type_names) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-15" data-line-number="15"><span class="st">  </span><span class="kw">inner_join</span>(method_names) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-16" data-line-number="16"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>bulk, <span class="dt">y=</span>simulated)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-17" data-line-number="17"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">colour=</span>cell_type_name), <span class="dt">size=</span>.<span class="dv">5</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-18" data-line-number="18"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>method_name, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>, <span class="dt">nrow=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-19" data-line-number="19"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-20" data-line-number="20"><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">data=</span>tmp_cor, <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">paste0</span>(<span class="st">&quot;r=&quot;</span>, pearson), <span class="dt">x=</span><span class="dv">0</span>, <span class="dt">y=</span><span class="dv">0</span>), <span class="dt">hjust=</span><span class="dv">0</span>, <span class="dt">vjust=</span><span class="op">-</span><span class="dv">8</span>, <span class="dt">size=</span><span class="fl">2.4</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-21" data-line-number="21"><span class="st">  </span><span class="kw">panel_border</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-22" data-line-number="22"><span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span>fun_breaks3) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-23" data-line-number="23"><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks=</span>fun_breaks3) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb35-24" data-line-number="24"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">strip.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">6</span>), </a>
<a class="sourceLine" id="cb35-25" data-line-number="25">        <span class="dt">panel.spacing =</span> <span class="kw">unit</span>(<span class="dv">2</span>, <span class="st">&quot;mm&quot;</span>), <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">8</span>))</a></code></pre></div>
<p><img src="_main_files/figure-html/method-validity-f1a-1.png" width="672" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" data-line-number="1">fun_breaks =<span class="st"> </span><span class="cf">function</span>(limits) {</a>
<a class="sourceLine" id="cb36-2" data-line-number="2">  breaks =<span class="st"> </span><span class="kw">signif</span>(<span class="kw">max</span>(limits) <span class="op">*</span><span class="st"> </span><span class="kw">c</span>(<span class="fl">0.25</span>, <span class="fl">0.75</span>),<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb36-3" data-line-number="3">  <span class="kw">names</span>(breaks) =<span class="st"> </span><span class="kw">attr</span>(breaks, <span class="st">&quot;labels&quot;</span>)</a>
<a class="sourceLine" id="cb36-4" data-line-number="4">  breaks</a>
<a class="sourceLine" id="cb36-5" data-line-number="5">}</a>
<a class="sourceLine" id="cb36-6" data-line-number="6">corr_annot =<span class="st"> </span>res_mixing_study<span class="op">$</span>correlations <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-7" data-line-number="7"><span class="st">  </span><span class="kw">select</span>(method, cell_type, pearson) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-8" data-line-number="8"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">pearson =</span> <span class="kw">round</span>(pearson, <span class="dv">2</span>)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-9" data-line-number="9"><span class="st">  </span><span class="kw">inner_join</span>(method_names) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb36-10" data-line-number="10"><span class="st">  </span><span class="kw">inner_join</span>(cell_type_names) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb36-11" data-line-number="11"><span class="st">  </span><span class="kw">distinct</span>()</a>
<a class="sourceLine" id="cb36-12" data-line-number="12">plot_mixing =<span class="st"> </span>res_mixing_study<span class="op">$</span>all_results <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-13" data-line-number="13"><span class="st">  </span><span class="kw">inner_join</span>(method_names) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb36-14" data-line-number="14"><span class="st">  </span><span class="kw">inner_join</span>(cell_type_names) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb36-15" data-line-number="15"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>true_fraction, <span class="dt">y=</span>estimate)) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-16" data-line-number="16"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size=</span>.<span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-17" data-line-number="17"><span class="st">    </span><span class="kw">geom_text</span>(<span class="dt">data=</span>corr_annot, <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">label=</span><span class="kw">paste0</span>(<span class="st">&quot;r=&quot;</span>, pearson), <span class="dt">x=</span><span class="op">-</span><span class="ot">Inf</span>, <span class="dt">y=</span><span class="op">-</span><span class="ot">Inf</span>), <span class="dt">hjust=</span><span class="op">-</span><span class="fl">0.1</span>, <span class="dt">vjust=</span><span class="op">-</span><span class="dv">8</span>, <span class="dt">size=</span><span class="fl">2.4</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-18" data-line-number="18"><span class="st">    </span><span class="kw">facet_grid</span>(method_name <span class="op">~</span><span class="st"> </span>cell_type_name, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-19" data-line-number="19"><span class="st">    </span><span class="kw">stat_smooth</span>(<span class="kw">aes</span>(<span class="dt">color=</span>method), <span class="dt">method=</span><span class="st">&quot;lm&quot;</span>, <span class="dt">size=</span>.<span class="dv">4</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-20" data-line-number="20"><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>color_scales<span class="op">$</span>methods, <span class="dt">na.value=</span><span class="st">&quot;grey&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-21" data-line-number="21"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(.<span class="dv">2</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-22" data-line-number="22"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks=</span>fun_breaks) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-23" data-line-number="23"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="dt">strip.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">6</span>), </a>
<a class="sourceLine" id="cb36-24" data-line-number="24">          <span class="dt">panel.spacing =</span> <span class="kw">unit</span>(.<span class="dv">5</span>, <span class="st">&quot;mm&quot;</span>), <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">8</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb36-25" data-line-number="25"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;estimated fraction&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-26" data-line-number="26"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;true fraction&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb36-27" data-line-number="27"><span class="st">    </span><span class="kw">panel_border</span>()</a>
<a class="sourceLine" id="cb36-28" data-line-number="28"></a>
<a class="sourceLine" id="cb36-29" data-line-number="29"><span class="kw">ggsave</span>(<span class="st">&quot;../results/figures/correlations_mixing.pdf&quot;</span>, <span class="dt">width=</span>WIDTH, <span class="dt">height=</span><span class="dv">140</span>, <span class="dt">units =</span> <span class="st">&quot;mm&quot;</span>)</a>
<a class="sourceLine" id="cb36-30" data-line-number="30"><span class="kw">ggsave</span>(<span class="st">&quot;../results/figures/correlations_mixing.png&quot;</span>, <span class="dt">width=</span>WIDTH, <span class="dt">height=</span><span class="dv">140</span>, <span class="dt">units =</span> <span class="st">&quot;mm&quot;</span>, <span class="dt">dpi=</span><span class="dv">600</span>)</a></code></pre></div>
<div id="benchmark-results" class="section level2">
<h2><span class="header-section-number">8.1</span> Benchmark results</h2>
<div class="figure"><span id="fig:unnamed-chunk-77"></span>
<img src="_main_files/figure-html/unnamed-chunk-77-1.png" alt="Benchmark results main figure. " width="672" />
<p class="caption">
Figure 8.1: Benchmark results main figure.
</p>
</div>
</div>
<div id="detection-limit-false-positive-figure" class="section level2">
<h2><span class="header-section-number">8.2</span> detection limit / false positive figure</h2>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1">data =<span class="st"> </span>res_sensitivity<span class="op">$</span>all_results <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2"><span class="st">  </span><span class="kw">filter</span>(cell_type <span class="op">==</span><span class="st"> </span>input_cell_type) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-3" data-line-number="3"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">frac_immune_cells =</span> <span class="kw">as.numeric</span>(frac_immune_cells)<span class="op">/</span><span class="dv">100</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-4" data-line-number="4"><span class="st">  </span><span class="kw">filter</span>(frac_immune_cells <span class="op">&lt;</span><span class="st"> </span><span class="fl">.4</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-5" data-line-number="5"><span class="st">  </span><span class="kw">inner_join</span>(method_names) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-6" data-line-number="6"><span class="st">  </span><span class="kw">inner_join</span>(cell_type_names) </a>
<a class="sourceLine" id="cb37-7" data-line-number="7"></a>
<a class="sourceLine" id="cb37-8" data-line-number="8">data_detection_limit =<span class="st"> </span>res_sensitivity<span class="op">$</span>sensitivity <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-9" data-line-number="9"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">cell_type =</span> input_cell_type) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-10" data-line-number="10"><span class="st">  </span><span class="kw">inner_join</span>(method_names) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-11" data-line-number="11"><span class="st">  </span><span class="kw">inner_join</span>(cell_type_names) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-12" data-line-number="12"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">min_frac =</span> <span class="kw">as.numeric</span>(min_frac)<span class="op">/</span><span class="dv">100</span>)</a>
<a class="sourceLine" id="cb37-13" data-line-number="13"></a>
<a class="sourceLine" id="cb37-14" data-line-number="14">data_false_positives =<span class="st"> </span>res_sensitivity<span class="op">$</span>specificity <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-15" data-line-number="15"><span class="st">  </span><span class="kw">group_by</span>(method, cell_type) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-16" data-line-number="16"><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">fp_prediction =</span> <span class="kw">mean</span>(estimate)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-17" data-line-number="17"><span class="st">  </span><span class="kw">inner_join</span>(method_names) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-18" data-line-number="18"><span class="st">  </span><span class="kw">inner_join</span>(cell_type_names)</a>
<a class="sourceLine" id="cb37-19" data-line-number="19"></a>
<a class="sourceLine" id="cb37-20" data-line-number="20">fun_breaks2 =<span class="st"> </span><span class="cf">function</span>(limits) {</a>
<a class="sourceLine" id="cb37-21" data-line-number="21">  breaks =<span class="st"> </span><span class="kw">signif</span>(<span class="kw">max</span>(limits) <span class="op">*</span><span class="st"> </span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>),<span class="dv">1</span>)</a>
<a class="sourceLine" id="cb37-22" data-line-number="22">  <span class="kw">names</span>(breaks) =<span class="st"> </span><span class="kw">attr</span>(breaks, <span class="st">&quot;labels&quot;</span>)</a>
<a class="sourceLine" id="cb37-23" data-line-number="23">  breaks</a>
<a class="sourceLine" id="cb37-24" data-line-number="24">}</a>
<a class="sourceLine" id="cb37-25" data-line-number="25"></a>
<a class="sourceLine" id="cb37-26" data-line-number="26">data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb37-27" data-line-number="27"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>frac_immune_cells, <span class="dt">y=</span>mean)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-28" data-line-number="28"><span class="st">    </span><span class="kw">geom_ribbon</span>(<span class="kw">aes</span>(<span class="dt">ymin=</span>mean<span class="op">-</span>ci, <span class="dt">ymax=</span>mean<span class="op">+</span>ci), <span class="dt">alpha=</span>.<span class="dv">2</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-29" data-line-number="29"><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size=</span>.<span class="dv">2</span>, <span class="dt">shape=</span><span class="dv">4</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-30" data-line-number="30"><span class="st">    </span><span class="co"># geom_errorbar(aes(ymin=mean-ci, ymax=mean+ci)) + </span></a>
<a class="sourceLine" id="cb37-31" data-line-number="31"><span class="st">    </span><span class="kw">panel_border</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-32" data-line-number="32"><span class="st">    </span><span class="kw">facet_grid</span>(method_name <span class="op">~</span><span class="st"> </span>cell_type_name, <span class="dt">scales =</span> <span class="st">&quot;free_y&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-33" data-line-number="33"><span class="st">    </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(.<span class="dv">1</span>, <span class="fl">.3</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb37-34" data-line-number="34"><span class="st">    </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks=</span>fun_breaks2) <span class="op">+</span></a>
<a class="sourceLine" id="cb37-35" data-line-number="35"><span class="st">    </span><span class="kw">geom_vline</span>(<span class="dt">data=</span>data_detection_limit, <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">xintercept=</span>min_frac, <span class="dt">colour=</span><span class="st">&quot;detection limit&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-36" data-line-number="36"><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">data=</span>data_false_positives, <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">yintercept=</span>fp_prediction, <span class="dt">colour=</span><span class="st">&quot;false positive fraction&quot;</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-37" data-line-number="37"><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;average estimate&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-38" data-line-number="38"><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;fraction of spike-in cells&quot;</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb37-39" data-line-number="39"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;top&quot;</span>, <span class="dt">strip.text=</span><span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">6</span>), </a>
<a class="sourceLine" id="cb37-40" data-line-number="40">          <span class="dt">panel.spacing =</span> <span class="kw">unit</span>(<span class="dv">1</span>, <span class="st">&quot;mm&quot;</span>), <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">8</span>)) </a></code></pre></div>
<pre><code>## Warning: Removed 3 rows containing missing values (geom_vline).</code></pre>
<p><img src="_main_files/figure-html/detection-limit-fp-1.png" width="662.4" /></p>
</div>
<div id="migration-charts-for-spillover-analysis" class="section level2">
<h2><span class="header-section-number">8.3</span> Migration charts for Spillover analysis</h2>
<div class="figure"><span id="fig:unnamed-chunk-81"></span>
<img src="_main_files/figure-html/unnamed-chunk-81-1.png" alt="Migration chart figure for paper. " width="768" />
<p class="caption">
Figure 8.2: Migration chart figure for paper.
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="spillover-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
