<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Immune Deconvolution Benchmark</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Immune Deconvolution Benchmark" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Immune Deconvolution Benchmark" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm">


<meta name="date" content="2018-05-08">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="validitiy-of-single-cell-rna-seq-as-reference.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#conceptual-differences-between-methods"><i class="fa fa-check"></i><b>1.1</b> Conceptual differences between methods</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#gene-set-based-methods-allow-comparison-between-samples-but-not-between-cell-types"><i class="fa fa-check"></i><b>1.1.1</b> Gene-set based methods (allow comparison between samples, but not between cell types)</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#deconvolution-based-methods-relative-fractions-allow-comparisons-between-cell-types-but-not-samples"><i class="fa fa-check"></i><b>1.1.2</b> Deconvolution-based methods (relative fractions, allow comparisons between cell types, but not samples)</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#deconvolution-based-methods-absolute-fractions-allow-comparisons-between-cell-types-and-samples"><i class="fa fa-check"></i><b>1.1.3</b> Deconvolution-based methods (absolute fractions, allow comparisons between cell types and samples)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.1</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#validation-data-rnaseq-matched-gold-standard"><i class="fa fa-check"></i><b>2.2</b> Validation data (RNAseq + matched gold standard)</a><ul>
<li class="chapter" data-level="2.2.1" data-path="input-data.html"><a href="input-data.html#hoeck-data"><i class="fa fa-check"></i><b>2.2.1</b> Hoeck data</a></li>
<li class="chapter" data-level="2.2.2" data-path="input-data.html"><a href="input-data.html#schelker-data"><i class="fa fa-check"></i><b>2.2.2</b> Schelker data</a></li>
<li class="chapter" data-level="2.2.3" data-path="input-data.html"><a href="input-data.html#racle-data"><i class="fa fa-check"></i><b>2.2.3</b> Racle data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-single-cell-sequencing-for-counting-immune-cells"><i class="fa fa-check"></i><b>3.1</b> Validity of single cell sequencing for counting immune cells</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data"><i class="fa fa-check"></i><b>3.2</b> Validity of simulated bulk tissues from single cell sequencing data</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-samplese-to-bulk-samples"><i class="fa fa-check"></i><b>3.2.1</b> Compare simululated samplese to bulk samples</a></li>
<li class="chapter" data-level="3.2.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-the-predicted-fractions"><i class="fa fa-check"></i><b>3.2.2</b> Compare the predicted fractions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>4</b> Spillover analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.1</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="4.1.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#run-timer"><i class="fa fa-check"></i><b>4.1.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#results"><i class="fa fa-check"></i><b>4.2</b> Results</a><ul>
<li class="chapter" data-level="4.2.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#calculate-signal-to-noise-ratio-for-each-cell-type"><i class="fa fa-check"></i><b>4.2.1</b> Calculate signal to noise ratio for each cell type</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="mixing-study.html"><a href="mixing-study.html"><i class="fa fa-check"></i><b>5</b> Mixing study</a><ul>
<li class="chapter" data-level="5.1" data-path="mixing-study.html"><a href="mixing-study.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>5.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="5.2" data-path="mixing-study.html"><a href="mixing-study.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>5.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="5.3" data-path="mixing-study.html"><a href="mixing-study.html#run-the-deconvolution-1"><i class="fa fa-check"></i><b>5.3</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="5.3.1" data-path="mixing-study.html"><a href="mixing-study.html#run-timer-1"><i class="fa fa-check"></i><b>5.3.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="mixing-study.html"><a href="mixing-study.html#results-1"><i class="fa fa-check"></i><b>5.4</b> Results</a><ul>
<li class="chapter" data-level="5.4.1" data-path="mixing-study.html"><a href="mixing-study.html#calculate-other-cells-for-all-absolute-methods"><i class="fa fa-check"></i><b>5.4.1</b> Calculate “Other cells” for all absolute methods</a></li>
<li class="chapter" data-level="5.4.2" data-path="mixing-study.html"><a href="mixing-study.html#correlation-plots"><i class="fa fa-check"></i><b>5.4.2</b> Correlation plots</a></li>
<li class="chapter" data-level="5.4.3" data-path="mixing-study.html"><a href="mixing-study.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>5.4.3</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="5.4.4" data-path="mixing-study.html"><a href="mixing-study.html#absolute-error-for-all-qualifying-methods"><i class="fa fa-check"></i><b>5.4.4</b> Absolute error for all qualifying methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>6</b> Validation with real data</a><ul>
<li class="chapter" data-level="6.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#run-the-deconvolution-2"><i class="fa fa-check"></i><b>6.1</b> Run the deconvolution</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>7</b> Conclusions</a></li>
<li class="chapter" data-level="8" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>8</b> Materials and Methods</a><ul>
<li class="chapter" data-level="8.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#data-acquisition"><i class="fa fa-check"></i><b>8.1</b> Data acquisition</a></li>
<li class="chapter" data-level="8.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#validation-datasets-flow-cytometry-scrnaseq-imageing-matched-rna-seq"><i class="fa fa-check"></i><b>8.2</b> Validation datasets (Flow cytometry, scRNAseq, Imageing + matched RNA seq)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Immune Deconvolution Benchmark</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="input-data" class="section level1">
<h1><span class="header-section-number">2</span> Input data</h1>
<div id="single-cell-data-for-simulated-mixtures" class="section level2">
<h2><span class="header-section-number">2.1</span> Single cell data for simulated mixtures</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;../data/single_cell_schelker/single_cell_schelker.rda&quot;</span>)</code></pre></div>
<p>In this study, we make use of the single cell dataset curated by <span class="citation">(Schelker et al. <a href="#ref-Schelker2017">2017</a>)</span>. They aggregated single cell sequencing data from different sources resulting in a set of ~12,000 single cells. They classified the cells using at set of 45 marker genes into 11 categories: 2 cancer types (Melanoma cells, Ovarian carcinoma cells), 7 immune cells (B cells, CD8+ T cells, CD4+ T cells, Macrophages/Monocytes, regulatory T cells, Dendritic cells, Natural killer cells), 2 other cells (Cancer associated fibroblasts, Endothelial cells) and <em>Unknown cells</em> which could not have been classified unambiguously.</p>
<p>Unknown cells are excluded from the downstream analysis.</p>
<table>
<caption>the ~12,000 samples by cell type</caption>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B cells</td>
<td align="right">1044</td>
</tr>
<tr class="even">
<td align="left">Cancer associated fibroblasts</td>
<td align="right">132</td>
</tr>
<tr class="odd">
<td align="left">CD4+ T cells</td>
<td align="right">3023</td>
</tr>
<tr class="even">
<td align="left">CD8+ T cells</td>
<td align="right">1884</td>
</tr>
<tr class="odd">
<td align="left">Dendritic cells</td>
<td align="right">166</td>
</tr>
<tr class="even">
<td align="left">Endothelial cells</td>
<td align="right">71</td>
</tr>
<tr class="odd">
<td align="left">Macrophages/Monocytes</td>
<td align="right">2934</td>
</tr>
<tr class="even">
<td align="left">Melanoma cells</td>
<td align="right">1310</td>
</tr>
<tr class="odd">
<td align="left">Natural killer cells</td>
<td align="right">425</td>
</tr>
<tr class="even">
<td align="left">Ovarian carcinoma cells</td>
<td align="right">300</td>
</tr>
<tr class="odd">
<td align="left">regulatory T cells</td>
<td align="right">145</td>
</tr>
<tr class="even">
<td align="left">Unknown</td>
<td align="right">325</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:unnamed-chunk-4"></span>
<img src="_main_files/figure-html/unnamed-chunk-4-1.png" alt="tSNE-clustering of the ~12000 single cells from [@Schelker2017]. " width="960" />
<p class="caption">
Figure 2.1: tSNE-clustering of the ~12000 single cells from <span class="citation">(Schelker et al. <a href="#ref-Schelker2017">2017</a>)</span>.
</p>
</div>
</div>
<div id="validation-data-rnaseq-matched-gold-standard" class="section level2">
<h2><span class="header-section-number">2.2</span> Validation data (RNAseq + matched gold standard)</h2>
<p>e.g. IHC, Flow cytometry, single cell</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">datasets =<span class="st"> </span><span class="kw">read_excel</span>(<span class="st">&quot;../tables/validation-datasets.xlsx&quot;</span>)
<span class="kw">kable</span>(datasets)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Study</th>
<th align="right"># Samples</th>
<th align="left">Sample Type</th>
<th align="left">Gene Expression by</th>
<th align="left">Validation by</th>
<th align="left">Cell types</th>
<th align="left">Cited by</th>
<th align="left">Comments</th>
<th align="left">Got TPM</th>
<th align="left">Got Validation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Finotello 2017</td>
<td align="right">9</td>
<td align="left">CRC</td>
<td align="left">RNAseq</td>
<td align="left">IHC</td>
<td align="left">CD3, CD8, CD45RO, FOXP3, PD1</td>
<td align="left">NA</td>
<td align="left">Available upon publication</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Finotello 2017</td>
<td align="right">9</td>
<td align="left">PBMC/PMN</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry</td>
<td align="left">B, Mono, Neu, NK, CD4, CD8, Treg, DC</td>
<td align="left">NA</td>
<td align="left">Available upon publication</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">Hoeck 2015</td>
<td align="right">8</td>
<td align="left">PBMC/PMN</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry</td>
<td align="left">T, B, NK, MDC, Mono, Neu</td>
<td align="left">Finotello 2017</td>
<td align="left">Available through personal communicaion</td>
<td align="left">x</td>
<td align="left">x</td>
</tr>
<tr class="even">
<td align="left">Johnson 2015</td>
<td align="right">31</td>
<td align="left">metastatic melanoma</td>
<td align="left">RNAseq</td>
<td align="left">IHC</td>
<td align="left">CD8, CD4, FOXP3+</td>
<td align="left">Finotello 2017</td>
<td align="left">How to get?</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">Racle 2017</td>
<td align="right">4</td>
<td align="left">metastatic melanoma</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry</td>
<td align="left">B, CD4, CD8, NK, Melanoma/other</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">x</td>
<td align="left">x</td>
</tr>
<tr class="even">
<td align="left">Zimmermann 2016</td>
<td align="right">12</td>
<td align="left">PBMC before influenza vacc</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry</td>
<td align="left">B, CD4, CD8, NK, Mono, Other</td>
<td align="left">Racle 2017</td>
<td align="left">obtained through Racle 2017</td>
<td align="left">NA</td>
<td align="left">x</td>
</tr>
<tr class="odd">
<td align="left">Schelker 2017</td>
<td align="right">3</td>
<td align="left">ovarian cancer ascites</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry + Single Cell RNAseq</td>
<td align="left">CD8, CD4, B, NK, DC, Mono, …</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">x</td>
<td align="left">x</td>
</tr>
<tr class="even">
<td align="left">Chung 2017</td>
<td align="right">11</td>
<td align="left">BRCA</td>
<td align="left">RNAseq</td>
<td align="left">Single Cell RNAseq + IHC for B and T</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<div id="hoeck-data" class="section level3">
<h3><span class="header-section-number">2.2.1</span> Hoeck data</h3>
<p>8 PBMC, RNAseq+flow cytometry</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;../data/hoeck/HoekPBMC_mixture.RData&quot;</span>)
hoeck_mixture =<span class="st"> </span>mix.mat
<span class="kw">load</span>(<span class="st">&quot;../data/hoeck/HoekPBMC_gtruth.RData&quot;</span>)
hoeck_ref =<span class="st"> </span>RefData</code></pre></div>
</div>
<div id="schelker-data" class="section level3">
<h3><span class="header-section-number">2.2.2</span> Schelker data</h3>
<p>Load the data and merge replicates</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ovarian_bulk_replicates =<span class="st"> </span><span class="kw">read_xls</span>(<span class="st">&quot;../data/single_cell_schelker/ascites_bulk_samples.xls&quot;</span>) 

samples =<span class="st"> </span><span class="kw">list</span>(<span class="st">&quot;7873M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7873_1&quot;</span>, <span class="st">&quot;ascites_7873_2&quot;</span>),
               <span class="st">&quot;7882M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7882_1&quot;</span>, <span class="st">&quot;ascites_7882_2&quot;</span>),
               <span class="st">&quot;7892M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7892_1&quot;</span>, <span class="st">&quot;ascites_7892_2&quot;</span>))

ovarian_bulk =<span class="st"> </span><span class="kw">lapply</span>(samples, <span class="cf">function</span>(cols) {
  <span class="kw">apply</span>(ovarian_bulk_replicates[,cols], <span class="dv">1</span>, mean)
}) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_cols</span>() 

ovarian_bulk <span class="op">%&lt;&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">gene_symbol =</span> ovarian_bulk_replicates<span class="op">$</span>Row)

ovarian_bulk_mat =<span class="st"> </span>ovarian_bulk <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;gene_symbol&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">as.matrix</span>()</code></pre></div>
<p>Load the ‘ground truth’</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">true_cell_count =<span class="st"> </span><span class="kw">pData</span>(single_cell_schelker2) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(source <span class="op">==</span><span class="st"> &quot;ascites&quot;</span>, donor <span class="op">%in%</span><span class="st"> </span><span class="kw">colnames</span>(ovarian_bulk)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(donor, cell_type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">count</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">rename</span>(<span class="dt">cell_count=</span>n) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">true_fraction=</span>cell_count<span class="op">/</span><span class="kw">sum</span>(cell_count)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">arrange</span>(cell_type, donor)

total_cells =<span class="st"> </span>true_cell_count <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="kw">sum</span>(cell_count))

total_cells</code></pre></div>
<pre><code>## # A tibble: 3 x 2
##   donor `sum(cell_count)`
##   &lt;chr&gt;             &lt;int&gt;
## 1 7873M               864
## 2 7882M               902
## 3 7892M               773</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">true_cell_count</code></pre></div>
<pre><code>## # A tibble: 25 x 4
## # Groups:   donor [3]
##    donor cell_type                     cell_count true_fraction
##    &lt;chr&gt; &lt;chr&gt;                              &lt;int&gt;         &lt;dbl&gt;
##  1 7873M B cells                                5       0.00579
##  2 7882M B cells                                4       0.00443
##  3 7892M B cells                                5       0.00647
##  4 7873M Cancer associated fibroblasts          9       0.0104 
##  5 7882M Cancer associated fibroblasts         11       0.0122 
##  6 7892M Cancer associated fibroblasts         16       0.0207 
##  7 7873M CD4+ T cells                          29       0.0336 
##  8 7882M CD4+ T cells                         137       0.152  
##  9 7892M CD4+ T cells                         100       0.129  
## 10 7873M CD8+ T cells                           4       0.00463
## # ... with 15 more rows</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">schelker_ref =<span class="st"> </span>true_cell_count <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="dt">sample=</span>donor, cell_type, true_fraction)</code></pre></div>
</div>
<div id="racle-data" class="section level3">
<h3><span class="header-section-number">2.2.3</span> Racle data</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ensg_hgnc =<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;../data/ensemble_hgnc.txt&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">rename</span>(<span class="dt">gene_id =</span> <span class="st">`</span><span class="dt">Gene stable ID</span><span class="st">`</span>, <span class="dt">hgnc_symbol=</span><span class="st">`</span><span class="dt">HGNC symbol</span><span class="st">`</span>)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   `Gene stable ID` = col_character(),
##   `HGNC symbol` = col_character()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">racle_cyto =<span class="st"> </span><span class="kw">read_excel</span>(<span class="st">&quot;../data/racle/racle2017_flow_cytometry.xlsx&quot;</span>)
racle_files =<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;../data/racle/GSE93722_RAW/&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)
racle_identifiers =<span class="st"> </span><span class="kw">str_extract</span>(racle_files, <span class="st">&quot;LAU</span><span class="ch">\\</span><span class="st">d+&quot;</span>)
racle_expression_raw =<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">list.files</span>(<span class="st">&quot;../data/racle/GSE93722_RAW/&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>), read_tsv) </code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   gene_id = col_character(),
##   `transcript_id(s)` = col_character(),
##   length = col_double(),
##   effective_length = col_double(),
##   expected_count = col_double(),
##   TPM = col_double(),
##   FPKM = col_double()
## )</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   gene_id = col_character(),
##   `transcript_id(s)` = col_character(),
##   length = col_double(),
##   effective_length = col_double(),
##   expected_count = col_double(),
##   TPM = col_double(),
##   FPKM = col_double()
## )
## Parsed with column specification:
## cols(
##   gene_id = col_character(),
##   `transcript_id(s)` = col_character(),
##   length = col_double(),
##   effective_length = col_double(),
##   expected_count = col_double(),
##   TPM = col_double(),
##   FPKM = col_double()
## )
## Parsed with column specification:
## cols(
##   gene_id = col_character(),
##   `transcript_id(s)` = col_character(),
##   length = col_double(),
##   effective_length = col_double(),
##   expected_count = col_double(),
##   TPM = col_double(),
##   FPKM = col_double()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gene_ids =<span class="st"> </span>racle_expression_raw[[<span class="dv">1</span>]]<span class="op">$</span>gene_id
racle_tpm =<span class="st"> </span><span class="kw">sapply</span>(racle_expression_raw, <span class="cf">function</span>(df) {<span class="kw">pull</span>(df, <span class="st">&quot;TPM&quot;</span>)}) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.tibble</span>()
<span class="kw">colnames</span>(racle_tpm) =<span class="st"> </span>racle_identifiers
racle_tpm<span class="op">$</span>gene_id =<span class="st"> </span>gene_ids
racle_expression =<span class="st"> </span>racle_tpm <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">inner_join</span>(ensg_hgnc) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(hgnc_symbol)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>gene_id) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(hgnc_symbol) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise_all</span>(sum)</code></pre></div>
<pre><code>## Joining, by = &quot;gene_id&quot;</code></pre>

</div>
</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Schelker2017">
<p>Schelker, Max, Sonia Feau, Jinyan Du, Nav Ranu, Edda Klipp, Gavin MacBeath, Birgit Schoeberl, and Andreas Raue. 2017. “Estimation of immune cell content in tumour tissue using single-cell RNA-seq data.” <em>Nature Communications</em> 8 (1): 2032. doi:<a href="https://doi.org/10.1038/s41467-017-02289-3">10.1038/s41467-017-02289-3</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="validitiy-of-single-cell-rna-seq-as-reference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
