<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="manuscript in preparation">
<meta name="author" content="Gregor Sturm^{1,2}">
<meta name="author" content="Tatsiana Aneichyk^1">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="validitiy-of-single-cell-rna-seq-as-reference.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-reference-samples"><i class="fa fa-check"></i><b>2.3</b> Immune cell reference samples</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#pbmc-samples-from-hoek2015"><i class="fa fa-check"></i><b>2.4</b> 8 PBMC samples from <span class="citation">Hoek et al. (<span>2015</span>)</span></a></li>
<li class="chapter" data-level="2.5" data-path="input-data.html"><a href="input-data.html#ovarian-cancer-ascites-samples-from-schelker2017"><i class="fa fa-check"></i><b>2.5</b> 3 ovarian cancer ascites samples from <span class="citation">Schelker et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.6" data-path="input-data.html"><a href="input-data.html#metastatic-melanoma-samples-from-epic2017"><i class="fa fa-check"></i><b>2.6</b> 4 metastatic melanoma samples from <span class="citation">Racle et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.7" data-path="input-data.html"><a href="input-data.html#data-sanity-checks"><i class="fa fa-check"></i><b>2.7</b> Data sanity checks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-to-genuine-bulk-samples"><i class="fa fa-check"></i><b>3.1</b> Compare simululated to genuine bulk samples</a><ul>
<li class="chapter" data-level="3.1.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#log-ratio"><i class="fa fa-check"></i><b>3.1.1</b> Log ratio</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#enrichment-analysis-test-for-systematic-bias"><i class="fa fa-check"></i><b>3.2</b> Enrichment analysis: Test for systematic Bias</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#immune-relevant-genes-are-consistent-between-sc-and-bulk-data."><i class="fa fa-check"></i><b>3.2.1</b> Immune-relevant genes are consistent between sc and bulk data.</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-predicted-fractions"><i class="fa fa-check"></i><b>3.3</b> Compare predicted fractions</a></li>
<li class="chapter" data-level="3.4" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlation-with-immune-reference-samples."><i class="fa fa-check"></i><b>3.4</b> Correlation with immune reference samples.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html"><i class="fa fa-check"></i><b>4</b> Simulation benchmark</a><ul>
<li class="chapter" data-level="4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>4.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>4.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.3</b> Run the deconvolution</a></li>
<li class="chapter" data-level="4.4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#results"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#correlation-plots"><i class="fa fa-check"></i><b>4.4.1</b> Correlation plots</a></li>
<li class="chapter" data-level="4.4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>4.4.2</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="4.4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#deviation-from-slope-xy-for-all-absolute-methods"><i class="fa fa-check"></i><b>4.4.3</b> Deviation from slope (x=y) for all absolute methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>5</b> Validation with real data</a><ul>
<li class="chapter" data-level="5.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between--and-within-sample-comparisons"><i class="fa fa-check"></i><b>5.1</b> Between- and within-sample comparisons</a></li>
<li class="chapter" data-level="5.2" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#within-sample-comparison"><i class="fa fa-check"></i><b>5.2</b> Within-sample comparison</a></li>
<li class="chapter" data-level="5.3" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between-sample-comparisons"><i class="fa fa-check"></i><b>5.3</b> Between-sample comparisons</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html"><i class="fa fa-check"></i><b>6</b> Detection limit and false positive predictions</a><ul>
<li class="chapter" data-level="6.1" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#predictions-with-increasing-immune-cell-content"><i class="fa fa-check"></i><b>6.1</b> Predictions with increasing immune cell content</a></li>
<li class="chapter" data-level="6.2" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#detection-limit"><i class="fa fa-check"></i><b>6.2</b> Detection Limit</a></li>
<li class="chapter" data-level="6.3" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#false-positive-predictions"><i class="fa fa-check"></i><b>6.3</b> False-positive predictions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>7</b> Spillover Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#complete-spillover-matrix"><i class="fa fa-check"></i><b>7.1</b> Complete Spillover Matrix</a></li>
<li class="chapter" data-level="7.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-signal-to-noise-ratio"><i class="fa fa-check"></i><b>7.2</b> Summary figure: Signal to noise ratio</a></li>
<li class="chapter" data-level="7.3" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-migration-charts"><i class="fa fa-check"></i><b>7.3</b> Summary figure: migration charts</a></li>
<li class="chapter" data-level="7.4" data-path="spillover-analysis.html"><a href="spillover-analysis.html#investigate-marker-genes"><i class="fa fa-check"></i><b>7.4</b> Investigate marker genes</a></li>
<li class="chapter" data-level="7.5" data-path="spillover-analysis.html"><a href="spillover-analysis.html#check-expression-of-markers-in-detection-limit-simulation-dataset."><i class="fa fa-check"></i><b>7.5</b> Check expression of markers in “detection limit” simulation dataset.</a><ul>
<li class="chapter" data-level="7.5.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#derive-marker-genes-from-quantiseqepic-signature-matrices."><i class="fa fa-check"></i><b>7.5.1</b> Derive marker genes from quanTIseq/EPIC signature matrices.</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="spillover-analysis.html"><a href="spillover-analysis.html#deconvolution-results-before-and-after"><i class="fa fa-check"></i><b>7.6</b> Deconvolution results before and after</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html"><i class="fa fa-check"></i><b>8</b> Creating publication ready figures</a><ul>
<li class="chapter" data-level="8.1" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#benchmark-results-detection-limit-false-positives-and-absolute-deviation"><i class="fa fa-check"></i><b>8.1</b> Benchmark results, detection limit, false positives and absolute deviation</a></li>
<li class="chapter" data-level="8.2" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#migration-charts-for-spillover-analysis"><i class="fa fa-check"></i><b>8.2</b> Migration charts for Spillover analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="input-data" class="section level1">
<h1><span class="header-section-number">2</span> Input data</h1>
<p>In this chapter, we load and preprocess the datasets we use in this study</p>
<table>
<colgroup>
<col width="44%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th>Dataset</th>
<th>Reference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>A dataset of more than 11,000 single cancer and immune cells, classified by cell type</td>
<td><span class="citation">Schelker et al. (<a href="#ref-Schelker2017">2017</a>)</span></td>
</tr>
<tr class="even">
<td>50 Immune cell reference samples from 5 studies</td>
<td>Curated by <span class="citation">Finotello et al. (<a href="#ref-quantiseq2017">2017</a>)</span></td>
</tr>
<tr class="odd">
<td>3 ovarian cancer ascites samples (RNAseq + FACS)</td>
<td><span class="citation">Schelker et al. (<a href="#ref-Schelker2017">2017</a>)</span></td>
</tr>
<tr class="even">
<td>8 PBMC samples (RNAseq + FACS)</td>
<td><span class="citation">Hoek et al. (<a href="#ref-Hoek2015">2015</a>)</span></td>
</tr>
<tr class="odd">
<td>4 metastatic melanoma samplese (RNAseq + FACS)</td>
<td><span class="citation">Racle et al. (<a href="#ref-EPIC2017">2017</a>)</span></td>
</tr>
</tbody>
</table>
<div id="cell-type-hierarchy" class="section level2">
<h2><span class="header-section-number">2.1</span> cell type hierarchy</h2>
<p>We use a hierachy of immune cell types to map the cell types between different methods and datasets.
The following figure shows this hierarchy visualized as a tree</p>
<div class="figure"><span id="fig:hierarchy"></span>
<div id="htmlwidget-bc5282a8a655db9f081b" style="width:768px;height:768px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-bc5282a8a655db9f081b">{"x":{"diagram":"digraph {\n\ngraph [rankdir = \"LR\"]\n\n\n\n  \"1\" [label = \"cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"2\" [label = \"immune cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"3\" [label = \"lymphoid\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"4\" [label = \"T cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"5\" [label = \"T cell CD8+\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"6\" [label = \"T cell CD8+ naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"7\" [label = \"T cell CD8+ memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"8\" [label = \"T cell CD4+\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"9\" [label = \"T cell CD4+ (non-regulatory)\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"10\" [label = \"T cell CD4+ naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"11\" [label = \"T cell CD4+ memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"12\" [label = \"T cell CD4+ memory resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"13\" [label = \"T cell CD4+ memory activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"14\" [label = \"T cell follicular helper\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"15\" [label = \"T cell regulatory (Tregs)\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"16\" [label = \"T cell gamma delta\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"17\" [label = \"B cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"18\" [label = \"B cell naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"19\" [label = \"B cell memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"20\" [label = \"B cell plasma\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"21\" [label = \"NK cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"22\" [label = \"NK cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"23\" [label = \"NK cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"24\" [label = \"myleoid\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"25\" [label = \"Neutrophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"26\" [label = \"Eosinophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"27\" [label = \"Mast cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"28\" [label = \"Mast cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"29\" [label = \"Mast cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"30\" [label = \"Monocytic lineage\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"31\" [label = \"Macrophage/Monocyte\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"32\" [label = \"Monocyte\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"33\" [label = \"Macrophage\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"34\" [label = \"Macrophage M0\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"35\" [label = \"Macrophage M1\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"36\" [label = \"Macrophage M2\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"37\" [label = \"Dendritic cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"38\" [label = \"Dendritic cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"39\" [label = \"Dendritic cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"40\" [label = \"other cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"41\" [label = \"Endothelial cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"42\" [label = \"Cancer associated fibroblast\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"43\" [label = \"cancer cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"44\" [label = \"Ovarian carcinoma cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"45\" [label = \"Melanoma cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n\"1\"->\"2\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"1\"->\"40\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"2\"->\"3\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"2\"->\"24\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"40\"->\"41\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"40\"->\"42\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"40\"->\"43\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"4\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"17\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"21\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"24\"->\"25\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"24\"->\"26\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"24\"->\"27\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"24\"->\"30\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"43\"->\"44\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"43\"->\"45\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"5\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"8\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"16\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"17\"->\"18\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"17\"->\"19\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"17\"->\"20\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"21\"->\"22\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"21\"->\"23\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"27\"->\"28\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"27\"->\"29\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"30\"->\"31\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"30\"->\"37\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"6\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"7\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"8\"->\"9\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"8\"->\"15\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"31\"->\"32\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"31\"->\"33\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"37\"->\"38\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"37\"->\"39\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"9\"->\"10\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"9\"->\"11\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"9\"->\"14\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"33\"->\"34\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"33\"->\"35\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"33\"->\"36\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"11\"->\"12\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"11\"->\"13\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n}","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 2.1: Hierarchy of immune cell types used for mapping cell types between methods and datasets.
</p>
</div>
</div>
<div id="single-cell-data-for-simulated-mixtures" class="section level2">
<h2><span class="header-section-number">2.2</span> Single cell data for simulated mixtures</h2>
<p>In this study, we make use of the single cell dataset curated by <span class="citation">Schelker et al. (<a href="#ref-Schelker2017">2017</a>)</span>. They aggregated single cell sequencing data from different sources resulting in a set of more than 11,000 single cells. They classified the cells using at set of 45 marker genes into 12 categories:</p>
<ul>
<li>2 cancer types (Melanoma cell, Ovarian carcinoma cell),</li>
<li>7 immune cells (B cell, T cell CD8+, T cell CD4+ (non-regulatory), Macrophage/Monocyte, T cell regulatory (Tregs), Dendritic cell, NK cell),</li>
<li>2 other cells (Cancer associated fibroblast, Endothelial cell) and</li>
<li><em>Unknown cells</em> which could not have been classified unambiguously.</li>
</ul>
<p>Unknown cells are excluded from the downstream analysis.</p>
<p>The dataset consists of single cells from PBMC, melanoma and ovarian cancer ascites.
As we are interested in the deconvolution of cancer samples, we exclude the PBMC cells from all downstream analyses.</p>
<table>
<caption><span id="tab:unnamed-chunk-2">Table 2.1: </span>The 11434 single cells by cell type</caption>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B cell</td>
<td align="right">646</td>
</tr>
<tr class="even">
<td align="left">Cancer associated fibroblast</td>
<td align="right">132</td>
</tr>
<tr class="odd">
<td align="left">Dendritic cell</td>
<td align="right">140</td>
</tr>
<tr class="even">
<td align="left">Endothelial cell</td>
<td align="right">71</td>
</tr>
<tr class="odd">
<td align="left">Macrophage/Monocyte</td>
<td align="right">2227</td>
</tr>
<tr class="even">
<td align="left">Melanoma cell</td>
<td align="right">1310</td>
</tr>
<tr class="odd">
<td align="left">NK cell</td>
<td align="right">198</td>
</tr>
<tr class="even">
<td align="left">Ovarian carcinoma cell</td>
<td align="right">300</td>
</tr>
<tr class="odd">
<td align="left">PBMC</td>
<td align="right">3942</td>
</tr>
<tr class="even">
<td align="left">T cell CD4+ (non-regulatory)</td>
<td align="right">1196</td>
</tr>
<tr class="odd">
<td align="left">T cell CD8+</td>
<td align="right">1130</td>
</tr>
<tr class="even">
<td align="left">T cell regulatory (Tregs)</td>
<td align="right">142</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:unnamed-chunk-3"></span>
<img src="_main_files/figure-html/unnamed-chunk-3-1.png" alt="tSNE-clustering of the ~12,000 single cells from @Schelker2017. " width="1248" />
<p class="caption">
Figure 2.2: tSNE-clustering of the ~12,000 single cells from <span class="citation">Schelker et al. (<a href="#ref-Schelker2017">2017</a>)</span>.
</p>
</div>
</div>
<div id="immune-cell-reference-samples" class="section level2">
<h2><span class="header-section-number">2.3</span> Immune cell reference samples</h2>
<p>RNAseq samples of pure immune cells of 10 types from 5 studies curated by <span class="citation">Finotello et al. (<a href="#ref-quantiseq2017">2017</a>)</span>.</p>
<table>
<caption>
<span id="tab:mytab">Table 2.2: </span>List of immune cell reference samples
</caption>
</table>
<div id="htmlwidget-c8bf611ac41e2d27ba76" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c8bf611ac41e2d27ba76">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50"],["SRR1551070","SRR1551056","SRR1551049","SRR1550988","SRR1740034","SRR1740062","SRR1740038","SRR1740066","SRR1740039","SRR1740040","SRR1740041","SRR1740067","SRR1740068","SRR1740069","SRR452330","SRR452329","SRR452333","SRR452332","SRR452331","SRR1551069","SRR1551055","SRR1551048","SRR1550987","SRR1740042","SRR1740070","SRR1551068","SRR1551054","SRR1551047","SRR1550986","SRR1740046","SRR1740074","SRR1551073","SRR1551059","SRR1551052","SRR1550991","SRR1740050","SRR1740078","SRR1551071","SRR1551057","SRR1551050","SRR1550989","SRR1551072","SRR1551058","SRR1551051","SRR1550990","ERR431583","ERR431606","ERR431609","ERR431601","ERR431589"],["Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Beyer et al.","Beyer et al.","Beyer et al.","Beyer et al.","Beyer et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Hoek et al.","Hoek et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Hoek et al.","Hoek et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Hoek et al.","Hoek et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","De Simone et al.","De Simone et al.","De Simone et al.","De Simone et al.","De Simone et al."],["B cell","B cell","B cell","B cell","B cell","B cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Macrophage M1","Macrophage M1","Macrophage M2","Macrophage M2","Macrophage M2","Monocyte","Monocyte","Monocyte","Monocyte","Monocyte","Monocyte","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","NK cell","NK cell","NK cell","NK cell","NK cell","NK cell","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD8+","T cell CD8+","T cell CD8+","T cell CD8+","T cell regulatory (Tregs)","T cell regulatory (Tregs)","T cell regulatory (Tregs)","T cell regulatory (Tregs)","T cell regulatory (Tregs)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>Study<\/th>\n      <th>cell_type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="pbmc-samples-from-hoek2015" class="section level2">
<h2><span class="header-section-number">2.4</span> 8 PBMC samples from <span class="citation">Hoek et al. (<a href="#ref-Hoek2015">2015</a>)</span></h2>
<table>
<caption>
<span id="tab:mytab">Table 2.2: </span>Flow cytometry estimates of Hoek et al.
</caption>
</table>
<div id="htmlwidget-7b78b2fdb2811b13a201" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7b78b2fdb2811b13a201">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40"],["HD30_PBMC_0","HD30_PBMC_1","HD30_PBMC_3","HD30_PBMC_7","HD31_PBMC_0","HD31_PBMC_1","HD31_PBMC_3","HD31_PBMC_7","HD30_PBMC_0","HD30_PBMC_1","HD30_PBMC_3","HD30_PBMC_7","HD31_PBMC_0","HD31_PBMC_1","HD31_PBMC_3","HD31_PBMC_7","HD30_PBMC_0","HD30_PBMC_1","HD30_PBMC_3","HD30_PBMC_7","HD31_PBMC_0","HD31_PBMC_1","HD31_PBMC_3","HD31_PBMC_7","HD30_PBMC_0","HD30_PBMC_1","HD30_PBMC_3","HD30_PBMC_7","HD31_PBMC_0","HD31_PBMC_1","HD31_PBMC_3","HD31_PBMC_7","HD30_PBMC_0","HD30_PBMC_1","HD30_PBMC_3","HD30_PBMC_7","HD31_PBMC_0","HD31_PBMC_1","HD31_PBMC_3","HD31_PBMC_7"],["Tcell","Tcell","Tcell","Tcell","Tcell","Tcell","Tcell","Tcell","mono","mono","mono","mono","mono","mono","mono","mono","Bcells","Bcells","Bcells","Bcells","Bcells","Bcells","Bcells","Bcells","DC","DC","DC","DC","DC","DC","DC","DC","NK","NK","NK","NK","NK","NK","NK","NK"],[0.514,0.457,0.58,0.578,0.59,0.623,0.654,0.636,0.155,0.214,0.153,0.135,0.181,0.17,0.175,0.144,0.043,0.048,0.042,0.066,0.091,0.093,0.061,0.097,0.0264,0.0251,0.0172,0.0184,0.0163,0.0152,0.0133,0.0161,0.178,0.168,0.148,0.132,0.045,0.034,0.049,0.036],["hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek","hoek"],["T cell","T cell","T cell","T cell","T cell","T cell","T cell","T cell","Macrophage/Monocyte","Macrophage/Monocyte","Macrophage/Monocyte","Macrophage/Monocyte","Macrophage/Monocyte","Macrophage/Monocyte","Macrophage/Monocyte","Macrophage/Monocyte","B cell","B cell","B cell","B cell","B cell","B cell","B cell","B cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","NK cell","NK cell","NK cell","NK cell","NK cell","NK cell","NK cell","NK cell"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>method_cell_type<\/th>\n      <th>true_fraction<\/th>\n      <th>method_dataset<\/th>\n      <th>cell_type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="ovarian-cancer-ascites-samples-from-schelker2017" class="section level2">
<h2><span class="header-section-number">2.5</span> 3 ovarian cancer ascites samples from <span class="citation">Schelker et al. (<a href="#ref-Schelker2017">2017</a>)</span></h2>
<p>Each sample has two technical replicates. We merge the two replicates by taking
the mean for each gene.</p>
<pre><code>## &lt;table&gt;
## &lt;caption&gt;(#tab:mytab)Flow cytometry estimates of Schelker et al. &lt;/caption&gt;
## &lt;/table&gt;</code></pre>
<div id="htmlwidget-a61e424180a07097cdfd" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a61e424180a07097cdfd">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30"],["7873M","7873M","7873M","7873M","7873M","7873M","7873M","7873M","7873M","7873M","7882M","7882M","7882M","7882M","7882M","7882M","7882M","7882M","7882M","7882M","7892M","7892M","7892M","7892M","7892M","7892M","7892M","7892M","7892M","7892M"],["NK cell","T cell CD4+ naive","T cell CD4+ memory","T cell regulatory (Tregs)","T cell CD8+ naive","T cell CD8+ memory","Macrophage/Monocyte","Neutrophil","B cell","Dendritic cell","NK cell","T cell CD4+ naive","T cell CD4+ memory","T cell regulatory (Tregs)","T cell CD8+ naive","T cell CD8+ memory","Macrophage/Monocyte","Neutrophil","B cell","Dendritic cell","NK cell","T cell CD4+ naive","T cell CD4+ memory","T cell regulatory (Tregs)","T cell CD8+ naive","T cell CD8+ memory","Macrophage/Monocyte","Neutrophil","B cell","Dendritic cell"],[0.02332,0.003432,0.074976,0.001496,0.005896,0.014344,0.678,0.0512,0.0021,0.0333,0.0330424,0.0071378,0.14837,0.0053734,0.0119498,0.0285512,0.329,0.19,0.00028,0.0262,0.0001918,0.007261,0.11371,0.001644,0.0091105,0.016166,null,0.213,0.00017,0.0165],["schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian","schelker_ovarian"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>cell_type<\/th>\n      <th>true_fraction<\/th>\n      <th>method_dataset<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>The samples have also been profiled by single cell RNA sequencing. The following table shows the
cell count for each sample.</p>
<table>
<caption><span id="tab:unnamed-chunk-6">Table 2.3: </span>Single cell count per ovarian cancer ascites sample.</caption>
<thead>
<tr class="header">
<th align="left">donor</th>
<th align="right">sum(cell_count)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">7873M</td>
<td align="right">864</td>
</tr>
<tr class="even">
<td align="left">7882M</td>
<td align="right">902</td>
</tr>
<tr class="odd">
<td align="left">7892M</td>
<td align="right">773</td>
</tr>
</tbody>
</table>
</div>
<div id="metastatic-melanoma-samples-from-epic2017" class="section level2">
<h2><span class="header-section-number">2.6</span> 4 metastatic melanoma samples from <span class="citation">Racle et al. (<a href="#ref-EPIC2017">2017</a>)</span></h2>
<table>
<caption>
<span id="tab:mytab">Table 2.2: </span>Flow cytometry estimates of Racle et al.
</caption>
</table>
<div id="htmlwidget-a2914097f1aff68ea197" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a2914097f1aff68ea197">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314"],["B cell","B cell","B cell","B cell","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD8+","T cell CD8+","T cell CD8+","T cell CD8+","NK cell","NK cell","NK cell","NK cell"],[0.1812,0.3248,0.0579,0.4667,0.0082,0.2315,0.0276,0.1815,0.0035,0.0582,0.0376,0.0454,0.005,0.0017,0.0017,0.0025]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>cell_type<\/th>\n      <th>true_fraction<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="data-sanity-checks" class="section level2">
<h2><span class="header-section-number">2.7</span> Data sanity checks</h2>
<p>Here, we plot the distributions of the different gene expression datasets to ensure
that everything looks like we expect it to, e.g. if all datasets are
on non-log scale.</p>
<div class="figure"><span id="fig:unnamed-chunk-7"></span>
<img src="_main_files/figure-html/unnamed-chunk-7-1.png" alt="Histogram of gene expression data of all datasets" width="960" />
<p class="caption">
Figure 2.3: Histogram of gene expression data of all datasets
</p>
</div>
<div class="figure"><span id="fig:unnamed-chunk-8"></span>
<img src="_main_files/figure-html/unnamed-chunk-8-1.png" alt="Histogram of log-tranformed gene expression data of all datasets" width="960" />
<p class="caption">
Figure 2.4: Histogram of log-tranformed gene expression data of all datasets
</p>
</div>
<p>The mean values of all datasets:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">mean</span>(<span class="kw">c</span>(<span class="kw">exprs</span>(single_cell_schelker<span class="op">$</span>eset)))</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw">mean</span>(<span class="kw">c</span>(racle<span class="op">$</span>expr_mat))</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw">mean</span>(<span class="kw">c</span>(schelker_ovarian<span class="op">$</span>expr_mat))</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">mean</span>(<span class="kw">c</span>(hoek<span class="op">$</span>expr_mat))</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw">mean</span>(<span class="kw">c</span>(immune_cell_reference<span class="op">$</span>expr_mat))</a></code></pre></div>
<pre><code>## [1] 0.7163828
## [1] 29.20031
## [1] 35.24549
## [1] 51.61823
## [1] 17.04449</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">save.image</span>(<span class="dt">file=</span><span class="st">&quot;../results/cache/input_data.rda&quot;</span>, <span class="dt">compress=</span><span class="ot">FALSE</span>)</a></code></pre></div>

</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Schelker2017">
<p>Schelker, Max, Sonia Feau, Jinyan Du, Nav Ranu, Edda Klipp, Gavin MacBeath, Birgit Schoeberl, and Andreas Raue. 2017. “Estimation of immune cell content in tumour tissue using single-cell RNA-seq data.” <em>Nature Communications</em> 8 (1): 2032. <a href="https://doi.org/10.1038/s41467-017-02289-3">https://doi.org/10.1038/s41467-017-02289-3</a>.</p>
</div>
<div id="ref-quantiseq2017">
<p>Finotello, Francesca, Clemens Mayer, Christina Plattner, Gerhard Laschober, Dietmar Rieder, Hubert Hackl, Anne Krogsdam, et al. 2017. “quanTIseq: quantifying immune contexture of human tumors.” <em>bioRxiv</em>, November. Cold Spring Harbor Laboratory, 223180. <a href="https://doi.org/10.1101/223180">https://doi.org/10.1101/223180</a>.</p>
</div>
<div id="ref-Hoek2015">
<p>Hoek, Kristen L, Parimal Samir, Leigh M Howard, Xinnan Niu, Nripesh Prasad, Allison Galassie, Qi Liu, et al. 2015. “A Cell-Based Systems Biology Assessment of Human Blood to Monitor Immune Responses after Influenza Vaccination.” <a href="https://doi.org/10.1371/journal.pone.0118528">https://doi.org/10.1371/journal.pone.0118528</a>.</p>
</div>
<div id="ref-EPIC2017">
<p>Racle, Julien, Kaat de Jonge, Petra Baumgaertner, Daniel E Speiser, and David Gfeller. 2017. “Simultaneous enumeration of cancer and immune cell types from bulk tumor gene expression data.” <em>eLife</em> 6 (November). eLife Sciences Publications Limited: e26476. <a href="https://doi.org/10.7554/eLife.26476">https://doi.org/10.7554/eLife.26476</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="validitiy-of-single-cell-rna-seq-as-reference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
