<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Immune Deconvolution Benchmark</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Immune Deconvolution Benchmark" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Immune Deconvolution Benchmark" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm">


<meta name="date" content="2018-06-19">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="validitiy-of-single-cell-rna-seq-as-reference.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#conceptual-differences-between-methods"><i class="fa fa-check"></i><b>1.1</b> Conceptual differences between methods</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#gene-set-based-methods-allow-comparison-between-samples-but-not-between-cell-types"><i class="fa fa-check"></i><b>1.1.1</b> Gene-set based methods (allow comparison between samples, but not between cell types)</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#deconvolution-based-methods-relative-fractions-allow-comparisons-between-cell-types-but-not-samples"><i class="fa fa-check"></i><b>1.1.2</b> Deconvolution-based methods (relative fractions, allow comparisons between cell types, but not samples)</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#deconvolution-based-methods-absolute-fractions-allow-comparisons-between-cell-types-and-samples"><i class="fa fa-check"></i><b>1.1.3</b> Deconvolution-based methods (absolute fractions, allow comparisons between cell types and samples)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-bulk-reference-samples-quantiseq-training-data"><i class="fa fa-check"></i><b>2.3</b> Immune cell bulk reference samples (quanTIseq training data)</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#validation-data-rnaseq-matched-gold-standard"><i class="fa fa-check"></i><b>2.4</b> Validation data (RNAseq + matched gold standard)</a><ul>
<li class="chapter" data-level="2.4.1" data-path="input-data.html"><a href="input-data.html#hoeck-data"><i class="fa fa-check"></i><b>2.4.1</b> Hoeck data</a></li>
<li class="chapter" data-level="2.4.2" data-path="input-data.html"><a href="input-data.html#schelker-data"><i class="fa fa-check"></i><b>2.4.2</b> Schelker data</a></li>
<li class="chapter" data-level="2.4.3" data-path="input-data.html"><a href="input-data.html#racle-data"><i class="fa fa-check"></i><b>2.4.3</b> Racle data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data"><i class="fa fa-check"></i><b>3.1</b> Validity of simulated bulk tissues from single cell sequencing data</a><ul>
<li class="chapter" data-level="3.1.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-samples-to-bulk-samples"><i class="fa fa-check"></i><b>3.1.1</b> Compare simululated samples to bulk samples</a></li>
<li class="chapter" data-level="3.1.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-the-predicted-fractions"><i class="fa fa-check"></i><b>3.1.2</b> Compare the predicted fractions</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlated-simulated-samples-with-bulk-of-the-same-tissue."><i class="fa fa-check"></i><b>3.2</b> Correlated simulated samples with bulk of the same tissue.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="mixing-study.html"><a href="mixing-study.html"><i class="fa fa-check"></i><b>4</b> Mixing study</a><ul>
<li class="chapter" data-level="4.1" data-path="mixing-study.html"><a href="mixing-study.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>4.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="4.2" data-path="mixing-study.html"><a href="mixing-study.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>4.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="4.3" data-path="mixing-study.html"><a href="mixing-study.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.3</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="4.3.1" data-path="mixing-study.html"><a href="mixing-study.html#run-timer"><i class="fa fa-check"></i><b>4.3.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="mixing-study.html"><a href="mixing-study.html#results"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="mixing-study.html"><a href="mixing-study.html#correlation-plots"><i class="fa fa-check"></i><b>4.4.1</b> Correlation plots</a></li>
<li class="chapter" data-level="4.4.2" data-path="mixing-study.html"><a href="mixing-study.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>4.4.2</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="4.4.3" data-path="mixing-study.html"><a href="mixing-study.html#absolute-error-for-all-qualifying-methods"><i class="fa fa-check"></i><b>4.4.3</b> Absolute error for all qualifying methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>5</b> Validation with real data</a></li>
<li class="chapter" data-level="6" data-path="correlation-of-methods-across-tcga-samples-.html"><a href="correlation-of-methods-across-tcga-samples-.html"><i class="fa fa-check"></i><b>6</b> Correlation of methods across TCGA samples.</a><ul>
<li class="chapter" data-level="6.1" data-path="correlation-of-methods-across-tcga-samples-.html"><a href="correlation-of-methods-across-tcga-samples-.html#heatmap-clustered-by-correlation"><i class="fa fa-check"></i><b>6.1</b> Heatmap clustered by correlation</a></li>
<li class="chapter" data-level="6.2" data-path="correlation-of-methods-across-tcga-samples-.html"><a href="correlation-of-methods-across-tcga-samples-.html#heatmap-unclustered-sorted-by-cell-type"><i class="fa fa-check"></i><b>6.2</b> Heatmap unclustered, sorted by cell type</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sensitivity-analysis.html"><a href="sensitivity-analysis.html"><i class="fa fa-check"></i><b>7</b> Sensitivity analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="sensitivity-analysis.html"><a href="sensitivity-analysis.html#predictions-with-increasing-immune-cell-content"><i class="fa fa-check"></i><b>7.1</b> Predictions with increasing immune cell content</a><ul>
<li class="chapter" data-level="7.1.1" data-path="sensitivity-analysis.html"><a href="sensitivity-analysis.html#summary-figure-at-which-percentage-the-method-captures-the-cell-type-above-noise-level"><i class="fa fa-check"></i><b>7.1.1</b> Summary figure: At which percentage the method captures the cell type above noise level?</a></li>
<li class="chapter" data-level="7.1.2" data-path="sensitivity-analysis.html"><a href="sensitivity-analysis.html#summary-figure-predicted-amount-of-cells-when-there-are-none"><i class="fa fa-check"></i><b>7.1.2</b> Summary figure: predicted amount of cells when there are none</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="specificity-analysis.html"><a href="specificity-analysis.html"><i class="fa fa-check"></i><b>8</b> Specificity Analysis</a><ul>
<li class="chapter" data-level="8.1" data-path="specificity-analysis.html"><a href="specificity-analysis.html#complete-spillover-matrix"><i class="fa fa-check"></i><b>8.1</b> Complete Spillover Matrix</a></li>
<li class="chapter" data-level="8.2" data-path="specificity-analysis.html"><a href="specificity-analysis.html#summary-figure-noise-ratio"><i class="fa fa-check"></i><b>8.2</b> Summary figure: Noise ratio</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="robustness-against-noise.html"><a href="robustness-against-noise.html"><i class="fa fa-check"></i><b>9</b> Robustness against noise</a><ul>
<li class="chapter" data-level="9.1" data-path="robustness-against-noise.html"><a href="robustness-against-noise.html#result-figures"><i class="fa fa-check"></i><b>9.1</b> Result figures</a><ul>
<li class="chapter" data-level="9.1.1" data-path="robustness-against-noise.html"><a href="robustness-against-noise.html#correlations"><i class="fa fa-check"></i><b>9.1.1</b> Correlations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>10</b> Conclusions</a></li>
<li class="chapter" data-level="11" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>11</b> Materials and Methods</a><ul>
<li class="chapter" data-level="11.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#data-acquisition"><i class="fa fa-check"></i><b>11.1</b> Data acquisition</a></li>
<li class="chapter" data-level="11.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#validation-datasets-flow-cytometry-scrnaseq-imageing-matched-rna-seq"><i class="fa fa-check"></i><b>11.2</b> Validation datasets (Flow cytometry, scRNAseq, Imageing + matched RNA seq)</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>12</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Immune Deconvolution Benchmark</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="input-data" class="section level1">
<h1><span class="header-section-number">2</span> Input data</h1>
<p>In this chapter, we load and preprocess the different input datasets. We use a dedicated environment for each dataset in order not to trash the global namespace.</p>
<div id="cell-type-hierarchy" class="section level2">
<h2><span class="header-section-number">2.1</span> cell type hierarchy</h2>
<p>We use a hierachy of immune cell types to map the cell types between different datasets. Here is the hierachy visualized:</p>
<div id="htmlwidget-72ddcc35dd03c5811504" style="width:768px;height:768px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-72ddcc35dd03c5811504">{"x":{"diagram":"digraph {\n\ngraph [rankdir = \"LR\"]\n\n\n\n  \"1\" [label = \"cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"2\" [label = \"immune cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"3\" [label = \"lymphoid\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"4\" [label = \"T cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"5\" [label = \"T cell CD8+\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"6\" [label = \"T cell CD8+ naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"7\" [label = \"T cell CD8+ memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"8\" [label = \"T cell CD4+\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"9\" [label = \"T cell CD4+ (non-regulatory)\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"10\" [label = \"T cell CD4+ naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"11\" [label = \"T cell CD4+ memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"12\" [label = \"T cell CD4+ memory resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"13\" [label = \"T cell CD4+ memory activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"14\" [label = \"T cell follicular helper\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"15\" [label = \"T cell regulatory (Tregs)\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"16\" [label = \"T cell gamma delta\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"17\" [label = \"B cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"18\" [label = \"B cell naive\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"19\" [label = \"B cell memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"20\" [label = \"B cell plasma\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"21\" [label = \"NK cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"22\" [label = \"NK cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"23\" [label = \"NK cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"24\" [label = \"myleoid\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"25\" [label = \"Neutrophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"26\" [label = \"Eosinophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"27\" [label = \"Mast cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"28\" [label = \"Mast cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"29\" [label = \"Mast cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"30\" [label = \"Monocytic lineage\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"31\" [label = \"Macrophage/Monocyte\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"32\" [label = \"Monocyte\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"33\" [label = \"Macrophage\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"34\" [label = \"Macrophage M0\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: TRUE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"35\" [label = \"Macrophage M1\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"36\" [label = \"Macrophage M2\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"37\" [label = \"Dendritic cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"38\" [label = \"Dendritic cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"39\" [label = \"Dendritic cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"40\" [label = \"other cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"41\" [label = \"Endothelial cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"42\" [label = \"Cancer associated fibroblast\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"43\" [label = \"cancer cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"44\" [label = \"Ovarian carcinoma cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"45\" [label = \"Melanoma cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- optional: FALSE\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n\"1\"->\"2\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"1\"->\"40\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"2\"->\"3\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"2\"->\"24\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"40\"->\"41\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"40\"->\"42\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"40\"->\"43\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"4\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"17\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"21\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"24\"->\"25\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"24\"->\"26\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"24\"->\"27\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"24\"->\"30\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"43\"->\"44\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"43\"->\"45\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"5\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"8\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"16\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"17\"->\"18\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"17\"->\"19\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"17\"->\"20\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"21\"->\"22\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"21\"->\"23\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"27\"->\"28\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"27\"->\"29\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"30\"->\"31\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"30\"->\"37\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"6\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"5\"->\"7\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"8\"->\"9\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"8\"->\"15\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"31\"->\"32\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"31\"->\"33\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"37\"->\"38\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"37\"->\"39\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"9\"->\"10\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"9\"->\"11\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"9\"->\"14\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"33\"->\"34\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"33\"->\"35\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"33\"->\"36\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"11\"->\"12\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"11\"->\"13\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n}","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="single-cell-data-for-simulated-mixtures" class="section level2">
<h2><span class="header-section-number">2.2</span> Single cell data for simulated mixtures</h2>
<p>In this study, we make use of the single cell dataset curated by <span class="citation">(Schelker et al. <a href="#ref-Schelker2017">2017</a>)</span>. They aggregated single cell sequencing data from different sources resulting in a set of ~12,000 single cells. They classified the cells using at set of 45 marker genes into 11 categories: 2 cancer types (Melanoma cell, Ovarian carcinoma cell), 0 immune cells (), 2 other cells (Cancer associated fibroblast, Endothelial cell) and <em>Unknown cells</em> which could not have been classified unambiguously.</p>
<p>Unknown cells are excluded from the downstream analysis.</p>
<p>The dataset consists of single cells from PBMC, melanoma and ovarian cancer ascites. As we are interested in the deconvolution of cancer samples, we exclude the PBMC cells from all downstream analyses.</p>
<table>
<caption>the ~12,000 samples by cell type</caption>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B cell</td>
<td align="right">646</td>
</tr>
<tr class="even">
<td align="left">Cancer associated fibroblast</td>
<td align="right">132</td>
</tr>
<tr class="odd">
<td align="left">Dendritic cell</td>
<td align="right">140</td>
</tr>
<tr class="even">
<td align="left">Endothelial cell</td>
<td align="right">71</td>
</tr>
<tr class="odd">
<td align="left">Macrophage/Monocyte</td>
<td align="right">2227</td>
</tr>
<tr class="even">
<td align="left">Melanoma cell</td>
<td align="right">1310</td>
</tr>
<tr class="odd">
<td align="left">NK cell</td>
<td align="right">198</td>
</tr>
<tr class="even">
<td align="left">Ovarian carcinoma cell</td>
<td align="right">300</td>
</tr>
<tr class="odd">
<td align="left">T cell CD4+ (non-regulatory)</td>
<td align="right">1196</td>
</tr>
<tr class="even">
<td align="left">T cell CD8+</td>
<td align="right">1130</td>
</tr>
<tr class="odd">
<td align="left">T cell regulatory (Tregs)</td>
<td align="right">142</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:unnamed-chunk-3"></span>
<img src="_main_files/figure-html/unnamed-chunk-3-1.png" alt="tSNE-clustering of the ~12000 single cells from [@Schelker2017]. " width="960" />
<p class="caption">
Figure 2.1: tSNE-clustering of the ~12000 single cells from <span class="citation">(Schelker et al. <a href="#ref-Schelker2017">2017</a>)</span>.
</p>
</div>
<!-- ### Attempting to distinguish between macrophages and monocytes -->
<!-- ```{r} -->
<!-- cd11b = exprs(single_cell_schelker$eset)["ITGAM",] != 0  -->
<!-- pdata = pData(single_cell_schelker$eset)  -->
<!-- pdata$ITGAM = cd11b -->
<!-- pdata %>% filter(cell_type == "Macrophage/Monocyte") %>%  -->
<!--   ggplot(aes(x=tsneX1, y=tsneX2, colour=ITGAM)) + -->
<!--    geom_point(size=1) +  -->
<!--    theme(legend.position="bottom")  -->
<!-- # pheatmap(exprs(single_cell_schelker$eset)[,pdata$cell_type == "Macrophage/Monocyte"], cluster_rows = FALSE) -->
<!-- ``` -->
</div>
<div id="immune-cell-bulk-reference-samples-quantiseq-training-data" class="section level2">
<h2><span class="header-section-number">2.3</span> Immune cell bulk reference samples (quanTIseq training data)</h2>
<div id="htmlwidget-9fe9fb32619ed3dc33b4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9fe9fb32619ed3dc33b4">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50"],["SRR1551070","SRR1551056","SRR1551049","SRR1550988","SRR1740034","SRR1740062","SRR1740038","SRR1740066","SRR1740039","SRR1740040","SRR1740041","SRR1740067","SRR1740068","SRR1740069","SRR452330","SRR452329","SRR452333","SRR452332","SRR452331","SRR1551069","SRR1551055","SRR1551048","SRR1550987","SRR1740042","SRR1740070","SRR1551068","SRR1551054","SRR1551047","SRR1550986","SRR1740046","SRR1740074","SRR1551073","SRR1551059","SRR1551052","SRR1550991","SRR1740050","SRR1740078","SRR1551071","SRR1551057","SRR1551050","SRR1550989","SRR1551072","SRR1551058","SRR1551051","SRR1550990","ERR431583","ERR431606","ERR431609","ERR431601","ERR431589"],["Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Hoek et al.","Beyer et al.","Beyer et al.","Beyer et al.","Beyer et al.","Beyer et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Hoek et al.","Hoek et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Hoek et al.","Hoek et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Hoek et al.","Hoek et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","Linsley et al.","De Simone et al.","De Simone et al.","De Simone et al.","De Simone et al.","De Simone et al."],["B cell","B cell","B cell","B cell","B cell","B cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Dendritic cell","Macrophage M1","Macrophage M1","Macrophage M2","Macrophage M2","Macrophage M2","Monocyte","Monocyte","Monocyte","Monocyte","Monocyte","Monocyte","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","Neutrophil","NK cell","NK cell","NK cell","NK cell","NK cell","NK cell","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD8+","T cell CD8+","T cell CD8+","T cell CD8+","T cell regulatory (Tregs)","T cell regulatory (Tregs)","T cell regulatory (Tregs)","T cell regulatory (Tregs)","T cell regulatory (Tregs)"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>Study<\/th>\n      <th>cell_type<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="validation-data-rnaseq-matched-gold-standard" class="section level2">
<h2><span class="header-section-number">2.4</span> Validation data (RNAseq + matched gold standard)</h2>
<p>e.g. IHC, Flow cytometry, single cell</p>
<div id="htmlwidget-5ab68ae6c94771b3fb1c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5ab68ae6c94771b3fb1c">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8"],["Finotello 2017","Finotello 2017","Hoeck 2015","Johnson 2015","Racle 2017","Zimmermann 2016","Schelker 2017","Chung 2017"],[9,9,8,31,4,12,3,11],["CRC","PBMC/PMN","PBMC/PMN","metastatic melanoma","metastatic melanoma","PBMC before influenza vacc","ovarian cancer ascites","BRCA"],["RNAseq","RNAseq","RNAseq","RNAseq","RNAseq","RNAseq","RNAseq","RNAseq"],["IHC","FlowCytometry","FlowCytometry","IHC","FlowCytometry","FlowCytometry","FlowCytometry + Single Cell RNAseq","Single Cell RNAseq + IHC for B and T"],["CD3, CD8, CD45RO, FOXP3, PD1","B, Mono, Neu, NK, CD4, CD8, Treg, DC","T, B, NK, MDC, Mono, Neu","CD8, CD4, FOXP3+","B, CD4, CD8, NK, Melanoma/other","B, CD4, CD8, NK, Mono, Other","CD8, CD4, B, NK, DC, Mono, ...",null],[null,null,"Finotello 2017","Finotello 2017",null,"Racle 2017",null,null],["Available upon publication","Available upon publication","Available through personal communicaion","How to get?",null,"obtained through Racle 2017",null,null],[null,null,"x",null,"x",null,"x",null],[null,null,"x",null,"x","x","x",null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Study<\/th>\n      <th># Samples<\/th>\n      <th>Sample Type<\/th>\n      <th>Gene Expression by<\/th>\n      <th>Validation by<\/th>\n      <th>Cell types<\/th>\n      <th>Cited by<\/th>\n      <th>Comments<\/th>\n      <th>Got TPM<\/th>\n      <th>Got Validation<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="hoeck-data" class="section level3">
<h3><span class="header-section-number">2.4.1</span> Hoeck data</h3>
<p>8 PBMC, RNAseq+flow cytometry</p>
<p>Measured cell types: T cell, Macrophage/Monocyte, B cell, Dendritic cell, NK cell.</p>
</div>
<div id="schelker-data" class="section level3">
<h3><span class="header-section-number">2.4.2</span> Schelker data</h3>
<p>Load the data and merge replicates. 3 ovarian cancer ascites samplese which have matched flow cytometry/single cell/bulk data.</p>
<p>Single cell count for the 3 samples</p>
<table>
<thead>
<tr class="header">
<th align="left">donor</th>
<th align="right">sum(cell_count)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">7873M</td>
<td align="right">864</td>
</tr>
<tr class="even">
<td align="left">7882M</td>
<td align="right">902</td>
</tr>
<tr class="odd">
<td align="left">7892M</td>
<td align="right">773</td>
</tr>
</tbody>
</table>
<div id="htmlwidget-849e24e3949e8c5a688f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-849e24e3949e8c5a688f">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25"],["7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7873M","7882M","7892M","7882M"],["B cell","B cell","B cell","Cancer associated fibroblast","Cancer associated fibroblast","Cancer associated fibroblast","Dendritic cell","Dendritic cell","Dendritic cell","Macrophage/Monocyte","Macrophage/Monocyte","Macrophage/Monocyte","NK cell","NK cell","NK cell","Ovarian carcinoma cell","Ovarian carcinoma cell","Ovarian carcinoma cell","T cell CD4+ (non-regulatory)","T cell CD4+ (non-regulatory)","T cell CD4+ (non-regulatory)","T cell CD8+","T cell CD8+","T cell CD8+","T cell regulatory (Tregs)"],[5,4,5,9,11,16,27,46,39,761,599,520,10,51,42,19,34,46,29,137,100,4,19,5,1],[0.00578703703703704,0.00443458980044346,0.00646830530401035,0.0104166666666667,0.0121951219512195,0.0206985769728331,0.03125,0.0509977827050998,0.0504527813712807,0.880787037037037,0.664079822616408,0.672703751617076,0.0115740740740741,0.0565410199556541,0.0543337645536869,0.0219907407407407,0.0376940133037694,0.0595084087968952,0.0335648148148148,0.151884700665188,0.129366106080207,0.00462962962962963,0.0210643015521064,0.00646830530401035,0.00110864745011086]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>donor<\/th>\n      <th>cell_type<\/th>\n      <th>cell_count<\/th>\n      <th>true_fraction<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="racle-data" class="section level3">
<h3><span class="header-section-number">2.4.3</span> Racle data</h3>
<p>4 metastatic melanoma, matched bulkRNAseq with FlowCytometry. <div id="htmlwidget-94dadbbdf20086f82a00" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-94dadbbdf20086f82a00">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314","LAU125","LAU355","LAU1255","LAU1314"],["B cell","B cell","B cell","B cell","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD4+","T cell CD8+","T cell CD8+","T cell CD8+","T cell CD8+","NK cell","NK cell","NK cell","NK cell"],[0.1812,0.3248,0.0579,0.4667,0.0082,0.2315,0.0276,0.1815,0.0035,0.0582,0.0376,0.0454,0.005,0.0017,0.0017,0.0025]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>cell_type<\/th>\n      <th>true_fraction<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script></p>

</div>
</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Schelker2017">
<p>Schelker, Max, Sonia Feau, Jinyan Du, Nav Ranu, Edda Klipp, Gavin MacBeath, Birgit Schoeberl, and Andreas Raue. 2017. “Estimation of immune cell content in tumour tissue using single-cell RNA-seq data.” <em>Nature Communications</em> 8 (1): 2032. doi:<a href="https://doi.org/10.1038/s41467-017-02289-3">10.1038/s41467-017-02289-3</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="validitiy-of-single-cell-rna-seq-as-reference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
