<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Immune Deconvolution Benchmark</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Immune Deconvolution Benchmark" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Immune Deconvolution Benchmark" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm">


<meta name="date" content="2018-06-04">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="validitiy-of-single-cell-rna-seq-as-reference.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#conceptual-differences-between-methods"><i class="fa fa-check"></i><b>1.1</b> Conceptual differences between methods</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#gene-set-based-methods-allow-comparison-between-samples-but-not-between-cell-types"><i class="fa fa-check"></i><b>1.1.1</b> Gene-set based methods (allow comparison between samples, but not between cell types)</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#deconvolution-based-methods-relative-fractions-allow-comparisons-between-cell-types-but-not-samples"><i class="fa fa-check"></i><b>1.1.2</b> Deconvolution-based methods (relative fractions, allow comparisons between cell types, but not samples)</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#deconvolution-based-methods-absolute-fractions-allow-comparisons-between-cell-types-and-samples"><i class="fa fa-check"></i><b>1.1.3</b> Deconvolution-based methods (absolute fractions, allow comparisons between cell types and samples)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-bulk-reference-samples-quantiseq-training-data"><i class="fa fa-check"></i><b>2.3</b> Immune cell bulk reference samples (quanTIseq training data)</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#validation-data-rnaseq-matched-gold-standard"><i class="fa fa-check"></i><b>2.4</b> Validation data (RNAseq + matched gold standard)</a><ul>
<li class="chapter" data-level="2.4.1" data-path="input-data.html"><a href="input-data.html#hoeck-data"><i class="fa fa-check"></i><b>2.4.1</b> Hoeck data</a></li>
<li class="chapter" data-level="2.4.2" data-path="input-data.html"><a href="input-data.html#schelker-data"><i class="fa fa-check"></i><b>2.4.2</b> Schelker data</a></li>
<li class="chapter" data-level="2.4.3" data-path="input-data.html"><a href="input-data.html#racle-data"><i class="fa fa-check"></i><b>2.4.3</b> Racle data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-single-cell-sequencing-for-counting-immune-cells"><i class="fa fa-check"></i><b>3.1</b> Validity of single cell sequencing for counting immune cells</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data"><i class="fa fa-check"></i><b>3.2</b> Validity of simulated bulk tissues from single cell sequencing data</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-samples-to-bulk-samples"><i class="fa fa-check"></i><b>3.2.1</b> Compare simululated samples to bulk samples</a></li>
<li class="chapter" data-level="3.2.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-the-predicted-fractions"><i class="fa fa-check"></i><b>3.2.2</b> Compare the predicted fractions</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlated-simulated-samples-with-bulk-of-the-same-tissue."><i class="fa fa-check"></i><b>3.3</b> Correlated simulated samples with bulk of the same tissue.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sensitivity-analysis.html"><a href="sensitivity-analysis.html"><i class="fa fa-check"></i><b>4</b> Sensitivity analysis</a></li>
<li class="chapter" data-level="5" data-path="specificity-analysis.html"><a href="specificity-analysis.html"><i class="fa fa-check"></i><b>5</b> Specificity Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="specificity-analysis.html"><a href="specificity-analysis.html#create-datasets"><i class="fa fa-check"></i><b>5.1</b> Create datasets</a></li>
<li class="chapter" data-level="5.2" data-path="specificity-analysis.html"><a href="specificity-analysis.html#run-the-deconvolution"><i class="fa fa-check"></i><b>5.2</b> Run the deconvolution</a></li>
<li class="chapter" data-level="5.3" data-path="specificity-analysis.html"><a href="specificity-analysis.html#results"><i class="fa fa-check"></i><b>5.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="correlation-of-methods-across-tcga-samples-.html"><a href="correlation-of-methods-across-tcga-samples-.html"><i class="fa fa-check"></i><b>6</b> Correlation of methods across TCGA samples.</a></li>
<li class="chapter" data-level="7" data-path="mixing-study.html"><a href="mixing-study.html"><i class="fa fa-check"></i><b>7</b> Mixing study</a><ul>
<li class="chapter" data-level="7.1" data-path="mixing-study.html"><a href="mixing-study.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>7.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="7.2" data-path="mixing-study.html"><a href="mixing-study.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>7.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="7.3" data-path="mixing-study.html"><a href="mixing-study.html#run-the-deconvolution-1"><i class="fa fa-check"></i><b>7.3</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="7.3.1" data-path="mixing-study.html"><a href="mixing-study.html#run-timer"><i class="fa fa-check"></i><b>7.3.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="mixing-study.html"><a href="mixing-study.html#results-1"><i class="fa fa-check"></i><b>7.4</b> Results</a><ul>
<li class="chapter" data-level="7.4.1" data-path="mixing-study.html"><a href="mixing-study.html#calculate-other-cells-for-all-absolute-methods"><i class="fa fa-check"></i><b>7.4.1</b> Calculate “Other cells” for all absolute methods</a></li>
<li class="chapter" data-level="7.4.2" data-path="mixing-study.html"><a href="mixing-study.html#correlation-plots"><i class="fa fa-check"></i><b>7.4.2</b> Correlation plots</a></li>
<li class="chapter" data-level="7.4.3" data-path="mixing-study.html"><a href="mixing-study.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>7.4.3</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="7.4.4" data-path="mixing-study.html"><a href="mixing-study.html#absolute-error-for-all-qualifying-methods"><i class="fa fa-check"></i><b>7.4.4</b> Absolute error for all qualifying methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>8</b> Validation with real data</a><ul>
<li class="chapter" data-level="8.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#run-the-deconvolution-2"><i class="fa fa-check"></i><b>8.1</b> Run the deconvolution</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>9</b> Conclusions</a></li>
<li class="chapter" data-level="10" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>10</b> Materials and Methods</a><ul>
<li class="chapter" data-level="10.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#data-acquisition"><i class="fa fa-check"></i><b>10.1</b> Data acquisition</a></li>
<li class="chapter" data-level="10.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#validation-datasets-flow-cytometry-scrnaseq-imageing-matched-rna-seq"><i class="fa fa-check"></i><b>10.2</b> Validation datasets (Flow cytometry, scRNAseq, Imageing + matched RNA seq)</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>11</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Immune Deconvolution Benchmark</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="input-data" class="section level1">
<h1><span class="header-section-number">2</span> Input data</h1>
<p>In this chapter, we load and preprocess the different input datasets. We use a dedicated environment for each dataset in order not to trash the global namespace.</p>
<div id="cell-type-hierarchy" class="section level2">
<h2><span class="header-section-number">2.1</span> cell type hierarchy</h2>
<p>We use a hierachy of immune cell types to map the cell types between different datasets. Here is the hierachy visualized:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell_type_hierarchy =<span class="st"> </span><span class="kw">new.env</span>()
<span class="kw">with</span>(cell_type_hierarchy, {
  adj_list =<span class="st"> </span><span class="kw">read_excel</span>(<span class="st">&quot;../tables/cell_type_mapping2.xlsx&quot;</span>, <span class="dt">sheet =</span> <span class="st">&quot;controlled_vocabulary&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(parent, cell_type)
  tree =<span class="st"> </span><span class="kw">FromDataFrameNetwork</span>(adj_list <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>())
  <span class="kw">SetGraphStyle</span>(tree, <span class="dt">rankdir =</span> <span class="st">&quot;LR&quot;</span>)
  <span class="kw">SetEdgeStyle</span>(tree, <span class="dt">arrowhead =</span> <span class="st">&quot;vee&quot;</span>, <span class="dt">color =</span> <span class="st">&quot;grey35&quot;</span>, <span class="dt">penwidth =</span> <span class="dv">2</span>)
  <span class="kw">SetNodeStyle</span>(tree, <span class="dt">style =</span> <span class="st">&quot;filled,rounded&quot;</span>, <span class="dt">shape =</span> <span class="st">&quot;box&quot;</span>, <span class="dt">fillcolor =</span> <span class="st">&quot;GreenYellow&quot;</span>, 
              <span class="dt">fontname =</span> <span class="st">&quot;helvetica&quot;</span>, <span class="dt">tooltip =</span> GetDefaultTooltip, <span class="dt">fontcolor=</span><span class="st">&quot;black&quot;</span>)
  <span class="kw">plot</span>(tree)
})</code></pre></div>
<div id="htmlwidget-2305c333580f0138db43" style="width:768px;height:768px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-2305c333580f0138db43">{"x":{"diagram":"digraph {\n\ngraph [rankdir = \"LR\"]\n\n\n\n  \"1\" [label = \"cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"2\" [label = \"immune cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: immune cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"3\" [label = \"lymphoid\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: lymphoid\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"4\" [label = \"T cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"5\" [label = \"T cell CD8+\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell CD8+\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"6\" [label = \"T cell CD4+\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell CD4+\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"7\" [label = \"T cell CD4+ (non-regulatory)\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell CD4+ (non-regulatory)\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"8\" [label = \"T cell CD4+ naïve\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell CD4+ naïve\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"9\" [label = \"T cell CD4+ memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell CD4+ memory\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"10\" [label = \"T cell CD4+ memory resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell CD4+ memory resting\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"11\" [label = \"T cell CD4+ memory activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell CD4+ memory activated\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"12\" [label = \"T cell follicular helper\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell follicular helper\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"13\" [label = \"T cell regulatory (Tregs)\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell regulatory (Tregs)\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"14\" [label = \"T cell gamma delta\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: T cell gamma delta\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"15\" [label = \"B cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: B cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"16\" [label = \"B cell naïve\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: B cell naïve\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"17\" [label = \"B cell memory\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: B cell memory\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"18\" [label = \"B cell plasma\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: B cell plasma\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"19\" [label = \"NK cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: NK cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"20\" [label = \"NK cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: NK cell resting\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"21\" [label = \"NK cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: NK cell activated\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"22\" [label = \"myleoid\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: myleoid\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"23\" [label = \"Neutrophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Neutrophil\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"24\" [label = \"Eosinophil\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Eosinophil\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"25\" [label = \"Mast cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Mast cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"26\" [label = \"Mast cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Mast cell resting\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"27\" [label = \"Mast cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Mast cell activated\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"28\" [label = \"Monocytic lineage\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Monocytic lineage\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"29\" [label = \"Macrophage/Monocyte\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Macrophage/Monocyte\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"30\" [label = \"Monocyte\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Monocyte\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"31\" [label = \"Macrophage\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Macrophage\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"32\" [label = \"Macrophage M0\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Macrophage M0\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"33\" [label = \"Macrophage M1\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Macrophage M1\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"34\" [label = \"Macrophage M2\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Macrophage M2\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"35\" [label = \"Dendritic cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Dendritic cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"36\" [label = \"Dendritic cell resting\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Dendritic cell resting\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"37\" [label = \"Dendritic cell activated\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: Dendritic cell activated\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"38\" [label = \"other cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: other cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"39\" [label = \"endothelial cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: endothelial cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"40\" [label = \"cancer associated fibroblast\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: cancer associated fibroblast\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"41\" [label = \"cancer cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: cancer cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"42\" [label = \"ovarian carcinoma cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: ovarian carcinoma cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n  \"43\" [label = \"melanoma cell\", style = \"filled,rounded\", shape = \"box\", fontname = \"helvetica\", tooltip = \"- name: melanoma cell\", fontcolor = \"black\", fillcolor = \"GreenYellow\"] \n\"1\"->\"2\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"1\"->\"38\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"2\"->\"3\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"2\"->\"22\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"38\"->\"39\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"38\"->\"40\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"38\"->\"41\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"4\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"15\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"3\"->\"19\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"22\"->\"23\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"22\"->\"24\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"22\"->\"25\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"22\"->\"28\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"41\"->\"42\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"41\"->\"43\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"5\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"6\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"4\"->\"14\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"15\"->\"16\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"15\"->\"17\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"15\"->\"18\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"19\"->\"20\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"19\"->\"21\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"25\"->\"26\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"25\"->\"27\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"28\"->\"29\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"28\"->\"35\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"7\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"8\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"9\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"12\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"6\"->\"13\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"29\"->\"30\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"29\"->\"31\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"35\"->\"36\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"35\"->\"37\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"9\"->\"10\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"9\"->\"11\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"31\"->\"32\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"31\"->\"33\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n\"31\"->\"34\" [arrowhead = \"vee\", color = \"grey35\", penwidth = \"2\"] \n}","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="single-cell-data-for-simulated-mixtures" class="section level2">
<h2><span class="header-section-number">2.2</span> Single cell data for simulated mixtures</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tmp_env =<span class="st"> </span><span class="kw">new.env</span>()
<span class="kw">with</span>(tmp_env, {
  <span class="kw">load</span>(<span class="st">&quot;../data/single_cell_schelker/single_cell_schelker.rda&quot;</span>)
  <span class="co"># symbol_remap = read_tsv(&quot;../data/single_cell_schelker/symbol_checker.tsv&quot;) %&gt;%</span>
  <span class="co">#   as.data.frame() %&gt;%</span>
  <span class="co">#   column_to_rownames(&quot;Input&quot;)</span>
  valid_cells =<span class="st"> </span><span class="kw">which</span>(<span class="kw">pData</span>(single_cell_schelker)<span class="op">$</span>cell_type<span class="op">!=</span><span class="st">&quot;Unknown&quot;</span>)
  eset =<span class="st"> </span>single_cell_schelker[,valid_cells]
  <span class="co"># rownames(eset) = symbol_remap[,rownames(eset)]$`Approved symbol`</span>
  <span class="kw">assert_that</span>(<span class="op">!</span><span class="kw">any</span>(<span class="kw">pData</span>(eset)<span class="op">$</span>cell_type <span class="op">==</span><span class="st"> &quot;Unknown&quot;</span>))
})</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">single_cell_schelker =<span class="st"> </span><span class="kw">new.env</span>()
single_cell_schelker<span class="op">$</span>eset =<span class="st"> </span>tmp_env<span class="op">$</span>eset</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell_types =<span class="st"> </span><span class="kw">as.environment</span>(<span class="kw">list</span>(
  <span class="dt">available=</span><span class="kw">pData</span>(single_cell_schelker<span class="op">$</span>eset) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(cell_type) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(cell_type),
  <span class="dt">cancer=</span><span class="kw">c</span>(<span class="st">&quot;Melanoma cells&quot;</span>, <span class="st">&quot;Ovarian carcinoma cells&quot;</span>),
  <span class="dt">other=</span><span class="kw">c</span>(<span class="st">&quot;Cancer associated fibroblasts&quot;</span>, <span class="st">&quot;Endothelial cells&quot;</span>)
))
cell_types<span class="op">$</span>immune_cells =<span class="st"> </span>cell_types<span class="op">$</span>available[<span class="op">!</span>cell_types<span class="op">$</span>available <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(cell_types<span class="op">$</span>cancer, cell_types<span class="op">$</span>other)]</code></pre></div>
<p>In this study, we make use of the single cell dataset curated by <span class="citation">(Schelker et al. <a href="#ref-Schelker2017">2017</a>)</span>. They aggregated single cell sequencing data from different sources resulting in a set of ~12,000 single cells. They classified the cells using at set of 45 marker genes into 11 categories: 2 cancer types (Melanoma cells, Ovarian carcinoma cells), 0 immune cells (), 2 other cells (Cancer associated fibroblasts, Endothelial cells) and <em>Unknown cells</em> which could not have been classified unambiguously.</p>
<p>Unknown cells are excluded from the downstream analysis.</p>
<table>
<caption>the ~12,000 samples by cell type</caption>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B cells</td>
<td align="right">1044</td>
</tr>
<tr class="even">
<td align="left">Cancer associated fibroblasts</td>
<td align="right">132</td>
</tr>
<tr class="odd">
<td align="left">CD4+ T cells</td>
<td align="right">3023</td>
</tr>
<tr class="even">
<td align="left">CD8+ T cells</td>
<td align="right">1884</td>
</tr>
<tr class="odd">
<td align="left">Dendritic cells</td>
<td align="right">166</td>
</tr>
<tr class="even">
<td align="left">Endothelial cells</td>
<td align="right">71</td>
</tr>
<tr class="odd">
<td align="left">Macrophages/Monocytes</td>
<td align="right">2934</td>
</tr>
<tr class="even">
<td align="left">Melanoma cells</td>
<td align="right">1310</td>
</tr>
<tr class="odd">
<td align="left">Natural killer cells</td>
<td align="right">425</td>
</tr>
<tr class="even">
<td align="left">Ovarian carcinoma cells</td>
<td align="right">300</td>
</tr>
<tr class="odd">
<td align="left">regulatory T cells</td>
<td align="right">145</td>
</tr>
</tbody>
</table>
<div class="figure"><span id="fig:unnamed-chunk-3"></span>
<img src="_main_files/figure-html/unnamed-chunk-3-1.png" alt="tSNE-clustering of the ~12000 single cells from [@Schelker2017]. " width="960" />
<p class="caption">
Figure 2.1: tSNE-clustering of the ~12000 single cells from <span class="citation">(Schelker et al. <a href="#ref-Schelker2017">2017</a>)</span>.
</p>
</div>
</div>
<div id="immune-cell-bulk-reference-samples-quantiseq-training-data" class="section level2">
<h2><span class="header-section-number">2.3</span> Immune cell bulk reference samples (quanTIseq training data)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">immune_cell_reference =<span class="st"> </span><span class="kw">new.env</span>()
<span class="kw">with</span>(immune_cell_reference, {
  tidy =<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;../data/immune_cell_reference_profiles/immune_cell_reference_tidy.tsv&quot;</span>)

  sample_description =<span class="st"> </span>tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(sample, CellType, Study) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>()

  expr =<span class="st"> </span>tidy <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">select</span>(sample, hgnc_id, TPM) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(sample, hgnc_id) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">TPM =</span> <span class="kw">sum</span>(TPM)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">spread</span>(sample, TPM)

  expr_mat =<span class="st"> </span>expr <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;hgnc_id&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">as.matrix</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span>.[, sample_description<span class="op">$</span>sample]
})</code></pre></div>
</div>
<div id="validation-data-rnaseq-matched-gold-standard" class="section level2">
<h2><span class="header-section-number">2.4</span> Validation data (RNAseq + matched gold standard)</h2>
<p>e.g. IHC, Flow cytometry, single cell</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">validation_datasets =<span class="st"> </span><span class="kw">read_excel</span>(<span class="st">&quot;../tables/validation-datasets.xlsx&quot;</span>)
<span class="kw">kable</span>(validation_datasets)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">Study</th>
<th align="right"># Samples</th>
<th align="left">Sample Type</th>
<th align="left">Gene Expression by</th>
<th align="left">Validation by</th>
<th align="left">Cell types</th>
<th align="left">Cited by</th>
<th align="left">Comments</th>
<th align="left">Got TPM</th>
<th align="left">Got Validation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Finotello 2017</td>
<td align="right">9</td>
<td align="left">CRC</td>
<td align="left">RNAseq</td>
<td align="left">IHC</td>
<td align="left">CD3, CD8, CD45RO, FOXP3, PD1</td>
<td align="left">NA</td>
<td align="left">Available upon publication</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">Finotello 2017</td>
<td align="right">9</td>
<td align="left">PBMC/PMN</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry</td>
<td align="left">B, Mono, Neu, NK, CD4, CD8, Treg, DC</td>
<td align="left">NA</td>
<td align="left">Available upon publication</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">Hoeck 2015</td>
<td align="right">8</td>
<td align="left">PBMC/PMN</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry</td>
<td align="left">T, B, NK, MDC, Mono, Neu</td>
<td align="left">Finotello 2017</td>
<td align="left">Available through personal communicaion</td>
<td align="left">x</td>
<td align="left">x</td>
</tr>
<tr class="even">
<td align="left">Johnson 2015</td>
<td align="right">31</td>
<td align="left">metastatic melanoma</td>
<td align="left">RNAseq</td>
<td align="left">IHC</td>
<td align="left">CD8, CD4, FOXP3+</td>
<td align="left">Finotello 2017</td>
<td align="left">How to get?</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">Racle 2017</td>
<td align="right">4</td>
<td align="left">metastatic melanoma</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry</td>
<td align="left">B, CD4, CD8, NK, Melanoma/other</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">x</td>
<td align="left">x</td>
</tr>
<tr class="even">
<td align="left">Zimmermann 2016</td>
<td align="right">12</td>
<td align="left">PBMC before influenza vacc</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry</td>
<td align="left">B, CD4, CD8, NK, Mono, Other</td>
<td align="left">Racle 2017</td>
<td align="left">obtained through Racle 2017</td>
<td align="left">NA</td>
<td align="left">x</td>
</tr>
<tr class="odd">
<td align="left">Schelker 2017</td>
<td align="right">3</td>
<td align="left">ovarian cancer ascites</td>
<td align="left">RNAseq</td>
<td align="left">FlowCytometry + Single Cell RNAseq</td>
<td align="left">CD8, CD4, B, NK, DC, Mono, …</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">x</td>
<td align="left">x</td>
</tr>
<tr class="even">
<td align="left">Chung 2017</td>
<td align="right">11</td>
<td align="left">BRCA</td>
<td align="left">RNAseq</td>
<td align="left">Single Cell RNAseq + IHC for B and T</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<div id="hoeck-data" class="section level3">
<h3><span class="header-section-number">2.4.1</span> Hoeck data</h3>
<p>8 PBMC, RNAseq+flow cytometry</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tmp_env =<span class="st"> </span><span class="kw">new.env</span>()
<span class="kw">with</span>(tmp_env, {
  <span class="kw">load</span>(<span class="st">&quot;../data/hoeck/HoekPBMC_mixture.RData&quot;</span>)
  <span class="kw">load</span>(<span class="st">&quot;../data/hoeck/HoekPBMC_gtruth.RData&quot;</span>)
})
hoeck =<span class="st"> </span><span class="kw">new.env</span>() 
hoeck<span class="op">$</span>expr_mat =<span class="st"> </span>tmp_env<span class="op">$</span>mix.mat
hoeck<span class="op">$</span>ref =<span class="st"> </span>tmp_env<span class="op">$</span>RefData</code></pre></div>
</div>
<div id="schelker-data" class="section level3">
<h3><span class="header-section-number">2.4.2</span> Schelker data</h3>
<p>Load the data and merge replicates. 3 ovarian cancer ascites samplese which have matched flow cytometry/single cell/bulk data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">schelker_ovarian =<span class="st"> </span><span class="kw">new.env</span>()
<span class="kw">with</span>(schelker_ovarian, {
  bulk_replicates =<span class="st"> </span><span class="kw">read_xls</span>(<span class="st">&quot;../data/single_cell_schelker/ascites_bulk_samples.xls&quot;</span>) 

  samples =<span class="st"> </span><span class="kw">list</span>(<span class="st">&quot;7873M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7873_1&quot;</span>, <span class="st">&quot;ascites_7873_2&quot;</span>),
                 <span class="st">&quot;7882M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7882_1&quot;</span>, <span class="st">&quot;ascites_7882_2&quot;</span>),
                 <span class="st">&quot;7892M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7892_1&quot;</span>, <span class="st">&quot;ascites_7892_2&quot;</span>))
  
  bulk =<span class="st"> </span><span class="kw">lapply</span>(samples, <span class="cf">function</span>(cols) {
    <span class="kw">apply</span>(bulk_replicates[,cols], <span class="dv">1</span>, mean)
  }) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_cols</span>() 
  
  bulk <span class="op">%&lt;&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">gene_symbol =</span> bulk_replicates<span class="op">$</span>Row)
  
  bulk_mat =<span class="st"> </span>bulk <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;gene_symbol&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">as.matrix</span>()
  
  <span class="co"># Load the ground truth</span>
  
  true_cell_count =<span class="st"> </span><span class="kw">pData</span>(single_cell_schelker<span class="op">$</span>eset) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">filter</span>(source <span class="op">==</span><span class="st"> &quot;ascites&quot;</span>, donor <span class="op">%in%</span><span class="st"> </span><span class="kw">colnames</span>(bulk)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(donor, cell_type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">count</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">rename</span>(<span class="dt">cell_count=</span>n) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">true_fraction=</span>cell_count<span class="op">/</span><span class="kw">sum</span>(cell_count)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">arrange</span>(cell_type, donor)

  

   ref =<span class="st"> </span>true_cell_count <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="dt">sample=</span>donor, cell_type, true_fraction)
})</code></pre></div>
<p>Single cell count for the 3 samples</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">schelker_ovarian<span class="op">$</span>true_cell_count <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="kw">sum</span>(cell_count)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">kable</span>()</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">donor</th>
<th align="right">sum(cell_count)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">7873M</td>
<td align="right">864</td>
</tr>
<tr class="even">
<td align="left">7882M</td>
<td align="right">902</td>
</tr>
<tr class="odd">
<td align="left">7892M</td>
<td align="right">773</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">schelker_ovarian<span class="op">$</span>true_cell_count <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">kable</span>()</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">donor</th>
<th align="left">cell_type</th>
<th align="right">cell_count</th>
<th align="right">true_fraction</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">7873M</td>
<td align="left">B cells</td>
<td align="right">5</td>
<td align="right">0.0057870</td>
</tr>
<tr class="even">
<td align="left">7882M</td>
<td align="left">B cells</td>
<td align="right">4</td>
<td align="right">0.0044346</td>
</tr>
<tr class="odd">
<td align="left">7892M</td>
<td align="left">B cells</td>
<td align="right">5</td>
<td align="right">0.0064683</td>
</tr>
<tr class="even">
<td align="left">7873M</td>
<td align="left">Cancer associated fibroblasts</td>
<td align="right">9</td>
<td align="right">0.0104167</td>
</tr>
<tr class="odd">
<td align="left">7882M</td>
<td align="left">Cancer associated fibroblasts</td>
<td align="right">11</td>
<td align="right">0.0121951</td>
</tr>
<tr class="even">
<td align="left">7892M</td>
<td align="left">Cancer associated fibroblasts</td>
<td align="right">16</td>
<td align="right">0.0206986</td>
</tr>
<tr class="odd">
<td align="left">7873M</td>
<td align="left">CD4+ T cells</td>
<td align="right">29</td>
<td align="right">0.0335648</td>
</tr>
<tr class="even">
<td align="left">7882M</td>
<td align="left">CD4+ T cells</td>
<td align="right">137</td>
<td align="right">0.1518847</td>
</tr>
<tr class="odd">
<td align="left">7892M</td>
<td align="left">CD4+ T cells</td>
<td align="right">100</td>
<td align="right">0.1293661</td>
</tr>
<tr class="even">
<td align="left">7873M</td>
<td align="left">CD8+ T cells</td>
<td align="right">4</td>
<td align="right">0.0046296</td>
</tr>
<tr class="odd">
<td align="left">7882M</td>
<td align="left">CD8+ T cells</td>
<td align="right">19</td>
<td align="right">0.0210643</td>
</tr>
<tr class="even">
<td align="left">7892M</td>
<td align="left">CD8+ T cells</td>
<td align="right">5</td>
<td align="right">0.0064683</td>
</tr>
<tr class="odd">
<td align="left">7873M</td>
<td align="left">Dendritic cells</td>
<td align="right">27</td>
<td align="right">0.0312500</td>
</tr>
<tr class="even">
<td align="left">7882M</td>
<td align="left">Dendritic cells</td>
<td align="right">46</td>
<td align="right">0.0509978</td>
</tr>
<tr class="odd">
<td align="left">7892M</td>
<td align="left">Dendritic cells</td>
<td align="right">39</td>
<td align="right">0.0504528</td>
</tr>
<tr class="even">
<td align="left">7873M</td>
<td align="left">Macrophages/Monocytes</td>
<td align="right">761</td>
<td align="right">0.8807870</td>
</tr>
<tr class="odd">
<td align="left">7882M</td>
<td align="left">Macrophages/Monocytes</td>
<td align="right">599</td>
<td align="right">0.6640798</td>
</tr>
<tr class="even">
<td align="left">7892M</td>
<td align="left">Macrophages/Monocytes</td>
<td align="right">520</td>
<td align="right">0.6727038</td>
</tr>
<tr class="odd">
<td align="left">7873M</td>
<td align="left">Natural killer cells</td>
<td align="right">10</td>
<td align="right">0.0115741</td>
</tr>
<tr class="even">
<td align="left">7882M</td>
<td align="left">Natural killer cells</td>
<td align="right">51</td>
<td align="right">0.0565410</td>
</tr>
<tr class="odd">
<td align="left">7892M</td>
<td align="left">Natural killer cells</td>
<td align="right">42</td>
<td align="right">0.0543338</td>
</tr>
<tr class="even">
<td align="left">7873M</td>
<td align="left">Ovarian carcinoma cells</td>
<td align="right">19</td>
<td align="right">0.0219907</td>
</tr>
<tr class="odd">
<td align="left">7882M</td>
<td align="left">Ovarian carcinoma cells</td>
<td align="right">34</td>
<td align="right">0.0376940</td>
</tr>
<tr class="even">
<td align="left">7892M</td>
<td align="left">Ovarian carcinoma cells</td>
<td align="right">46</td>
<td align="right">0.0595084</td>
</tr>
<tr class="odd">
<td align="left">7882M</td>
<td align="left">regulatory T cells</td>
<td align="right">1</td>
<td align="right">0.0011086</td>
</tr>
</tbody>
</table>
</div>
<div id="racle-data" class="section level3">
<h3><span class="header-section-number">2.4.3</span> Racle data</h3>
<p>4 metastatic melanoma, matched bulkRNAseq with FlowCytometry.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tmp_env =<span class="st"> </span><span class="kw">new.env</span>()
<span class="kw">with</span>(tmp_env, {
  ensg_hgnc =<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;../data/ensemble_hgnc.txt&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">rename</span>(<span class="dt">gene_id =</span> <span class="st">`</span><span class="dt">Gene stable ID</span><span class="st">`</span>, <span class="dt">hgnc_symbol=</span><span class="st">`</span><span class="dt">HGNC symbol</span><span class="st">`</span>)
  racle_cyto =<span class="st"> </span><span class="kw">read_excel</span>(<span class="st">&quot;../data/racle/racle2017_flow_cytometry.xlsx&quot;</span>)
  racle_files =<span class="st"> </span><span class="kw">list.files</span>(<span class="st">&quot;../data/racle/GSE93722_RAW/&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)
  racle_identifiers =<span class="st"> </span><span class="kw">str_extract</span>(racle_files, <span class="st">&quot;LAU</span><span class="ch">\\</span><span class="st">d+&quot;</span>)
  racle_expression_raw =<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">list.files</span>(<span class="st">&quot;../data/racle/GSE93722_RAW/&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>), read_tsv) 
  gene_ids =<span class="st"> </span>racle_expression_raw[[<span class="dv">1</span>]]<span class="op">$</span>gene_id
  racle_tpm =<span class="st"> </span><span class="kw">sapply</span>(racle_expression_raw, <span class="cf">function</span>(df) {<span class="kw">pull</span>(df, <span class="st">&quot;TPM&quot;</span>)}) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.tibble</span>()
  <span class="kw">colnames</span>(racle_tpm) =<span class="st"> </span>racle_identifiers
  racle_tpm<span class="op">$</span>gene_id =<span class="st"> </span>gene_ids
  racle_expression =<span class="st"> </span>racle_tpm <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">inner_join</span>(ensg_hgnc) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(hgnc_symbol)) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">select</span>(<span class="op">-</span>gene_id) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">group_by</span>(hgnc_symbol) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">summarise_all</span>(sum)
})

racle =<span class="st"> </span><span class="kw">new.env</span>()
racle<span class="op">$</span>expr =<span class="st"> </span>tmp_env<span class="op">$</span>racle_expression
racle<span class="op">$</span>expr_mat =<span class="st"> </span>racle<span class="op">$</span>expr <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;hgnc_symbol&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.matrix</span>()
racle<span class="op">$</span>ref =<span class="st"> </span>tmp_env<span class="op">$</span>racle_cyto</code></pre></div>

</div>
</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Schelker2017">
<p>Schelker, Max, Sonia Feau, Jinyan Du, Nav Ranu, Edda Klipp, Gavin MacBeath, Birgit Schoeberl, and Andreas Raue. 2017. “Estimation of immune cell content in tumour tissue using single-cell RNA-seq data.” <em>Nature Communications</em> 8 (1): 2032. doi:<a href="https://doi.org/10.1038/s41467-017-02289-3">10.1038/s41467-017-02289-3</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="validitiy-of-single-cell-rna-seq-as-reference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
