<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Immune Deconvolution Benchmark</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Immune Deconvolution Benchmark" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Immune Deconvolution Benchmark" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm">


<meta name="date" content="2018-05-08">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="spillover-analysis.html">
<link rel="next" href="validation-with-real-data.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#conceptual-differences-between-methods"><i class="fa fa-check"></i><b>1.1</b> Conceptual differences between methods</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#gene-set-based-methods-allow-comparison-between-samples-but-not-between-cell-types"><i class="fa fa-check"></i><b>1.1.1</b> Gene-set based methods (allow comparison between samples, but not between cell types)</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#deconvolution-based-methods-relative-fractions-allow-comparisons-between-cell-types-but-not-samples"><i class="fa fa-check"></i><b>1.1.2</b> Deconvolution-based methods (relative fractions, allow comparisons between cell types, but not samples)</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#deconvolution-based-methods-absolute-fractions-allow-comparisons-between-cell-types-and-samples"><i class="fa fa-check"></i><b>1.1.3</b> Deconvolution-based methods (absolute fractions, allow comparisons between cell types and samples)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.1</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#validation-data-rnaseq-matched-gold-standard"><i class="fa fa-check"></i><b>2.2</b> Validation data (RNAseq + matched gold standard)</a><ul>
<li class="chapter" data-level="2.2.1" data-path="input-data.html"><a href="input-data.html#hoeck-data"><i class="fa fa-check"></i><b>2.2.1</b> Hoeck data</a></li>
<li class="chapter" data-level="2.2.2" data-path="input-data.html"><a href="input-data.html#schelker-data"><i class="fa fa-check"></i><b>2.2.2</b> Schelker data</a></li>
<li class="chapter" data-level="2.2.3" data-path="input-data.html"><a href="input-data.html#racle-data"><i class="fa fa-check"></i><b>2.2.3</b> Racle data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-single-cell-sequencing-for-counting-immune-cells"><i class="fa fa-check"></i><b>3.1</b> Validity of single cell sequencing for counting immune cells</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data"><i class="fa fa-check"></i><b>3.2</b> Validity of simulated bulk tissues from single cell sequencing data</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-samplese-to-bulk-samples"><i class="fa fa-check"></i><b>3.2.1</b> Compare simululated samplese to bulk samples</a></li>
<li class="chapter" data-level="3.2.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-the-predicted-fractions"><i class="fa fa-check"></i><b>3.2.2</b> Compare the predicted fractions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>4</b> Spillover analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.1</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="4.1.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#run-timer"><i class="fa fa-check"></i><b>4.1.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#results"><i class="fa fa-check"></i><b>4.2</b> Results</a><ul>
<li class="chapter" data-level="4.2.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#calculate-signal-to-noise-ratio-for-each-cell-type"><i class="fa fa-check"></i><b>4.2.1</b> Calculate signal to noise ratio for each cell type</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="mixing-study.html"><a href="mixing-study.html"><i class="fa fa-check"></i><b>5</b> Mixing study</a><ul>
<li class="chapter" data-level="5.1" data-path="mixing-study.html"><a href="mixing-study.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>5.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="5.2" data-path="mixing-study.html"><a href="mixing-study.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>5.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="5.3" data-path="mixing-study.html"><a href="mixing-study.html#run-the-deconvolution-1"><i class="fa fa-check"></i><b>5.3</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="5.3.1" data-path="mixing-study.html"><a href="mixing-study.html#run-timer-1"><i class="fa fa-check"></i><b>5.3.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="mixing-study.html"><a href="mixing-study.html#results-1"><i class="fa fa-check"></i><b>5.4</b> Results</a><ul>
<li class="chapter" data-level="5.4.1" data-path="mixing-study.html"><a href="mixing-study.html#calculate-other-cells-for-all-absolute-methods"><i class="fa fa-check"></i><b>5.4.1</b> Calculate “Other cells” for all absolute methods</a></li>
<li class="chapter" data-level="5.4.2" data-path="mixing-study.html"><a href="mixing-study.html#correlation-plots"><i class="fa fa-check"></i><b>5.4.2</b> Correlation plots</a></li>
<li class="chapter" data-level="5.4.3" data-path="mixing-study.html"><a href="mixing-study.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>5.4.3</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="5.4.4" data-path="mixing-study.html"><a href="mixing-study.html#absolute-error-for-all-qualifying-methods"><i class="fa fa-check"></i><b>5.4.4</b> Absolute error for all qualifying methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>6</b> Validation with real data</a><ul>
<li class="chapter" data-level="6.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#run-the-deconvolution-2"><i class="fa fa-check"></i><b>6.1</b> Run the deconvolution</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>7</b> Conclusions</a></li>
<li class="chapter" data-level="8" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>8</b> Materials and Methods</a><ul>
<li class="chapter" data-level="8.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#data-acquisition"><i class="fa fa-check"></i><b>8.1</b> Data acquisition</a></li>
<li class="chapter" data-level="8.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#validation-datasets-flow-cytometry-scrnaseq-imageing-matched-rna-seq"><i class="fa fa-check"></i><b>8.2</b> Validation datasets (Flow cytometry, scRNAseq, Imageing + matched RNA seq)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Immune Deconvolution Benchmark</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mixing-study" class="section level1">
<h1><span class="header-section-number">5</span> Mixing study</h1>
<p>In this chapter, we will use single cell data from <span class="citation">(Schelker et al. <a href="#ref-Schelker2017">2017</a>)</span> to create simulated bulk RNAseq samples of which we know the true cell proportions. We futher use these samples to benchmark the performance of different recently published immune deconvolution tools.</p>
<div id="average-fraction-of-tumour-cells" class="section level2">
<h2><span class="header-section-number">5.1</span> Average fraction of tumour cells</h2>
<p>To obtain representatitive simulated samples, we are interested in the average fraction of tumour cells vs immune cells in a mixture.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell_type_table =<span class="st"> </span><span class="kw">pData</span>(single_cell_schelker2) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">filter</span>(source <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;melanoma&quot;</span>, <span class="st">&quot;ascites&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">sample=</span><span class="kw">paste</span>(source, donor, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(source, sample, cell_type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">n=</span><span class="kw">n</span>()) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">freq=</span>n<span class="op">/</span><span class="kw">sum</span>(n))

<span class="kw">ggplot</span>(cell_type_table, <span class="kw">aes</span>(<span class="dt">x=</span>sample, <span class="dt">y=</span>freq)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="kw">aes</span>(<span class="dt">fill=</span>cell_type), <span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust=</span><span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>color_scales<span class="op">$</span>cell_types)</code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cancer_cells =<span class="st"> </span>cell_type_table <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(cell_type <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Melanoma cells&quot;</span>, <span class="st">&quot;Ovarian carcinoma cells&quot;</span>)) 


cancer_cell_param =<span class="st"> </span>MASS<span class="op">::</span><span class="kw">fitdistr</span>(cancer_cells<span class="op">$</span>freq, <span class="st">&quot;normal&quot;</span>)</code></pre></div>
</div>
<div id="create-simulated-bulk-tissues" class="section level2">
<h2><span class="header-section-number">5.2</span> Create simulated bulk tissues</h2>
<p>The fractions of a sample are randomly assigned in the following procedure:</p>
<ol style="list-style-type: decimal">
<li>Draw a random tumour cell content from the distribution fitted above</li>
<li>The first half of the samples will use melanoma cells, the second half ovarian cancer cells.</li>
<li>Assign the remaining fraction (=not cancer cells) randomly to the remaining cell types (B cells, CD8+ T cells, Melanoma cells, CD4+ T cells, Macrophages/Monocytes, regulatory T cells, Cancer associated fibroblasts, Dendritic cells, Endothelial cells, Natural killer cells, Ovarian carcinoma cells)</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">42</span>)

cancer_cells =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Melanoma cells&quot;</span>, <span class="st">&quot;Ovarian carcinoma cells&quot;</span>)
immune_cells =<span class="st"> </span>available_cell_types[<span class="op">!</span>(available_cell_types <span class="op">%in%</span><span class="st"> </span>cancer_cells)]
is_melanoma =<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="ot">TRUE</span>, <span class="dv">50</span>), <span class="kw">rep</span>(<span class="ot">FALSE</span>, <span class="dv">50</span>))

cell_fractions =<span class="st"> </span><span class="kw">lapply</span>(is_melanoma, <span class="cf">function</span>(melanoma) {
  cancer_fraction =<span class="st"> </span><span class="kw">rnorm</span>(<span class="dv">1</span>, <span class="dt">mean=</span>cancer_cell_param<span class="op">$</span>estimate[<span class="dv">1</span>], <span class="dt">sd=</span>cancer_cell_param<span class="op">$</span>estimate[<span class="dv">2</span>])
  cancer_fraction =<span class="st"> </span><span class="kw">ifelse</span>(cancer_fraction <span class="op">&lt;</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">0</span>, cancer_fraction)
  cancer_fraction =<span class="st"> </span><span class="kw">ifelse</span>(cancer_fraction <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>, <span class="dv">1</span>, cancer_fraction)
  
  <span class="co"># associate fraction randomly ot Melanoma or Ovarian</span>
  <span class="cf">if</span>(melanoma) {
    cancer_fractions =<span class="st"> </span><span class="kw">list</span>(cancer_fraction, <span class="dv">0</span>)
  } <span class="cf">else</span> {
    cancer_fractions =<span class="st"> </span><span class="kw">list</span>(<span class="dv">0</span>, cancer_fraction)
  }
  <span class="kw">names</span>(cancer_fractions) =<span class="st"> </span>cancer_cells
  
  <span class="co"># compute random fractions for other cells</span>
  remaining_fraction =<span class="st"> </span><span class="dv">1</span> <span class="op">-</span><span class="st"> </span>cancer_fraction
  rnd =<span class="st"> </span><span class="kw">sample</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">100</span>, <span class="kw">length</span>(immune_cells), <span class="ot">TRUE</span>)
  tmp_fractions =<span class="st"> </span>(remaining_fraction <span class="op">*</span><span class="st"> </span>rnd) <span class="op">/</span><span class="st"> </span><span class="kw">sum</span>(rnd)
  <span class="kw">names</span>(tmp_fractions) =<span class="st"> </span>immune_cells
  
  <span class="kw">c</span>(cancer_fractions, tmp_fractions)
}) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_rows</span>()

testthat<span class="op">::</span><span class="kw">test_that</span>(<span class="st">&quot;all fractions sum up to 1&quot;</span>, <span class="kw">expect_equal</span>(<span class="kw">apply</span>(cell_fractions, <span class="dv">1</span>, sum), <span class="kw">rep</span>(<span class="dv">1</span>, <span class="dv">100</span>)))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">42</span>)
bulk_eset =<span class="st"> </span><span class="kw">make_bulk_eset</span>(<span class="dt">eset=</span>single_cell_schelker2, <span class="dt">cell_fractions =</span> cell_fractions, <span class="dt">n_cells=</span><span class="dv">500</span>)</code></pre></div>
</div>
<div id="run-the-deconvolution-1" class="section level2">
<h2><span class="header-section-number">5.3</span> Run the deconvolution</h2>
<p>We first run all methods that are integrated in the <code>immune_deconvolution_methods</code> are package automatically in a loop.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results =<span class="st"> </span><span class="kw">lapply</span>(immunedeconv<span class="op">::</span>deconvolution_methods, <span class="cf">function</span>(method) {
  <span class="kw">deconvolute</span>(bulk_eset, method, <span class="dt">column=</span><span class="st">&quot;gene_symbol&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">method=</span>method)
})</code></pre></div>
<pre><code>## Warning in EPIC::EPIC(bulk = gene_expression_matrix, ...): The optimization didn&#39;t fully converge for some samples:
## value4; value5; value7; value34; value51; value62; value64; value66; value78
##  - check fit.gof for the convergeCode and convergeMessage</code></pre>
<pre><code>## Warning in EPIC::EPIC(bulk = gene_expression_matrix, ...): mRNA_cell value
## unknown for some cell types: CAFs, Endothelial - using the default value of
## 0.4 for these but this might bias the true cell proportions from all cell
## types.</code></pre>
<pre><code>## 
## Running quanTIseq deconvolution module</code></pre>
<pre><code>## Gene expression normalization and re-annotation (arrays: FALSE)</code></pre>
<pre><code>## Removing 17 noisy genes</code></pre>
<pre><code>## Signature genes found in data set: 140/153 (91.5%)</code></pre>
<pre><code>## Mixture deconvolution (method: lsei)</code></pre>
<pre><code>## Deconvolution sucessful!</code></pre>
<pre><code>## [1] &quot;Num. of genes: 10462&quot;
## Estimating ssGSEA scores for 489 gene sets.
## 
  |                                                                       
  |                                                                 |   0%Using parallel with 4 cores
## 
  |                                                                       
  |=                                                                |   1%
  |                                                                       
  |=                                                                |   2%
  |                                                                       
  |==                                                               |   3%
  |                                                                       
  |===                                                              |   4%
  |                                                                       
  |===                                                              |   5%
  |                                                                       
  |====                                                             |   6%
  |                                                                       
  |=====                                                            |   7%
  |                                                                       
  |=====                                                            |   8%
  |                                                                       
  |======                                                           |   9%
  |                                                                       
  |======                                                           |  10%
  |                                                                       
  |=======                                                          |  11%
  |                                                                       
  |========                                                         |  12%
  |                                                                       
  |========                                                         |  13%
  |                                                                       
  |=========                                                        |  14%
  |                                                                       
  |==========                                                       |  15%
  |                                                                       
  |==========                                                       |  16%
  |                                                                       
  |===========                                                      |  17%
  |                                                                       
  |============                                                     |  18%
  |                                                                       
  |============                                                     |  19%
  |                                                                       
  |=============                                                    |  20%
  |                                                                       
  |==============                                                   |  21%
  |                                                                       
  |==============                                                   |  22%
  |                                                                       
  |===============                                                  |  23%
  |                                                                       
  |================                                                 |  24%
  |                                                                       
  |================                                                 |  25%
  |                                                                       
  |=================                                                |  26%
  |                                                                       
  |==================                                               |  27%
  |                                                                       
  |==================                                               |  28%
  |                                                                       
  |===================                                              |  29%
  |                                                                       
  |====================                                             |  30%
  |                                                                       
  |====================                                             |  31%
  |                                                                       
  |=====================                                            |  32%
  |                                                                       
  |=====================                                            |  33%
  |                                                                       
  |======================                                           |  34%
  |                                                                       
  |=======================                                          |  35%
  |                                                                       
  |=======================                                          |  36%
  |                                                                       
  |========================                                         |  37%
  |                                                                       
  |=========================                                        |  38%
  |                                                                       
  |=========================                                        |  39%
  |                                                                       
  |==========================                                       |  40%
  |                                                                       
  |===========================                                      |  41%
  |                                                                       
  |===========================                                      |  42%
  |                                                                       
  |============================                                     |  43%
  |                                                                       
  |=============================                                    |  44%
  |                                                                       
  |=============================                                    |  45%
  |                                                                       
  |==============================                                   |  46%
  |                                                                       
  |===============================                                  |  47%
  |                                                                       
  |===============================                                  |  48%
  |                                                                       
  |================================                                 |  49%
  |                                                                       
  |================================                                 |  50%
  |                                                                       
  |=================================                                |  51%
  |                                                                       
  |==================================                               |  52%
  |                                                                       
  |==================================                               |  53%
  |                                                                       
  |===================================                              |  54%
  |                                                                       
  |====================================                             |  55%
  |                                                                       
  |====================================                             |  56%
  |                                                                       
  |=====================================                            |  57%
  |                                                                       
  |======================================                           |  58%
  |                                                                       
  |======================================                           |  59%
  |                                                                       
  |=======================================                          |  60%
  |                                                                       
  |========================================                         |  61%
  |                                                                       
  |========================================                         |  62%
  |                                                                       
  |=========================================                        |  63%
  |                                                                       
  |==========================================                       |  64%
  |                                                                       
  |==========================================                       |  65%
  |                                                                       
  |===========================================                      |  66%
  |                                                                       
  |============================================                     |  67%
  |                                                                       
  |============================================                     |  68%
  |                                                                       
  |=============================================                    |  69%
  |                                                                       
  |==============================================                   |  70%
  |                                                                       
  |==============================================                   |  71%
  |                                                                       
  |===============================================                  |  72%
  |                                                                       
  |===============================================                  |  73%
  |                                                                       
  |================================================                 |  74%
  |                                                                       
  |=================================================                |  75%
  |                                                                       
  |=================================================                |  76%
  |                                                                       
  |==================================================               |  77%
  |                                                                       
  |===================================================              |  78%
  |                                                                       
  |===================================================              |  79%
  |                                                                       
  |====================================================             |  80%
  |                                                                       
  |=====================================================            |  81%
  |                                                                       
  |=====================================================            |  82%
  |                                                                       
  |======================================================           |  83%
  |                                                                       
  |=======================================================          |  84%
  |                                                                       
  |=======================================================          |  85%
  |                                                                       
  |========================================================         |  86%
  |                                                                       
  |=========================================================        |  87%
  |                                                                       
  |=========================================================        |  88%
  |                                                                       
  |==========================================================       |  89%
  |                                                                       
  |==========================================================       |  90%
  |                                                                       
  |===========================================================      |  91%
  |                                                                       
  |============================================================     |  92%
  |                                                                       
  |============================================================     |  93%
  |                                                                       
  |=============================================================    |  94%
  |                                                                       
  |==============================================================   |  95%
  |                                                                       
  |==============================================================   |  96%
  |                                                                       
  |===============================================================  |  97%
  |                                                                       
  |================================================================ |  98%
  |                                                                       
  |================================================================ |  99%
  |                                                                       
  |=================================================================| 100%</code></pre>
<div id="run-timer-1" class="section level3">
<h3><span class="header-section-number">5.3.1</span> run TIMER</h3>
<p>TIMER <span class="citation">(B. Li et al. <a href="#ref-TIMER2016">2016</a>)</span> is only available as a web resource. Moreover, the algorithm is adapted for each cancer type, so that we have to choose the corresponding option (Melanoma/Ovarian) when running the algorithm.</p>
<p>We therefore export the data required to run the algorithm and re-import the results obtained from the web-tool.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">eset_mat =<span class="st"> </span><span class="kw">eset_to_matrix</span>(bulk_eset, <span class="st">&quot;gene_symbol&quot;</span>)
immunedeconv<span class="op">::</span><span class="kw">export_for_timer</span>(eset_mat[,is_melanoma], <span class="dt">path=</span><span class="st">&quot;../results/timer_input_melanoma.tsv&quot;</span>)
immunedeconv<span class="op">::</span><span class="kw">export_for_timer</span>(eset_mat[,<span class="op">!</span>is_melanoma], <span class="dt">path=</span><span class="st">&quot;../results/timer_input_ovarian.tsv&quot;</span>)

res_timer_melanoma =<span class="st"> </span>immunedeconv<span class="op">::</span><span class="kw">import_from_timer</span>(<span class="st">&quot;../data/timer/timer_result_melanoma.csv&quot;</span>)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   sampleID = col_character(),
##   B_cell = col_double(),
##   T_cell.CD4 = col_double(),
##   T_cell.CD8 = col_double(),
##   Neutrophil = col_double(),
##   Macrophage = col_double(),
##   DC = col_double()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_timer_ovarian =<span class="st"> </span>immunedeconv<span class="op">::</span><span class="kw">import_from_timer</span>(<span class="st">&quot;../data/timer/timer_result_ovarian.csv&quot;</span>)</code></pre></div>
<pre><code>## Parsed with column specification:
## cols(
##   sampleID = col_character(),
##   B_cell = col_double(),
##   T_cell.CD4 = col_double(),
##   T_cell.CD8 = col_double(),
##   Neutrophil = col_double(),
##   Macrophage = col_double(),
##   DC = col_double()
## )</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res_timer =<span class="st"> </span><span class="kw">inner_join</span>(res_timer_melanoma, res_timer_ovarian, <span class="dt">by=</span><span class="st">&quot;cell_type&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">method=</span><span class="st">&quot;timer&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">abs_methods =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;cibersort_abs&quot;</span>, <span class="st">&quot;epic&quot;</span>, <span class="st">&quot;quantiseq&quot;</span>, <span class="st">&quot;xcell&quot;</span>)</code></pre></div>
</div>
</div>
<div id="results-1" class="section level2">
<h2><span class="header-section-number">5.4</span> Results</h2>
<p>Here, we map the results back to the “gold standard”. We aggregate the results of the different methods into a single table and clean it up for further processing.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results2 =<span class="st"> </span>all_results <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">c</span>(<span class="kw">list</span>(res_timer))

<span class="co"># summarise cell_types from the method that only map to a single cell_type in the reference dataset (namely monocytes and macrophages)</span>
all_results3 =<span class="st"> </span><span class="kw">lapply</span>(all_results2, <span class="cf">function</span>(res) {
  res <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">map_results_to_dataset</span>(<span class="st">&quot;single_cell_schelker&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">    </span><span class="kw">group_by</span>(method, single_cell_schelker) <span class="op">%&gt;%</span><span class="st">   </span>
<span class="st">    </span><span class="kw">summarise_all</span>(<span class="kw">funs</span>(sum)) 
})

all_results_tidy =<span class="st"> </span>all_results3 <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">bind_rows</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rename</span>(<span class="dt">cell_type =</span> single_cell_schelker) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(sample, estimate, <span class="op">-</span>cell_type, <span class="op">-</span>method) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">arrange</span>(method, cell_type, sample)

gold_standard =<span class="st"> </span><span class="kw">pData</span>(bulk_eset) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;sample&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(cell_type, true_fraction, <span class="op">-</span>sample)</code></pre></div>
<div id="calculate-other-cells-for-all-absolute-methods" class="section level3">
<h3><span class="header-section-number">5.4.1</span> Calculate “Other cells” for all absolute methods</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">other_cells_estimates =<span class="st"> </span>all_results_tidy <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(method <span class="op">%in%</span><span class="st"> </span>abs_methods) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(method, sample) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">cell_type=</span><span class="st">&quot;Other&quot;</span>, <span class="dt">estimate=</span><span class="kw">max</span>(<span class="dv">0</span>, <span class="dv">1</span><span class="op">-</span><span class="kw">sum</span>(estimate)))

other_cells_true_fractions =<span class="st"> </span>gold_standard <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">inner_join</span>(all_results_tidy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(method, sample) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">distinct</span>(), <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">anti_join</span>(all_results_tidy, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;method&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(method, sample) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="dt">cell_type=</span><span class="st">&quot;Other&quot;</span>, <span class="dt">true_fraction=</span><span class="kw">sum</span>(true_fraction))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results_tidy2 =<span class="st"> </span><span class="kw">bind_rows</span>(all_results_tidy, other_cells_estimates)

<span class="co"># we need to add the methods (full join) to each cell type as </span>
<span class="co"># the &quot;Other&quot; fractions are method specific</span>
<span class="co"># (all cell types that are not predicted by the method)</span>
gold_standard2 =<span class="st"> </span>gold_standard <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">crossing</span>(<span class="dt">method=</span><span class="kw">unique</span>(all_results_tidy<span class="op">$</span>method)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">bind_rows</span>(other_cells_true_fractions)

results_with_gold_standard =<span class="st"> </span>all_results_tidy2 <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">inner_join</span>(gold_standard2, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;cell_type&quot;</span>, <span class="st">&quot;method&quot;</span>))</code></pre></div>
</div>
<div id="correlation-plots" class="section level3">
<h3><span class="header-section-number">5.4.2</span> Correlation plots</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">results_with_gold_standard <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>true_fraction, <span class="dt">y=</span>estimate)) <span class="op">+</span>
<span class="st">  </span><span class="kw">facet_grid</span>(method <span class="op">~</span><span class="st"> </span>cell_type, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>() <span class="op">+</span>
<span class="st">  </span><span class="kw">stat_smooth</span>(<span class="kw">aes</span>(<span class="dt">color=</span>method), <span class="dt">method=</span><span class="st">&quot;lm&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>color_scales<span class="op">$</span>methods) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">scale_x_continuous</span>(<span class="dt">breaks=</span><span class="kw">c</span>(.<span class="dv">2</span>, .<span class="dv">5</span>))</code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-35-1.png" width="1344" /></p>
</div>
<div id="calculate-correlations-for-each-method-and-cell-type" class="section level3">
<h3><span class="header-section-number">5.4.3</span> Calculate correlations for each method and cell type</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">make_cor =<span class="st"> </span><span class="cf">function</span>(estimate, true_fraction) {
  test_res =<span class="st"> </span><span class="kw">cor.test</span>(estimate, true_fraction, <span class="dt">method=</span><span class="st">&quot;pearson&quot;</span>)
  <span class="kw">data.frame</span>(
    <span class="dt">pearson=</span>test_res<span class="op">$</span>estimate,
    <span class="dt">conf_int_lower=</span>test_res<span class="op">$</span>conf.int[<span class="dv">1</span>],
    <span class="dt">conf_int_upper=</span>test_res<span class="op">$</span>conf.int[<span class="dv">2</span>]
  )
}

correlations =<span class="st"> </span>results_with_gold_standard <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">group_by</span>(cell_type, method) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">do</span>(<span class="kw">make_cor</span>(.<span class="op">$</span>estimate, .<span class="op">$</span>true_fraction))

correlations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>method, <span class="dt">y=</span>pearson)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_bar</span>(<span class="kw">aes</span>(<span class="dt">fill=</span>method), <span class="dt">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_errorbar</span>(<span class="kw">aes</span>(<span class="dt">ymin=</span>conf_int_lower, <span class="dt">ymax=</span>conf_int_upper), <span class="dt">width=</span>.<span class="dv">2</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>cell_type, <span class="dt">nrow=</span><span class="dv">2</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust=</span><span class="dv">1</span>),
        <span class="dt">legend.position=</span><span class="st">&quot;top&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>color_scales<span class="op">$</span>methods)</code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-36-1.png" width="576" /></p>
</div>
<div id="absolute-error-for-all-qualifying-methods" class="section level3">
<h3><span class="header-section-number">5.4.4</span> Absolute error for all qualifying methods</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">results_with_gold_standard <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(method <span class="op">%in%</span><span class="st"> </span>abs_methods) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">error=</span><span class="kw">abs</span>(estimate<span class="op">-</span>true_fraction)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">square_error=</span>error<span class="op">^</span><span class="dv">2</span>) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>method, <span class="dt">y=</span>square_error)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>cell_type, <span class="dt">nrow=</span><span class="dv">1</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="kw">aes</span>(<span class="dt">fill=</span>method), <span class="dt">width=</span><span class="dv">1</span>, <span class="dt">position=</span><span class="st">&quot;dodge&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust=</span><span class="dv">1</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_manual</span>(<span class="dt">values=</span>color_scales<span class="op">$</span>methods) </code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-37-1.png" width="960" /></p>

</div>
</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Schelker2017">
<p>Schelker, Max, Sonia Feau, Jinyan Du, Nav Ranu, Edda Klipp, Gavin MacBeath, Birgit Schoeberl, and Andreas Raue. 2017. “Estimation of immune cell content in tumour tissue using single-cell RNA-seq data.” <em>Nature Communications</em> 8 (1): 2032. doi:<a href="https://doi.org/10.1038/s41467-017-02289-3">10.1038/s41467-017-02289-3</a>.</p>
</div>
<div id="ref-TIMER2016">
<p>Li, Bo, Eric Severson, Jean-Christophe Pignon, Haoquan Zhao, Taiwen Li, Jesse Novak, Peng Jiang, et al. 2016. “Comprehensive analyses of tumor immunity: implications for cancer immunotherapy.” <em>Genome Biology</em> 17 (1): 174. doi:<a href="https://doi.org/10.1186/s13059-016-1028-7">10.1186/s13059-016-1028-7</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="spillover-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="validation-with-real-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
