<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>A quantitative comparison of cell type deconvolution methods for immuno-oncology</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="A quantitative comparison of cell type deconvolution methods for immuno-oncology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A quantitative comparison of cell type deconvolution methods for immuno-oncology" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm (1,2,*), Francesca Finotello (3), Florent Petitprez (4,5), Jitao David Zhang (6), Jan Baumbach (1), Wolf H. Fridman (4), Markus List (7), Tatsiana Aneichyk (2)">


<meta name="date" content="2018-11-14">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="validitiy-of-single-cell-rna-seq-as-reference.html">
<link rel="next" href="validation-with-real-data.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#authors-affiliations"><i class="fa fa-check"></i><b>1.1</b> Author’s affiliations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-reference-samples"><i class="fa fa-check"></i><b>2.3</b> Immune cell reference samples</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#pbmc-samples-from-hoek2015"><i class="fa fa-check"></i><b>2.4</b> 8 PBMC samples from <span class="citation">Hoek et al. (<span>2015</span>)</span></a></li>
<li class="chapter" data-level="2.5" data-path="input-data.html"><a href="input-data.html#ovarian-cancer-ascites-samples-from-schelker2017"><i class="fa fa-check"></i><b>2.5</b> 3 ovarian cancer ascites samples from <span class="citation">Schelker et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.6" data-path="input-data.html"><a href="input-data.html#metastatic-melanoma-samples-from-epic2017"><i class="fa fa-check"></i><b>2.6</b> 4 metastatic melanoma samples from <span class="citation">Racle et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.7" data-path="input-data.html"><a href="input-data.html#data-sanity-checks"><i class="fa fa-check"></i><b>2.7</b> Data sanity checks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-to-genuine-bulk-samples"><i class="fa fa-check"></i><b>3.1</b> Compare simululated to genuine bulk samples</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#enrichment-analysis-test-for-systematic-bias"><i class="fa fa-check"></i><b>3.2</b> Enrichment analysis: Test for systematic Bias</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#immune-relevant-genes-are-consistent-between-sc-and-bulk-data."><i class="fa fa-check"></i><b>3.2.1</b> Immune-relevant genes are consistent between sc and bulk data.</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-predicted-fractions"><i class="fa fa-check"></i><b>3.3</b> Compare predicted fractions</a></li>
<li class="chapter" data-level="3.4" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlation-with-immune-reference-samples."><i class="fa fa-check"></i><b>3.4</b> Correlation with immune reference samples.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html"><i class="fa fa-check"></i><b>4</b> Simulation benchmark</a><ul>
<li class="chapter" data-level="4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>4.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>4.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.3</b> Run the deconvolution</a></li>
<li class="chapter" data-level="4.4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#results"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#correlation-plots"><i class="fa fa-check"></i><b>4.4.1</b> Correlation plots</a></li>
<li class="chapter" data-level="4.4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>4.4.2</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="4.4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#absolute-deviation"><i class="fa fa-check"></i><b>4.4.3</b> Absolute deviation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>5</b> Validation with real data</a><ul>
<li class="chapter" data-level="5.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between--and-within-sample-comparisons"><i class="fa fa-check"></i><b>5.1</b> Between- and within-sample comparisons</a><ul>
<li class="chapter" data-level="5.1.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#absolute-scores"><i class="fa fa-check"></i><b>5.1.1</b> Absolute scores</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#within-sample-comparison"><i class="fa fa-check"></i><b>5.2</b> Within-sample comparison</a></li>
<li class="chapter" data-level="5.3" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between-sample-comparisons"><i class="fa fa-check"></i><b>5.3</b> Between-sample comparisons</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html"><i class="fa fa-check"></i><b>6</b> Detection limit and false positive predictions</a><ul>
<li class="chapter" data-level="6.1" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#predictions-with-increasing-immune-cell-content"><i class="fa fa-check"></i><b>6.1</b> Predictions with increasing immune cell content</a></li>
<li class="chapter" data-level="6.2" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#detection-limit-and-false-positives"><i class="fa fa-check"></i><b>6.2</b> Detection Limit and False positives</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>7</b> Spillover Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#complete-spillover-matrix"><i class="fa fa-check"></i><b>7.1</b> Complete Spillover Matrix</a></li>
<li class="chapter" data-level="7.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-signal-to-noise-ratio"><i class="fa fa-check"></i><b>7.2</b> Summary figure: Signal to noise ratio</a></li>
<li class="chapter" data-level="7.3" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-migration-charts"><i class="fa fa-check"></i><b>7.3</b> Summary figure: migration charts</a></li>
<li class="chapter" data-level="7.4" data-path="spillover-analysis.html"><a href="spillover-analysis.html#investigate-marker-genes"><i class="fa fa-check"></i><b>7.4</b> Investigate marker genes</a></li>
<li class="chapter" data-level="7.5" data-path="spillover-analysis.html"><a href="spillover-analysis.html#check-expression-of-markers-in-detection-limit-simulation-dataset."><i class="fa fa-check"></i><b>7.5</b> Check expression of markers in “detection limit” simulation dataset.</a><ul>
<li class="chapter" data-level="7.5.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#expression-of-marker-genes-in-the-cell-populations"><i class="fa fa-check"></i><b>7.5.1</b> Expression of marker genes in the cell populations</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="spillover-analysis.html"><a href="spillover-analysis.html#deconvolution-results-before-and-after"><i class="fa fa-check"></i><b>7.6</b> Deconvolution results before and after</a><ul>
<li class="chapter" data-level="7.6.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#the-same-analysis-but-for-t-cd4-vs.non-immune-cells."><i class="fa fa-check"></i><b>7.6.1</b> The same analysis, but for T CD4 vs. non-immune cells.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html"><i class="fa fa-check"></i><b>8</b> Creating publication ready figures</a><ul>
<li class="chapter" data-level="8.1" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#predictions-on-simulated-vs.genuine-bulk"><i class="fa fa-check"></i><b>8.1</b> Predictions on simulated vs. genuine bulk</a></li>
<li class="chapter" data-level="8.2" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#benchmark-results"><i class="fa fa-check"></i><b>8.2</b> Benchmark results</a></li>
<li class="chapter" data-level="8.3" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#detection-limit-false-positive-figure"><i class="fa fa-check"></i><b>8.3</b> detection limit / false positive figure</a></li>
<li class="chapter" data-level="8.4" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#migration-charts-for-spillover-analysis"><i class="fa fa-check"></i><b>8.4</b> Migration charts for Spillover analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A quantitative comparison of cell type deconvolution methods for immuno-oncology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="simulation-benchmark" class="section level1">
<h1><span class="header-section-number">4</span> Simulation benchmark</h1>
<p>In this chapter, we will use single cell data from <span class="citation">Schelker et al. (<a href="#ref-Schelker2017">2017</a>)</span> to create simulated bulk RNAseq samples
of which we know the true cell proportions (=artificial gold standard). Using these data, we
can assess the performance of immune deconvolution tools.</p>
<div id="average-fraction-of-tumour-cells" class="section level2">
<h2><span class="header-section-number">4.1</span> Average fraction of tumour cells</h2>
<p>To obtain representatitive simulated samples, we are interested in the average fraction of tumour cells vs immune cells in a mixture.</p>
<div class="figure"><span id="fig:tumor-ct-fractions"></span>
<img src="_main_files/figure-html/tumor-ct-fractions-1.png" alt="proportion of cell types by tumor sample" width="672" />
<p class="caption">
Figure 4.1: proportion of cell types by tumor sample
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">cancer_cell_param =<span class="st"> </span>MASS<span class="op">::</span><span class="kw">fitdistr</span>(cancer_cells<span class="op">$</span>freq, <span class="st">&quot;normal&quot;</span>)</a></code></pre></div>
<p>The empirical distribution of the cancer fraction is <span class="math inline">\(\sim\mathcal{N}(0.33, 0.3)\)</span>.</p>
</div>
<div id="create-simulated-bulk-tissues" class="section level2">
<h2><span class="header-section-number">4.2</span> Create simulated bulk tissues</h2>
<p>The fractions of a sample are randomly assigned in the following procedure:</p>
<ol style="list-style-type: decimal">
<li>Draw a random tumour cell content from the distribution fitted above</li>
<li>The first half of the samples will use melanoma cells, the second half ovarian cancer cells.</li>
<li>Assign the remaining fraction (=not cancer cells) randomly to the remaining cell types (B cell, T cell CD8+, Melanoma cell, T cell CD4+ (non-regulatory), Macrophage/Monocyte, T cell regulatory (Tregs), Cancer associated fibroblast, Dendritic cell, Endothelial cell, NK cell, PBMC, Ovarian carcinoma cell)</li>
</ol>
<p>Here, we generate a simulated bulk RNA-seq ExpressionSet:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="kw">set.seed</span>(<span class="dv">42</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">bulk_eset =<span class="st"> </span><span class="kw">make_bulk_eset</span>(<span class="dt">eset=</span>single_cell_schelker<span class="op">$</span>eset,</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                           <span class="dt">cell_fractions =</span> cell_fractions,</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">                           <span class="dt">n_cells=</span><span class="dv">500</span>)</a></code></pre></div>
</div>
<div id="run-the-deconvolution" class="section level2">
<h2><span class="header-section-number">4.3</span> Run the deconvolution</h2>
<p>We run the methods with <code>tumor = TRUE</code> and <code>scale_mrna = FALSE</code> where applicable, as
the mRNA scaling is inappropriate for already normalized single cells. In any case,
the mRNA scaling factors have no influence on the correlations between estimated and
known fractions.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">timer_indications =<span class="st"> </span><span class="kw">ifelse</span>(is_melanoma, <span class="st">&quot;SKCM&quot;</span>, <span class="st">&quot;OV&quot;</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">all_results =<span class="st"> </span><span class="kw">foreach</span>(<span class="dt">method=</span>config<span class="op">$</span>deconvolution_methods,</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">                      <span class="dt">.final =</span> <span class="cf">function</span>(x) {<span class="kw">setNames</span>(x, config<span class="op">$</span>deconvolution_methods)}) <span class="op">%dopar%</span><span class="st"> </span>{</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">  <span class="kw">deconvolute</span>(bulk_eset, method, <span class="dt">column=</span><span class="st">&quot;gene_symbol&quot;</span>, <span class="dt">indications =</span> timer_indications, <span class="dt">scale_mrna =</span> <span class="ot">FALSE</span>, <span class="dt">tumor =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">}</a></code></pre></div>
</div>
<div id="results" class="section level2">
<h2><span class="header-section-number">4.4</span> Results</h2>
<p>We perform an evaluation for the following cell types.
Note that some cell types are redundant (T cell CD4+ is a super-category of Tregs and non-regulatory CD4+ T cells).
As some methods provide deconvolution only at the CD4+ level, we compare both categories:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">show_cell_types =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;B cell&quot;</span>, <span class="st">&quot;Dendritic cell&quot;</span>, <span class="st">&quot;Macrophage/Monocyte&quot;</span>,</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">                    <span class="st">&quot;NK cell&quot;</span>, <span class="st">&quot;T cell CD4+&quot;</span>, <span class="st">&quot;T cell CD4+ (non-regulatory)&quot;</span>,</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">                    <span class="st">&quot;T cell regulatory (Tregs)&quot;</span>, <span class="st">&quot;T cell CD8+&quot;</span>,</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">                    <span class="st">&quot;Cancer associated fibroblast&quot;</span>, <span class="st">&quot;Endothelial cell&quot;</span>)</a></code></pre></div>
<p>The following methods do not have signatures to quantify “Macrophages/Monocytes”. EPIC and TIMER only provide a Macrophage signature, while MCP-counter only provides a signature “Monocytic lineage”, which also includes myeloid dendritic cells.
Unfortunately, our single cell dataset does not distinguish between Macrophages and Monocytes.
As we are only comparing correlations here, the subsitute signatures should still yield appropriate results, yet, we label the results
accordingly</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">substitute_macrophage_signature =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;epic&quot;</span>, <span class="st">&quot;timer&quot;</span>, <span class="st">&quot;mcp_counter&quot;</span>)</a></code></pre></div>
<p>Here, we map the results back to the “gold standard”. We aggregate the results of the different methods
into a single table and clean it up for further processing.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">results_with_gold_standard =<span class="st"> </span><span class="kw">inner_join</span>(all_results_tidy, gold_standard,</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">                                        <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;cell_type&quot;</span>))</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">res_mixing_study<span class="op">$</span>all_results =<span class="st"> </span>results_with_gold_standard</a></code></pre></div>
<div id="correlation-plots" class="section level3">
<h3><span class="header-section-number">4.4.1</span> Correlation plots</h3>
<div class="figure"><span id="fig:mixing-correlation-plot"></span>
<img src="_main_files/figure-html/mixing-correlation-plot-1.png" alt="The figure shows the correlation of predicted vs. known fractions on 100 simulated bulk RNA seq samples." width="1728" />
<p class="caption">
Figure 4.2: The figure shows the correlation of predicted vs. known fractions on 100 simulated bulk RNA seq samples.
</p>
</div>
</div>
<div id="calculate-correlations-for-each-method-and-cell-type" class="section level3">
<h3><span class="header-section-number">4.4.2</span> Calculate correlations for each method and cell type</h3>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">correlations_tab =<span class="st"> </span>correlations <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(cell_type, method, pearson) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="st">  </span><span class="kw">spread</span>(cell_type, pearson)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="kw">write_tsv</span>(correlations_tab, <span class="st">&quot;../results/tables/mixing_study_correlations.tsv&quot;</span>, <span class="dt">na=</span><span class="st">&quot;&quot;</span>)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-32"></span>
<img src="_main_files/figure-html/unnamed-chunk-32-1.png" alt="Correlations of predicted vs. known fractions on 100 simulated bulk RNA-seq samples, organized by cell type. " width="768" />
<p class="caption">
Figure 4.3: Correlations of predicted vs. known fractions on 100 simulated bulk RNA-seq samples, organized by cell type.
</p>
</div>
</div>
<div id="absolute-deviation" class="section level3">
<h3><span class="header-section-number">4.4.3</span> Absolute deviation</h3>
<p><strong>treat with caution!</strong></p>
<p>Next, we were interested in the absolute deviation of the values for methods
that compute absolute scores. We assess the absolute deviation using two
measures:</p>
<ul>
<li>by fitting a linear model to the values and denoting the slope. If the
absolute quantification was perfect, the slope would equal 1. Values
&lt; 1 indicate an underprediction and values &gt; 1 an overprediction of the
respective cell type.</li>
<li>by calculating the root mean square error (RMSE)</li>
</ul>
<p>Note that only EPIC and quanTIseq provide scores that can be interpreted as a cell fraction.
xCell and CIBERSORT abs. scale the output score to be ‘absolute’:</p>
<blockquote>
<p>xCell does an attempt to make the scores resemble percentages, but it is a hard problem, and is very platform and experiment specific. (<a href="https://github.com/dviraran/xCell/blob/060764fa254904f80ed776a19bdd9ecbfeedc2be/README.Md">xCell on github</a></p>
</blockquote>
<blockquote>
<p>Absolute mode scales relative cellular fractions into a score of arbitrary units that reflects the absolute proportion
of each cell type in a mixture. Although not currently expressed as a fraction, the absolute score can be directly
compared across cell types (i.e., relative differences between cell types are maintained) (<a href="https://cibersort.stanford.edu/">cibersort
FAQ</a>)</p>
</blockquote>
<p><em>Also note that these values are not necessarily
accurate for quanTIseq and EPIC, which account for the different mRNA contents
of different cell types.</em> As single cell data is already normalized on
a per-cell level, the scaling factors are most likely not appropriate with the
simulated data. We generated the same plots on the validation data sets (geniune bulk RNA-seq + FACS),
which we expect to be more representative!</p>
<div class="figure"><span id="fig:unnamed-chunk-33"></span>
<img src="_main_files/figure-html/unnamed-chunk-33-1.png" alt="Absolute deviation of the predictions represented as the slope of a linear model. Confidence intervals are computed using `confint` on the result of `lm`." width="1152" />
<p class="caption">
Figure 4.4: Absolute deviation of the predictions represented as the slope of a linear model. Confidence intervals are computed using <code>confint</code> on the result of <code>lm</code>.
</p>
</div>
<div class="figure"><span id="fig:abs-rmse"></span>
<img src="_main_files/figure-html/abs-rmse-1.png" alt="Absolute deviation of the predictions represented as RMSE." width="1152" />
<p class="caption">
Figure 4.5: Absolute deviation of the predictions represented as RMSE.
</p>
</div>

</div>
</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Schelker2017">
<p>Schelker, Max, Sonia Feau, Jinyan Du, Nav Ranu, Edda Klipp, Gavin MacBeath, Birgit Schoeberl, and Andreas Raue. 2017. “Estimation of immune cell content in tumour tissue using single-cell RNA-seq data.” <em>Nature Communications</em> 8 (1): 2032. <a href="https://doi.org/10.1038/s41467-017-02289-3">https://doi.org/10.1038/s41467-017-02289-3</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="validitiy-of-single-cell-rna-seq-as-reference.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="validation-with-real-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
