<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="manuscript in preparation">
<meta name="author" content="Gregor Sturm^{1,2}">
<meta name="author" content="Tatsiana Aneichyk^1">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="detection-limit-and-false-positive-predictions.html">
<link rel="next" href="creating-publication-ready-figures.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-reference-samples"><i class="fa fa-check"></i><b>2.3</b> Immune cell reference samples</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#pbmc-samples-from-hoek2015"><i class="fa fa-check"></i><b>2.4</b> 8 PBMC samples from <span class="citation">Hoek et al. (<span>2015</span>)</span></a></li>
<li class="chapter" data-level="2.5" data-path="input-data.html"><a href="input-data.html#ovarian-cancer-ascites-samples-from-schelker2017"><i class="fa fa-check"></i><b>2.5</b> 3 ovarian cancer ascites samples from <span class="citation">Schelker et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.6" data-path="input-data.html"><a href="input-data.html#metastatic-melanoma-samples-from-epic2017"><i class="fa fa-check"></i><b>2.6</b> 4 metastatic melanoma samples from <span class="citation">Racle et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.7" data-path="input-data.html"><a href="input-data.html#data-sanity-checks"><i class="fa fa-check"></i><b>2.7</b> Data sanity checks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-to-genuine-bulk-samples"><i class="fa fa-check"></i><b>3.1</b> Compare simululated to genuine bulk samples</a><ul>
<li class="chapter" data-level="3.1.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#log-ratio"><i class="fa fa-check"></i><b>3.1.1</b> Log ratio</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#enrichment-analysis-test-for-systematic-bias"><i class="fa fa-check"></i><b>3.2</b> Enrichment analysis: Test for systematic Bias</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#immune-relevant-genes-are-consistent-between-sc-and-bulk-data."><i class="fa fa-check"></i><b>3.2.1</b> Immune-relevant genes are consistent between sc and bulk data.</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-predicted-fractions"><i class="fa fa-check"></i><b>3.3</b> Compare predicted fractions</a></li>
<li class="chapter" data-level="3.4" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlation-with-immune-reference-samples."><i class="fa fa-check"></i><b>3.4</b> Correlation with immune reference samples.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html"><i class="fa fa-check"></i><b>4</b> Simulation benchmark</a><ul>
<li class="chapter" data-level="4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>4.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>4.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.3</b> Run the deconvolution</a></li>
<li class="chapter" data-level="4.4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#results"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#correlation-plots"><i class="fa fa-check"></i><b>4.4.1</b> Correlation plots</a></li>
<li class="chapter" data-level="4.4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>4.4.2</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="4.4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#deviation-from-slope-xy-for-all-absolute-methods"><i class="fa fa-check"></i><b>4.4.3</b> Deviation from slope (x=y) for all absolute methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>5</b> Validation with real data</a><ul>
<li class="chapter" data-level="5.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between--and-within-sample-comparisons"><i class="fa fa-check"></i><b>5.1</b> Between- and within-sample comparisons</a></li>
<li class="chapter" data-level="5.2" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#within-sample-comparison"><i class="fa fa-check"></i><b>5.2</b> Within-sample comparison</a></li>
<li class="chapter" data-level="5.3" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between-sample-comparisons"><i class="fa fa-check"></i><b>5.3</b> Between-sample comparisons</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html"><i class="fa fa-check"></i><b>6</b> Detection limit and false positive predictions</a><ul>
<li class="chapter" data-level="6.1" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#predictions-with-increasing-immune-cell-content"><i class="fa fa-check"></i><b>6.1</b> Predictions with increasing immune cell content</a></li>
<li class="chapter" data-level="6.2" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#detection-limit"><i class="fa fa-check"></i><b>6.2</b> Detection Limit</a></li>
<li class="chapter" data-level="6.3" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#false-positive-predictions"><i class="fa fa-check"></i><b>6.3</b> False-positive predictions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>7</b> Spillover Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#complete-spillover-matrix"><i class="fa fa-check"></i><b>7.1</b> Complete Spillover Matrix</a></li>
<li class="chapter" data-level="7.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-signal-to-noise-ratio"><i class="fa fa-check"></i><b>7.2</b> Summary figure: Signal to noise ratio</a></li>
<li class="chapter" data-level="7.3" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-migration-charts"><i class="fa fa-check"></i><b>7.3</b> Summary figure: migration charts</a></li>
<li class="chapter" data-level="7.4" data-path="spillover-analysis.html"><a href="spillover-analysis.html#investigate-marker-genes"><i class="fa fa-check"></i><b>7.4</b> Investigate marker genes</a></li>
<li class="chapter" data-level="7.5" data-path="spillover-analysis.html"><a href="spillover-analysis.html#check-expression-of-markers-in-detection-limit-simulation-dataset."><i class="fa fa-check"></i><b>7.5</b> Check expression of markers in “detection limit” simulation dataset.</a><ul>
<li class="chapter" data-level="7.5.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#derive-marker-genes-from-quantiseqepic-signature-matrices."><i class="fa fa-check"></i><b>7.5.1</b> Derive marker genes from quanTIseq/EPIC signature matrices.</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="spillover-analysis.html"><a href="spillover-analysis.html#deconvolution-results-before-and-after"><i class="fa fa-check"></i><b>7.6</b> Deconvolution results before and after</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html"><i class="fa fa-check"></i><b>8</b> Creating publication ready figures</a><ul>
<li class="chapter" data-level="8.1" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#benchmark-results-detection-limit-false-positives-and-absolute-deviation"><i class="fa fa-check"></i><b>8.1</b> Benchmark results, detection limit, false positives and absolute deviation</a></li>
<li class="chapter" data-level="8.2" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#migration-charts-for-spillover-analysis"><i class="fa fa-check"></i><b>8.2</b> Migration charts for Spillover analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="spillover-analysis" class="section level1">
<h1><span class="header-section-number">7</span> Spillover Analysis</h1>
<p>In this chapter, we investigate which other cell types a method predicts,
if there is actually only a certain cell type present. In FACS, this phenomenon is
known as “spillover”.</p>
<p>To this end, we use three datasets</p>
<ul>
<li><em>immune reference</em>: bulk RNA seq profiles from sorted immune cells (=quanTIseq training data)</li>
<li><em>artificial bulk</em>: simulated bulk RNA seq profiles from single cells (e.g. only T cells)</li>
<li><em>artificial bulk with background</em>: simulated bulk RNA seq profiles from single cells with ~80% other cells (cancer, fibroblasts, …)</li>
</ul>
<p>We test the following cell types:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1">show_cell_types =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;B cell&quot;</span>, <span class="st">&quot;Dendritic cell&quot;</span>, <span class="st">&quot;Macrophage/Monocyte&quot;</span>,</a>
<a class="sourceLine" id="cb33-2" data-line-number="2">                    <span class="st">&quot;NK cell&quot;</span>, <span class="st">&quot;T cell CD4+&quot;</span>, <span class="st">&quot;T cell CD8+&quot;</span>)</a></code></pre></div>
<div id="complete-spillover-matrix" class="section level2">
<h2><span class="header-section-number">7.1</span> Complete Spillover Matrix</h2>
<div class="figure"><span id="fig:unnamed-chunk-56"></span>
<img src="_main_files/figure-html/unnamed-chunk-56-1.png" alt="This figure shows the spillover for all methods, cell types and datasets" width="2304" />
<p class="caption">
Figure 7.1: This figure shows the spillover for all methods, cell types and datasets
</p>
</div>
</div>
<div id="summary-figure-signal-to-noise-ratio" class="section level2">
<h2><span class="header-section-number">7.2</span> Summary figure: Signal to noise ratio</h2>
<p>Noise is defined as the sum of all predictions of other cell types than the one that is actually present.
Signal is defined as the predicted fraction of the cell type that is actually present.
The signal ratio is defined as <span class="math inline">\(\frac{\text{signal}}{\text{signal+noise}}\)</span>. Higher values
indicate less noise (i.e. predictions of cell types that are not there).</p>
<div class="figure"><span id="fig:unnamed-chunk-58"></span>
<img src="_main_files/figure-html/unnamed-chunk-58-1.png" alt="Signal ratios for each cell type and dataset. " width="960" />
<p class="caption">
Figure 7.2: Signal ratios for each cell type and dataset.
</p>
</div>
</div>
<div id="summary-figure-migration-charts" class="section level2">
<h2><span class="header-section-number">7.3</span> Summary figure: migration charts</h2>
<p>Download this figure as high quality <a href="figures/spillover_migration_all.pdf">pdf</a></p>
<div class="figure"><span id="fig:unnamed-chunk-60"></span>
<img src="_main_files/figure-html/unnamed-chunk-60-1.png" alt="migration charts for all three dataset. The value in the middle of the chart indicates the 'noise ratio', i.e. the sum of all false positive predictions." width="960" />
<p class="caption">
Figure 7.3: migration charts for all three dataset. The value in the middle of the chart indicates the ‘noise ratio’, i.e. the sum of all false positive predictions.
</p>
</div>

</div>
<div id="investigate-marker-genes" class="section level2">
<h2><span class="header-section-number">7.4</span> Investigate marker genes</h2>
<p>We asked which marker genes drive the spillover from Dendritic cells to B cells.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1">mat =<span class="st"> </span><span class="kw">cbind</span>(<span class="kw">pData</span>(single_cell_schelker<span class="op">$</span>eset), <span class="kw">t</span>(<span class="kw">exprs</span>(single_cell_schelker<span class="op">$</span>eset))) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>() </a>
<a class="sourceLine" id="cb34-2" data-line-number="2">mat =<span class="st"> </span>mat <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(source <span class="op">!=</span><span class="st"> &quot;pbmc&quot;</span>)</a></code></pre></div>
First, we suspected that the effect might be due to misannotations in the single cell dataset.
Here, we show the B cells and DCs in the t-SNE plot of the single cell dataset.
<div class="figure"><span id="fig:unnamed-chunk-62"></span>
<img src="_main_files/figure-html/unnamed-chunk-62-1.png" alt="B cell and DC clusters in the single cell dataset. " width="672" />
<p class="caption">
Figure 7.4: B cell and DC clusters in the single cell dataset.
</p>
</div>
<p>We observe that both clusters are distinct and non-overlapping.</p>
<p>Next, we look for the expression of typical marker genes of DCs and B cells.</p>
<pre><code>## [[1]]</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-63-1.png" width="480" /></p>
<pre><code>## 
## [[2]]</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-63-2.png" width="480" /></p>
<pre><code>## 
## [[3]]</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-63-3.png" width="480" /></p>
<pre><code>## 
## [[4]]</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-63-4.png" width="480" /></p>
<pre><code>## 
## [[5]]</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-63-5.png" width="480" /></p>
<pre><code>## 
## [[6]]</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-63-6.png" width="480" /></p>
<pre><code>## 
## [[7]]</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-63-7.png" width="480" /></p>
<p><strong>Conclusion</strong>: The B cell and DC cluster separate well, the marker genes do not overlap.
The DC cluster are plasmacytoid DCs, mDCs are somewhere hidden in the Macrophage/Monocyte cluster
and cannot be distinguished.</p>
</div>
<div id="check-expression-of-markers-in-detection-limit-simulation-dataset." class="section level2">
<h2><span class="header-section-number">7.5</span> Check expression of markers in “detection limit” simulation dataset.</h2>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">markers =<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb42-2" data-line-number="2">  <span class="st">&quot;B cell&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;MS4A1&quot;</span>, <span class="co">#=CD20</span></a>
<a class="sourceLine" id="cb42-3" data-line-number="3">               <span class="st">&quot;CD19&quot;</span>, <span class="st">&quot;CD22&quot;</span></a>
<a class="sourceLine" id="cb42-4" data-line-number="4">  ), </a>
<a class="sourceLine" id="cb42-5" data-line-number="5">  <span class="st">&quot;myleoid&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ITGAM&quot;</span>, <span class="st">&quot;ITGAX&quot;</span>), <span class="co">#=CD11B,CD11C</span></a>
<a class="sourceLine" id="cb42-6" data-line-number="6">  <span class="st">&quot;MHC class II&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;HLA-DRB1&quot;</span>, <span class="st">&quot;HLA-DRA&quot;</span>),</a>
<a class="sourceLine" id="cb42-7" data-line-number="7">  <span class="st">&quot;plasmacytoid DC&quot;</span> =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;CLEC4C&quot;</span>, <span class="st">&quot;IL3RA&quot;</span>))</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-68"></span>
<img src="_main_files/figure-html/unnamed-chunk-68-1.png" alt="Correlation of marker genes with increase of the amount of a certain cell type. Along x-axis: cell type used for simulation. Along y-axis: groups of marker genes. " width="1152" />
<p class="caption">
Figure 7.5: Correlation of marker genes with increase of the amount of a certain cell type. Along x-axis: cell type used for simulation. Along y-axis: groups of marker genes.
</p>
</div>
<div class="figure"><span id="fig:unnamed-chunk-69"></span>
<img src="_main_files/figure-html/unnamed-chunk-69-1.png" alt="Even though the number of dendritic cells does correlate with MS4A1, a B cell marker, the absolute expression is really low. This effect can be driven by few misclassified cells but cannot explain the spillover effects we observe. " width="288" />
<p class="caption">
Figure 7.6: Even though the number of dendritic cells does correlate with MS4A1, a B cell marker, the absolute expression is really low. This effect can be driven by few misclassified cells but cannot explain the spillover effects we observe.
</p>
</div>
<div id="derive-marker-genes-from-quantiseqepic-signature-matrices." class="section level3">
<h3><span class="header-section-number">7.5.1</span> Derive marker genes from quanTIseq/EPIC signature matrices.</h3>
<p>We use gini-index (Zhang et al. (2017), BMC genomics) to filter for highly discriminating genes
within the matrix. We select all genes with a gini index &gt; 0.7. We assign all selected genes as marker gene
to the cell type in which it is expressed the most.
In the end, we receive a list of signature genes for each cell type.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1">til10 =<span class="st"> </span><span class="kw">read_tsv</span>(<span class="st">&quot;../immunedeconv/inst/extdata/quantiseq/TIL10_signature.txt&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb43-2" data-line-number="2"><span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;ID&quot;</span>)</a></code></pre></div>
<p>The same procedure is applied to EPIC and CIBERSORT.</p>
<p>See figures:
* <a href="figures/marker_gene_expression_epic.pdf">EPIC</a>
* <a href="figures/marker_gene_expression_quantiseq.pdf">quanTIseq</a>
* <a href="figures/marker_gene_expression_cibersort.pdf">CIBERSORT</a></p>
<p>The figures show the expression of marker genes across cell types. Along x-axis: cell types used for simulation. Along y-axis: signature genes for the cell types derived from the signature matrices of the deconvolution methods.</p>
</div>
</div>
<div id="deconvolution-results-before-and-after" class="section level2">
<h2><span class="header-section-number">7.6</span> Deconvolution results before and after</h2>
<div class="figure"><span id="fig:unnamed-chunk-75"></span>
<img src="_main_files/figure-html/unnamed-chunk-75-1.png" alt="Predictions on 10 simulated DC samples before and after removal of the five genes. " width="1152" />
<p class="caption">
Figure 7.7: Predictions on 10 simulated DC samples before and after removal of the five genes.
</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="detection-limit-and-false-positive-predictions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="creating-publication-ready-figures.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
