<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>A quantitative comparison of cell type deconvolution methods for immuno-oncology</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="A quantitative comparison of cell type deconvolution methods for immuno-oncology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A quantitative comparison of cell type deconvolution methods for immuno-oncology" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm (1,2,*), Francesca Finotello (3), Florent Petitprez (4,5), Jitao David Zhang (6), Jan Baumbach (1), Wolf H. Fridman (4), Markus List (7), Tatsiana Aneichyk (2)">


<meta name="date" content="2018-10-22">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="simulation-benchmark.html">
<link rel="next" href="detection-limit-and-false-positive-predictions.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#authors-affiliations"><i class="fa fa-check"></i><b>1.1</b> Author’s affiliations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-reference-samples"><i class="fa fa-check"></i><b>2.3</b> Immune cell reference samples</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#pbmc-samples-from-hoek2015"><i class="fa fa-check"></i><b>2.4</b> 8 PBMC samples from <span class="citation">Hoek et al. (<span>2015</span>)</span></a></li>
<li class="chapter" data-level="2.5" data-path="input-data.html"><a href="input-data.html#ovarian-cancer-ascites-samples-from-schelker2017"><i class="fa fa-check"></i><b>2.5</b> 3 ovarian cancer ascites samples from <span class="citation">Schelker et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.6" data-path="input-data.html"><a href="input-data.html#metastatic-melanoma-samples-from-epic2017"><i class="fa fa-check"></i><b>2.6</b> 4 metastatic melanoma samples from <span class="citation">Racle et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.7" data-path="input-data.html"><a href="input-data.html#data-sanity-checks"><i class="fa fa-check"></i><b>2.7</b> Data sanity checks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-to-genuine-bulk-samples"><i class="fa fa-check"></i><b>3.1</b> Compare simululated to genuine bulk samples</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#enrichment-analysis-test-for-systematic-bias"><i class="fa fa-check"></i><b>3.2</b> Enrichment analysis: Test for systematic Bias</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#immune-relevant-genes-are-consistent-between-sc-and-bulk-data."><i class="fa fa-check"></i><b>3.2.1</b> Immune-relevant genes are consistent between sc and bulk data.</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-predicted-fractions"><i class="fa fa-check"></i><b>3.3</b> Compare predicted fractions</a></li>
<li class="chapter" data-level="3.4" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlation-with-immune-reference-samples."><i class="fa fa-check"></i><b>3.4</b> Correlation with immune reference samples.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html"><i class="fa fa-check"></i><b>4</b> Simulation benchmark</a><ul>
<li class="chapter" data-level="4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>4.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>4.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.3</b> Run the deconvolution</a></li>
<li class="chapter" data-level="4.4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#results"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#correlation-plots"><i class="fa fa-check"></i><b>4.4.1</b> Correlation plots</a></li>
<li class="chapter" data-level="4.4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>4.4.2</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="4.4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#absolute-deviation"><i class="fa fa-check"></i><b>4.4.3</b> Absolute deviation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>5</b> Validation with real data</a><ul>
<li class="chapter" data-level="5.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between--and-within-sample-comparisons"><i class="fa fa-check"></i><b>5.1</b> Between- and within-sample comparisons</a><ul>
<li class="chapter" data-level="5.1.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#absolute-scores"><i class="fa fa-check"></i><b>5.1.1</b> Absolute scores</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#within-sample-comparison"><i class="fa fa-check"></i><b>5.2</b> Within-sample comparison</a></li>
<li class="chapter" data-level="5.3" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between-sample-comparisons"><i class="fa fa-check"></i><b>5.3</b> Between-sample comparisons</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html"><i class="fa fa-check"></i><b>6</b> Detection limit and false positive predictions</a><ul>
<li class="chapter" data-level="6.1" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#predictions-with-increasing-immune-cell-content"><i class="fa fa-check"></i><b>6.1</b> Predictions with increasing immune cell content</a></li>
<li class="chapter" data-level="6.2" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#detection-limit-and-false-positives"><i class="fa fa-check"></i><b>6.2</b> Detection Limit and False positives</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>7</b> Spillover Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#complete-spillover-matrix"><i class="fa fa-check"></i><b>7.1</b> Complete Spillover Matrix</a></li>
<li class="chapter" data-level="7.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-signal-to-noise-ratio"><i class="fa fa-check"></i><b>7.2</b> Summary figure: Signal to noise ratio</a></li>
<li class="chapter" data-level="7.3" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-migration-charts"><i class="fa fa-check"></i><b>7.3</b> Summary figure: migration charts</a></li>
<li class="chapter" data-level="7.4" data-path="spillover-analysis.html"><a href="spillover-analysis.html#investigate-marker-genes"><i class="fa fa-check"></i><b>7.4</b> Investigate marker genes</a></li>
<li class="chapter" data-level="7.5" data-path="spillover-analysis.html"><a href="spillover-analysis.html#check-expression-of-markers-in-detection-limit-simulation-dataset."><i class="fa fa-check"></i><b>7.5</b> Check expression of markers in “detection limit” simulation dataset.</a><ul>
<li class="chapter" data-level="7.5.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#expression-of-marker-genes-in-the-cell-populations"><i class="fa fa-check"></i><b>7.5.1</b> Expression of marker genes in the cell populations</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="spillover-analysis.html"><a href="spillover-analysis.html#deconvolution-results-before-and-after"><i class="fa fa-check"></i><b>7.6</b> Deconvolution results before and after</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html"><i class="fa fa-check"></i><b>8</b> Creating publication ready figures</a><ul>
<li class="chapter" data-level="8.1" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#predictions-on-simulated-vs.genuine-bulk"><i class="fa fa-check"></i><b>8.1</b> Predictions on simulated vs. genuine bulk</a></li>
<li class="chapter" data-level="8.2" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#benchmark-results"><i class="fa fa-check"></i><b>8.2</b> Benchmark results</a></li>
<li class="chapter" data-level="8.3" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#detection-limit-false-positive-figure"><i class="fa fa-check"></i><b>8.3</b> detection limit / false positive figure</a></li>
<li class="chapter" data-level="8.4" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#migration-charts-for-spillover-analysis"><i class="fa fa-check"></i><b>8.4</b> Migration charts for Spillover analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A quantitative comparison of cell type deconvolution methods for immuno-oncology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="validation-with-real-data" class="section level1">
<h1><span class="header-section-number">5</span> Validation with real data</h1>
<p>We use datasets which estimate the immune cell proportions using flow cytometry as an additional validation
for our simulation benchmark. We use the following three validation datasets (see section <a href="input-data.html#input-data">2</a>):</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">datasets =<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">  <span class="dt">racle=</span>racle,</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">  <span class="dt">hoek=</span>hoek,</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">  <span class="dt">schelker_ovarian=</span>schelker_ovarian</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">)</a></code></pre></div>
<p>We use the following cell types, which are available in (some of) the datasets</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1">use_cell_types =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;T cell&quot;</span>, <span class="st">&quot;T cell CD8+&quot;</span>, <span class="st">&quot;T cell CD4+&quot;</span>,</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">                   <span class="st">&quot;Monocyte&quot;</span>, <span class="st">&quot;B cell&quot;</span>,</a>
<a class="sourceLine" id="cb15-3" data-line-number="3">                   <span class="st">&quot;Dendritic cell&quot;</span>, <span class="st">&quot;NK cell&quot;</span>)</a></code></pre></div>
<p>Here, we combine the predictions with the ‘gold standard’ reference data.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">all_results_ref =<span class="st"> </span><span class="kw">inner_join</span>(all_results, all_refs,</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">                             <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;sample&quot;</span> =<span class="st"> &quot;sample&quot;</span>,</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">                                    <span class="st">&quot;dataset&quot;</span> =<span class="st"> &quot;dataset&quot;</span>,</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">                                    <span class="st">&quot;cell_type&quot;</span> =<span class="st"> &quot;cell_type&quot;</span>))</a>
<a class="sourceLine" id="cb16-5" data-line-number="5"></a>
<a class="sourceLine" id="cb16-6" data-line-number="6">res_validation<span class="op">$</span>all_results =<span class="st"> </span>all_results_ref</a></code></pre></div>
<p>The scores of the methods have different properties and not all of them are directly comparable.
We distinguish between three types of comparisons:</p>
<ul>
<li>absolute scores: allow to compare within <em>and</em> between samples.</li>
<li>scores relative to the total immune cell content: allow to compare within a sample</li>
<li>scores in arbitrary units, that only allow to compare between samples.</li>
</ul>
<div id="between--and-within-sample-comparisons" class="section level2">
<h2><span class="header-section-number">5.1</span> Between- and within-sample comparisons</h2>
<p>Only works for methods providing an absolute score (cibersort_abs, quantiseq, epic).
All other methods are included for reference.</p>
<div class="figure"><span id="fig:unnamed-chunk-40"></span>
<img src="_main_files/figure-html/unnamed-chunk-40-1.png" alt="Comparison of absolute predictions for three validation dataset. " width="960" />
<p class="caption">
Figure 5.1: Comparison of absolute predictions for three validation dataset.
</p>
</div>
<div id="absolute-scores" class="section level3">
<h3><span class="header-section-number">5.1.1</span> Absolute scores</h3>
<p>Both EPIC and quanTIseq provide scores that can be interpreted as absolute cell fractions. Both methods account for the different mRNA contents of the different cell types by applying a scaling factor. As the expression values of the single cells we use for the simulation have already been normalized, these scaling factors are not appropriate. While this has no effect on the correlations discussed in the previous section, it is crucial for deriving correct absolute values. For this reason, assessing the absolute quantification performance is fair on the validation datasets only, which provide geniune bulk RNA-seq data.</p>
<p>We use two measures to assess the absolute deviation: (1) the slope of a linear model fitted to the estimated versus known fractions and (2) the root mean square error (RMSE). The slope quickly allows to evaluate whether a method over- or under-predicts a certain cell type while RMSE gives insight about the true deviations.</p>
<div class="figure"><span id="fig:unnamed-chunk-41"></span>
<img src="_main_files/figure-html/unnamed-chunk-41-1.png" alt="Comparison of absolute methods. Values indicate the slope of a linear model fitted to predictions vs. true fractions. Values &lt; 1 indicate an under-prediction of the cell type, values &gt; 1 an over-prediction respectively. The error bars have been computed using `confint` on the result of `lm`. " width="864" />
<p class="caption">
Figure 5.2: Comparison of absolute methods. Values indicate the slope of a linear model fitted to predictions vs. true fractions. Values &lt; 1 indicate an under-prediction of the cell type, values &gt; 1 an over-prediction respectively. The error bars have been computed using <code>confint</code> on the result of <code>lm</code>.
</p>
</div>
<div class="figure"><span id="fig:abs-rmse-val"></span>
<img src="_main_files/figure-html/abs-rmse-val-1.png" alt="Comparison of absolute methods. The values show the RMSE" width="864" />
<p class="caption">
Figure 5.3: Comparison of absolute methods. The values show the RMSE
</p>
</div>
</div>
</div>
<div id="within-sample-comparison" class="section level2">
<h2><span class="header-section-number">5.2</span> Within-sample comparison</h2>
<p>only works with methods that provide an absolute score, or a score that is relative to total immune
cell content (cibersort, cibersort_abs, quantiseq, epic).</p>
<div class="figure"><span id="fig:unnamed-chunk-42"></span>
<img src="_main_files/figure-html/unnamed-chunk-42-1.png" alt="Comparison of predictions within each individual sample" width="768" />
<p class="caption">
Figure 5.4: Comparison of predictions within each individual sample
</p>
</div>
Compute the average over all samples:
<div class="figure"><span id="fig:unnamed-chunk-43"></span>
<img src="_main_files/figure-html/unnamed-chunk-43-1.png" alt="Correlations of within-sample comparisons. The last column shows the mean over all samples. " width="768" />
<p class="caption">
Figure 5.5: Correlations of within-sample comparisons. The last column shows the mean over all samples.
</p>
</div>
</div>
<div id="between-sample-comparisons" class="section level2">
<h2><span class="header-section-number">5.3</span> Between-sample comparisons</h2>
<p>For this, we need to look at every cell type independently.
Works for all methods except CIBERSORT.</p>
<div class="figure"><span id="fig:unnamed-chunk-44"></span>
<img src="_main_files/figure-html/unnamed-chunk-44-1.png" alt="Correlations of known vs. predicted fractions for each cell type independenctly. The 'Tcell' column corresponds to the amount of profiled total T cells or the sum of CD4+ and CD8+ T cells respectively. " width="1152" />
<p class="caption">
Figure 5.6: Correlations of known vs. predicted fractions for each cell type independenctly. The ‘Tcell’ column corresponds to the amount of profiled total T cells or the sum of CD4+ and CD8+ T cells respectively.
</p>
</div>
<p>Compute the average over all cell types.
We only use cell types with at least 5 samples to obtain reasonable
correlation estimates. We also exclude the T-cell supercategory to
avoid redundancies.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">use_cell_types2 =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;B cell&quot;</span>, <span class="st">&quot;Dendritic cell&quot;</span>, <span class="st">&quot;Macrophage/Monocyte&quot;</span>, <span class="st">&quot;NK cell&quot;</span>, <span class="st">&quot;T cell CD4+&quot;</span>, <span class="st">&quot;T cell CD8+&quot;</span>)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-46"></span>
<img src="_main_files/figure-html/unnamed-chunk-46-1.png" alt="Performance on validation datasets by cell type. The last column shows the mean over all cell types. " width="480" />
<p class="caption">
Figure 5.7: Performance on validation datasets by cell type. The last column shows the mean over all cell types.
</p>
</div>
<p>Note that, although the correlation for CD8+ T cells looks bad, it does not necessarily mean the predictions are bad. There is just little variance between the samples and CD8+ T cell abundance is generally low, which is correlctly predicted by the methods.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="simulation-benchmark.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="detection-limit-and-false-positive-predictions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
