<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>A quantitative comparison of cell type deconvolution methods for immuno-oncology</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="A quantitative comparison of cell type deconvolution methods for immuno-oncology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A quantitative comparison of cell type deconvolution methods for immuno-oncology" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm (1,2,*), Francesca Finotello (3), Florent Petitprez (4,5), Jitao David Zhang (6), Jan Baumbach (1), Wolf H. Fridman (4), Markus List (7), Tatsiana Aneichyk (2)">


<meta name="date" content="2018-10-30">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="input-data.html">
<link rel="next" href="simulation-benchmark.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#authors-affiliations"><i class="fa fa-check"></i><b>1.1</b> Author’s affiliations</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-reference-samples"><i class="fa fa-check"></i><b>2.3</b> Immune cell reference samples</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#pbmc-samples-from-hoek2015"><i class="fa fa-check"></i><b>2.4</b> 8 PBMC samples from <span class="citation">Hoek et al. (<span>2015</span>)</span></a></li>
<li class="chapter" data-level="2.5" data-path="input-data.html"><a href="input-data.html#ovarian-cancer-ascites-samples-from-schelker2017"><i class="fa fa-check"></i><b>2.5</b> 3 ovarian cancer ascites samples from <span class="citation">Schelker et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.6" data-path="input-data.html"><a href="input-data.html#metastatic-melanoma-samples-from-epic2017"><i class="fa fa-check"></i><b>2.6</b> 4 metastatic melanoma samples from <span class="citation">Racle et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.7" data-path="input-data.html"><a href="input-data.html#data-sanity-checks"><i class="fa fa-check"></i><b>2.7</b> Data sanity checks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-to-genuine-bulk-samples"><i class="fa fa-check"></i><b>3.1</b> Compare simululated to genuine bulk samples</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#enrichment-analysis-test-for-systematic-bias"><i class="fa fa-check"></i><b>3.2</b> Enrichment analysis: Test for systematic Bias</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#immune-relevant-genes-are-consistent-between-sc-and-bulk-data."><i class="fa fa-check"></i><b>3.2.1</b> Immune-relevant genes are consistent between sc and bulk data.</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-predicted-fractions"><i class="fa fa-check"></i><b>3.3</b> Compare predicted fractions</a></li>
<li class="chapter" data-level="3.4" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlation-with-immune-reference-samples."><i class="fa fa-check"></i><b>3.4</b> Correlation with immune reference samples.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html"><i class="fa fa-check"></i><b>4</b> Simulation benchmark</a><ul>
<li class="chapter" data-level="4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>4.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>4.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.3</b> Run the deconvolution</a></li>
<li class="chapter" data-level="4.4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#results"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#correlation-plots"><i class="fa fa-check"></i><b>4.4.1</b> Correlation plots</a></li>
<li class="chapter" data-level="4.4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>4.4.2</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="4.4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#absolute-deviation"><i class="fa fa-check"></i><b>4.4.3</b> Absolute deviation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>5</b> Validation with real data</a><ul>
<li class="chapter" data-level="5.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between--and-within-sample-comparisons"><i class="fa fa-check"></i><b>5.1</b> Between- and within-sample comparisons</a><ul>
<li class="chapter" data-level="5.1.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#absolute-scores"><i class="fa fa-check"></i><b>5.1.1</b> Absolute scores</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#within-sample-comparison"><i class="fa fa-check"></i><b>5.2</b> Within-sample comparison</a></li>
<li class="chapter" data-level="5.3" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between-sample-comparisons"><i class="fa fa-check"></i><b>5.3</b> Between-sample comparisons</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html"><i class="fa fa-check"></i><b>6</b> Detection limit and false positive predictions</a><ul>
<li class="chapter" data-level="6.1" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#predictions-with-increasing-immune-cell-content"><i class="fa fa-check"></i><b>6.1</b> Predictions with increasing immune cell content</a></li>
<li class="chapter" data-level="6.2" data-path="detection-limit-and-false-positive-predictions.html"><a href="detection-limit-and-false-positive-predictions.html#detection-limit-and-false-positives"><i class="fa fa-check"></i><b>6.2</b> Detection Limit and False positives</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>7</b> Spillover Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#complete-spillover-matrix"><i class="fa fa-check"></i><b>7.1</b> Complete Spillover Matrix</a></li>
<li class="chapter" data-level="7.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-signal-to-noise-ratio"><i class="fa fa-check"></i><b>7.2</b> Summary figure: Signal to noise ratio</a></li>
<li class="chapter" data-level="7.3" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-migration-charts"><i class="fa fa-check"></i><b>7.3</b> Summary figure: migration charts</a></li>
<li class="chapter" data-level="7.4" data-path="spillover-analysis.html"><a href="spillover-analysis.html#investigate-marker-genes"><i class="fa fa-check"></i><b>7.4</b> Investigate marker genes</a></li>
<li class="chapter" data-level="7.5" data-path="spillover-analysis.html"><a href="spillover-analysis.html#check-expression-of-markers-in-detection-limit-simulation-dataset."><i class="fa fa-check"></i><b>7.5</b> Check expression of markers in “detection limit” simulation dataset.</a><ul>
<li class="chapter" data-level="7.5.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#expression-of-marker-genes-in-the-cell-populations"><i class="fa fa-check"></i><b>7.5.1</b> Expression of marker genes in the cell populations</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="spillover-analysis.html"><a href="spillover-analysis.html#deconvolution-results-before-and-after"><i class="fa fa-check"></i><b>7.6</b> Deconvolution results before and after</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html"><i class="fa fa-check"></i><b>8</b> Creating publication ready figures</a><ul>
<li class="chapter" data-level="8.1" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#predictions-on-simulated-vs.genuine-bulk"><i class="fa fa-check"></i><b>8.1</b> Predictions on simulated vs. genuine bulk</a></li>
<li class="chapter" data-level="8.2" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#benchmark-results"><i class="fa fa-check"></i><b>8.2</b> Benchmark results</a></li>
<li class="chapter" data-level="8.3" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#detection-limit-false-positive-figure"><i class="fa fa-check"></i><b>8.3</b> detection limit / false positive figure</a></li>
<li class="chapter" data-level="8.4" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#migration-charts-for-spillover-analysis"><i class="fa fa-check"></i><b>8.4</b> Migration charts for Spillover analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A quantitative comparison of cell type deconvolution methods for immuno-oncology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="validitiy-of-single-cell-rna-seq-as-reference" class="section level1">
<h1><span class="header-section-number">3</span> Validitiy of single cell RNA-seq as reference</h1>
<p>Since we rely on simulated bulk RNA-seq samples generated from scRNA-seq data, we
first investigate if this approach is valid and will give reasonable
estimates of the methods’s performances.</p>
<p>We conduct the following analyses:</p>
<ul>
<li>Compare simulated bulk samples with a corresponding genuine bulk sample.</li>
<li>Check if immune-related genes are subject to a systematic bias.</li>
<li>Compare the predicted fractions of all methods on bulk vs. simulated samples.</li>
<li>Compare simulated bulk samples with a reference immune cell sample of the same cell type.</li>
</ul>
<div id="compare-simululated-to-genuine-bulk-samples" class="section level2">
<h2><span class="header-section-number">3.1</span> Compare simululated to genuine bulk samples</h2>
<p> provide three samples for which paired single cell and bulk
RNA-seq has been performed. We generate simulated bulk RNA-seq samples by
taking the average of the single cells and compare the simulated to the genuine
samples.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">bulk_mean =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">colnames</span>(schelker_ovarian<span class="op">$</span>expr_mat), <span class="cf">function</span>(donor) {</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">  ind =<span class="st"> </span><span class="kw">pData</span>(single_cell_schelker<span class="op">$</span>eset)<span class="op">$</span>donor <span class="op">==</span><span class="st"> </span>donor</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  <span class="kw">apply</span>(<span class="kw">exprs</span>(single_cell_schelker<span class="op">$</span>eset)[,ind], <span class="dv">1</span>, mean)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">})</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">bulk_mean =<span class="st"> </span><span class="kw">apply</span>(bulk_mean, <span class="dv">2</span>, scale_to_million)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-15"></span>
<img src="_main_files/figure-html/unnamed-chunk-15-1.png" alt="Correlation of simulated bulk samples with corresponding genuine bulk RNA-seq samples. Genes used in the signatures of quanTIseq, EPIC, CIBERSORT and MCP-counter are shown in blue. " width="1152" />
<p class="caption">
Figure 3.1: Correlation of simulated bulk samples with corresponding genuine bulk RNA-seq samples. Genes used in the signatures of quanTIseq, EPIC, CIBERSORT and MCP-counter are shown in blue.
</p>
</div>
</div>
<div id="enrichment-analysis-test-for-systematic-bias" class="section level2">
<h2><span class="header-section-number">3.2</span> Enrichment analysis: Test for systematic Bias</h2>
<p>Next, we calculate the sc-bulk log ratio, by <span class="math display">\[log(sc+1)/log(bulk+1)\]</span>,</p>
<p>We observe that the sc-bulk log ratio showes asymmetry: more genes show increased sc-aggregate expression rather than decreased sc-aggregate expression.</p>
<div class="figure"><span id="fig:scBulkLogRatio"></span>
<img src="_main_files/figure-html/scBulkLogRatio-1.png" alt="log-ratio of single cell agregates to geniune bulk. We observe that the distribution is skewed in direction of the single cells, i.e. more genes have increased gene expression in sc-aggregates rather than decreased." width="672" />
<p class="caption">
Figure 3.2: log-ratio of single cell agregates to geniune bulk. We observe that the distribution is skewed in direction of the single cells, i.e. more genes have increased gene expression in sc-aggregates rather than decreased.
</p>
</div>
<p>What are these genes that seem to be higher expressed in single-cell aggregates than bulk? We were wondering if there is a systematic bias towards a certain group of genes that tend to be over/underrepresented in single-cell vs. bulk RNA seq.
To this end, we take the geometric mean of the log ratio and run a gene set enrichment test with BioQC, using gene ontology (BP and CC terms) as the knowledgebase.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">geomMean &lt;-<span class="st"> </span><span class="cf">function</span>(x) <span class="dv">10</span><span class="op">^</span>(<span class="kw">mean</span>(<span class="kw">log10</span>(x)))</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">avgData &lt;-<span class="st"> </span>expr_all <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(gene_symbol) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summarise</span>(<span class="dt">MeanBulk=</span><span class="kw">geomMean</span>(bulk),</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                                                            <span class="dt">MeanSc=</span><span class="kw">geomMean</span>(single_cell_aggregate),</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                                                            <span class="dt">MeanScBulkLogRatio=</span><span class="kw">geomMean</span>(ScBulkLogRatio))</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">## translate gene symbol into EntrezIds</span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6">gsIDs &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db, avgData<span class="op">$</span>gene_symbol, <span class="kw">c</span>(<span class="st">&quot;ENTREZID&quot;</span>), <span class="st">&quot;SYMBOL&quot;</span>)</a>
<a class="sourceLine" id="cb5-7" data-line-number="7">annoAvgData &lt;-<span class="st"> </span>avgData <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">inner_join</span>(gsIDs, <span class="dt">by=</span><span class="kw">c</span>(<span class="st">&quot;gene_symbol&quot;</span>=<span class="st">&quot;SYMBOL&quot;</span>)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="st">  </span><span class="kw">rename</span>(<span class="dt">GeneID=</span>ENTREZID) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(GeneID)) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">GeneSymbol =</span> gene_symbol) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(GeneID, GeneSymbol, MeanBulk, MeanSc, MeanScBulkLogRatio)</a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">## build up a list of GO gene-sets</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12">go &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(org.Hs.eg.db, annoAvgData<span class="op">$</span>GeneID, <span class="kw">c</span>(<span class="st">&quot;GO&quot;</span>), <span class="st">&quot;ENTREZID&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="st">  </span><span class="kw">filter</span>(ONTOLOGY <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;BP&quot;</span>,<span class="st">&quot;CC&quot;</span>),</a>
<a class="sourceLine" id="cb5-14" data-line-number="14">         EVIDENCE <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;EXP&quot;</span>, <span class="st">&quot;IDA&quot;</span>, <span class="st">&quot;IPI&quot;</span>, <span class="st">&quot;IMP&quot;</span>, <span class="st">&quot;IGI&quot;</span>, <span class="st">&quot;IEP&quot;</span>))</a>
<a class="sourceLine" id="cb5-15" data-line-number="15">goList &lt;-<span class="st"> </span><span class="kw">with</span>(go, <span class="kw">split</span>(ENTREZID, GO))</a>
<a class="sourceLine" id="cb5-16" data-line-number="16">goMatchList &lt;-<span class="st"> </span><span class="kw">sapply</span>(goList, <span class="cf">function</span>(x) <span class="kw">match</span>(x, annoAvgData<span class="op">$</span>GeneID))</a>
<a class="sourceLine" id="cb5-17" data-line-number="17">goMatchListLen &lt;-<span class="st"> </span><span class="kw">sapply</span>(goMatchList, length)</a>
<a class="sourceLine" id="cb5-18" data-line-number="18">goBioQClist &lt;-<span class="st"> </span>goMatchList[goMatchListLen<span class="op">&gt;=</span><span class="dv">5</span> <span class="op">&amp;</span><span class="st"> </span>goMatchListLen<span class="op">&lt;=</span><span class="dv">1000</span>]</a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">## Run two-sided Wilcoxon-Mann-Whitney test using sc-bulk log ratios</span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20">goBioQCp &lt;-<span class="st"> </span><span class="kw">wmwTest</span>(annoAvgData<span class="op">$</span>MeanScBulkLogRatio, goBioQClist,</a>
<a class="sourceLine" id="cb5-21" data-line-number="21">                      <span class="dt">valType=</span><span class="st">&quot;p.two.sided&quot;</span>)</a>
<a class="sourceLine" id="cb5-22" data-line-number="22">goBioQCq &lt;-<span class="st"> </span><span class="kw">wmwTest</span>(annoAvgData<span class="op">$</span>MeanScBulkLogRatio, goBioQClist,</a>
<a class="sourceLine" id="cb5-23" data-line-number="23">                      <span class="dt">valType=</span><span class="st">&quot;Q&quot;</span>)</a>
<a class="sourceLine" id="cb5-24" data-line-number="24">goBioQClistAnno &lt;-<span class="st"> </span>AnnotationDbi<span class="op">::</span><span class="kw">select</span>(GO.db, <span class="kw">names</span>(goBioQClist), <span class="kw">c</span>(<span class="st">&quot;ONTOLOGY&quot;</span>, <span class="st">&quot;TERM&quot;</span>), <span class="st">&quot;GOID&quot;</span>)</a>
<a class="sourceLine" id="cb5-25" data-line-number="25">goBioQCres &lt;-<span class="st"> </span><span class="kw">cbind</span>(goBioQClistAnno,</a>
<a class="sourceLine" id="cb5-26" data-line-number="26">                    <span class="dt">GeneCount=</span><span class="kw">sapply</span>(goBioQClist, length),</a>
<a class="sourceLine" id="cb5-27" data-line-number="27">                    <span class="dt">BioQC.twosided.pvalue=</span>goBioQCp,</a>
<a class="sourceLine" id="cb5-28" data-line-number="28">                    <span class="dt">BioQC.Qvalue=</span>goBioQCq,</a>
<a class="sourceLine" id="cb5-29" data-line-number="29">                    <span class="dt">FDR=</span><span class="kw">p.adjust</span>(goBioQCp, <span class="st">&quot;BH&quot;</span>),</a>
<a class="sourceLine" id="cb5-30" data-line-number="30">                    <span class="dt">Bonferroni=</span><span class="kw">p.adjust</span>(goBioQCp, <span class="st">&quot;bonferroni&quot;</span>))</a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="co">#TODO table width and round pvalues</span></a></code></pre></div>
<p>Below we display the terms that, under moderate stringency of filtering (Benjamini-Hochberg FDR&lt;0.01), shows significant enrichment in either direction.</p>
<div id="htmlwidget-309e0a1cba4a4d414e74" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-309e0a1cba4a4d414e74">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68"],["GO:0005739","GO:0005730","GO:0005813","GO:0005925","GO:0005762","GO:0043234","GO:0043231","GO:0005759","GO:0071013","GO:0000776","GO:0005783","GO:0000784","GO:0005794","GO:0007080","GO:0031965","GO:0016607","GO:0048471","GO:0007059","GO:0022625","GO:0000209","GO:0000502","GO:0032212","GO:0000785","GO:0016604","GO:0005874","GO:0070979","GO:0043065","GO:0006468","GO:0030496","GO:0016567","GO:0070936","GO:0032981","GO:0005747","GO:0022627","GO:0042645","GO:0031012","GO:0000387","GO:0006511","GO:0042542","GO:0015630","GO:0050821","GO:0000922","GO:0006974","GO:0051973","GO:0006413","GO:0001650","GO:0007051","GO:0000281","GO:0005680","GO:1904874","GO:0007005","GO:0000070","GO:0034644","GO:0043161","GO:0060348","GO:0005743","GO:0030529","GO:0007052","GO:0007568","GO:0008584","GO:0005887","GO:0008076","GO:0071805","GO:0016180","GO:0032039","GO:0052695","GO:0016579","GO:0001580"],["CC","CC","CC","CC","CC","CC","CC","CC","CC","CC","CC","CC","CC","BP","CC","CC","CC","BP","CC","BP","CC","BP","CC","CC","CC","BP","BP","BP","CC","BP","BP","BP","CC","CC","CC","CC","BP","BP","BP","CC","BP","CC","BP","BP","BP","CC","BP","BP","CC","BP","BP","BP","BP","BP","BP","CC","CC","BP","BP","BP","CC","CC","BP","BP","CC","BP","BP","BP"],["mitochondrion","nucleolus","centrosome","focal adhesion","mitochondrial large ribosomal subunit","protein complex","intracellular membrane-bounded organelle","mitochondrial matrix","catalytic step 2 spliceosome","kinetochore","endoplasmic reticulum","nuclear chromosome, telomeric region","Golgi apparatus","mitotic metaphase plate congression","nuclear membrane","nuclear speck","perinuclear region of cytoplasm","chromosome segregation","cytosolic large ribosomal subunit","protein polyubiquitination","proteasome complex","positive regulation of telomere maintenance via telomerase","chromatin","nuclear body","microtubule","protein K11-linked ubiquitination","positive regulation of apoptotic process","protein phosphorylation","midbody","protein ubiquitination","protein K48-linked ubiquitination","mitochondrial respiratory chain complex I assembly","mitochondrial respiratory chain complex I","cytosolic small ribosomal subunit","mitochondrial nucleoid","extracellular matrix","spliceosomal snRNP assembly","ubiquitin-dependent protein catabolic process","response to hydrogen peroxide","microtubule cytoskeleton","protein stabilization","spindle pole","cellular response to DNA damage stimulus","positive regulation of telomerase activity","translational initiation","fibrillar center","spindle organization","mitotic cytokinesis","anaphase-promoting complex","positive regulation of telomerase RNA localization to Cajal body","mitochondrion organization","mitotic sister chromatid segregation","cellular response to UV","proteasome-mediated ubiquitin-dependent protein catabolic process","bone development","mitochondrial inner membrane","intracellular ribonucleoprotein complex","mitotic spindle organization","aging","male gonad development","integral component of plasma membrane","voltage-gated potassium channel complex","potassium ion transmembrane transport","snRNA processing","integrator complex","cellular glucuronidation","protein deubiquitination","detection of chemical stimulus involved in sensory perception of bitter taste"],[675,572,357,335,46,430,495,50,77,70,491,90,506,35,138,258,285,36,48,59,36,32,49,241,61,25,153,197,99,127,40,40,53,37,33,265,22,36,13,100,117,53,153,35,17,118,13,38,19,15,26,10,32,56,8,51,71,20,10,24,234,48,42,12,11,12,55,28],[4.31289290539951e-50,1.48815469779605e-20,2.42335484693185e-15,3.40519842670489e-12,2.35505019264333e-11,6.00446756264733e-11,2.2715102641934e-09,2.34938911157566e-08,4.05562232552102e-08,4.17014818464245e-08,7.47804699446757e-08,1.213743392586e-07,2.40373765129501e-07,4.23290702720989e-07,1.0234975246862e-06,1.38948536497238e-06,1.42974696386979e-06,1.58432359228078e-06,2.07179223376082e-06,2.07220001032709e-06,2.62012066013313e-06,3.13662530780998e-06,4.16511097493554e-06,4.74860927194148e-06,5.04168223423748e-06,5.345390295956e-06,5.63264659316927e-06,5.94301258211935e-06,6.62825067109957e-06,7.87879169950541e-06,1.05220875550247e-05,1.15909258733967e-05,1.25710284238149e-05,1.46331747246465e-05,1.72619823445859e-05,1.91975239170231e-05,4.99195325724216e-05,5.15538581892309e-05,5.96763468154735e-05,9.74591041542503e-05,0.000105620773074035,0.000111173641991272,0.000113005018360853,0.00012287039585591,0.000166033391694623,0.000168920012512817,0.000175875173060332,0.000186299010850216,0.000194236161396105,0.000195417632419454,0.000210179781968441,0.000220409421435904,0.000222205750093119,0.000226674040570089,0.000245704187577495,0.000249671727658744,0.000267800113132071,0.000293224378732129,0.00026492654353019,0.00024750519024726,0.000142978800530671,0.000133786091996753,7.06839863085969e-05,9.97627444732266e-06,3.93745362676092e-06,3.60318587943718e-06,2.8411923037989e-07,4.58104636284924e-10],[-49.3652313258029,-19.8273519203311,-14.6155829883466,-11.4678575759356,-10.627999832418,-10.2215254969798,-8.64368529641375,-7.62904504835002,-7.39194249489245,-7.37984851226148,-7.12621180992614,-6.9158731213033,-6.6191129338458,-6.37336127039042,-5.98991320333506,-5.85714602315448,-5.84474081702364,-5.80015611063217,-5.68365379924313,-5.6835683284687,-5.58167870836598,-5.5035373577669,-5.38037342279019,-5.32343356383994,-5.29742453039026,-5.27202057910035,-5.24928749667008,-5.22599335028374,-5.17860107553684,-5.10354038130156,-4.97789808872914,-4.93588187157334,-4.90062919164437,-4.83466144176119,-4.76290933191864,-4.71675478267767,-4.30172948987548,-4.28773882742037,-4.22419777076398,-4.01117758497867,-3.97625065808821,-3.95399816684236,-3.94690226980832,-3.91055274254198,-3.7798045603313,-3.77231889507659,-3.75479546221329,-3.72978945097297,-3.71166991329101,-3.70903625271045,-3.67740906281704,-3.65676984543652,-3.65324470686499,-3.64459821380691,-3.60958744170347,-3.60263063352759,-3.57218924385641,-3.53279992520178,3.57687452667378,3.60641568936835,3.8447283506123,3.87358903218114,4.15067896591388,5.00103161192343,5.40478454806214,5.44331333312262,6.54649937041213,9.33903531288934],[9.88083764627029e-47,1.70468120632537e-17,1.85063531810695e-12,1.95032739889522e-09,1.07908399826917e-08,2.29270586433751e-08,6.50503751908385e-07,5.98050050513316e-06,8.68528135546896e-06,8.68528135546896e-06,1.42768380536043e-05,2.13898931724194e-05,3.93354497079776e-05,6.06099374958616e-05,0.000137931342885652,0.000172397383906616,0.000172397383906616,0.000181484267495763,0.000215791373802699,0.000215791373802699,0.000260986801407174,0.000299417024174694,0.000353417379391752,0.000388537994357783,0.000398292896504761,0.000408209638934507,0.000416270753062929,0.000425482557051107,0.000460161281439064,0.000530891523046085,0.000669613960793378,0.000717697599350049,0.00075790068734105,0.000859605212670902,0.000988680038786157,0.0010727201779,0.00272299164579566,0.00274674160724484,0.00310723887623295,0.00485388712211712,0.00514845087473646,0.005283561164586,0.005283561164586,0.00562992153811782,0.00716658793827527,0.00716658793827527,0.00732600039056763,0.00762162560460437,0.00771899648056843,0.00771899648056843,0.00816138780491015,0.00834546513874322,0.00834546513874322,0.00837597140235602,0.00879996812409512,0.00879996812409512,0.00915716506247127,0.00987907428934275,0.00915716506247127,0.00879996812409512,0.00629931600030321,0.00600988111303061,0.00359860028073323,0.000653018421680463,0.00034695024072728,0.000330195953991623,4.33944771200218e-05,1.49931103104109e-07],[9.88083764627029e-47,3.40936241265074e-17,5.55190595432086e-12,7.8013095955809e-09,5.39541999134586e-08,1.3756235186025e-07,5.20403001526708e-06,5.38245045461984e-05,9.29143074776865e-05,9.55380949101585e-05,0.000171322056643252,0.000278068611241452,0.000550696295911687,0.000969758999933785,0.00234483282905609,0.00318331097115171,0.0032755502942257,0.00362968534991527,0.00474647600754605,0.00474741022365937,0.006002696432365,0.00718600858019266,0.00954226924357731,0.0108790638420179,0.0115504939986381,0.0122462891680352,0.0129043933449508,0.0136154418256354,0.0151853222874891,0.0180503117835669,0.0241061025885616,0.0265548111759518,0.0288002261189599,0.0335246032941652,0.0395472015514463,0.0439815272939,0.114365649123418,0.118109889111528,0.13671851055425,0.223278807617387,0.241977191112614,0.254698813802005,0.258894497064714,0.281496076905891,0.380382500372382,0.386995748666865,0.40293002148122,0.426811033857845,0.444995045758475,0.447701795872969,0.481521880489699,0.504957984509656,0.509073373463336,0.519310226946073,0.562908293740041,0.571997928066183,0.613530059185575,0.671777051675307,0.606946711227666,0.567034390856473,0.327564432015767,0.306503936764561,0.161937012632996,0.0228556447588162,0.00902070625890927,0.00825489884979057,0.000650917156800327,1.04951772172876e-06]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>GOID<\/th>\n      <th>ONTOLOGY<\/th>\n      <th>TERM<\/th>\n      <th>GeneCount<\/th>\n      <th>BioQC.twosided.pvalue<\/th>\n      <th>BioQC.Qvalue<\/th>\n      <th>FDR<\/th>\n      <th>Bonferroni<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":"500px","columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nDTWidget.formatSignif(this, row, data, 5, 2, 3, ',', '.');\nDTWidget.formatSignif(this, row, data, 6, 2, 3, ',', '.');\nDTWidget.formatSignif(this, row, data, 7, 2, 3, ',', '.');\nDTWidget.formatSignif(this, row, data, 8, 2, 3, ',', '.');\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>Interstingly, there are many terms that are highly significantly over- or underrepresented in simulated versus genuine bulk RNA-seq samples. We currently have no explanation for that.</p>
<div id="immune-relevant-genes-are-consistent-between-sc-and-bulk-data." class="section level3">
<h3><span class="header-section-number">3.2.1</span> Immune-relevant genes are consistent between sc and bulk data.</h3>
<p>We notice that <em>among these significantly enriched gene-sets above, no gene-sets are directly involved in immune response, cytokine/interleukin/chemokine response</em>.</p>
<p>Alternatively, we show these immune-relevant gene-sets and their results by an incomprehensive keyword search, to further demonstrating that they do not show strong difference between sc-aggregated and bulk data.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">isImmune &lt;-<span class="st"> </span><span class="kw">with</span>(goBioQCres,</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                 <span class="kw">grepl</span>(<span class="st">&quot;immune</span><span class="ch">\\</span><span class="st">b&quot;</span>, TERM, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>) <span class="op">|</span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="st">                                   </span><span class="kw">grepl</span>(<span class="st">&quot;inflammation</span><span class="ch">\\</span><span class="st">b&quot;</span>, TERM, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>) <span class="op">|</span></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="st">                                  </span><span class="kw">grepl</span>(<span class="st">&quot;cytokine</span><span class="ch">\\</span><span class="st">b&quot;</span>, TERM, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>) <span class="op">|</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="st">                                     </span><span class="kw">grepl</span>(<span class="st">&quot;chemokine</span><span class="ch">\\</span><span class="st">b&quot;</span>, TERM, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>) <span class="op">|</span></a>
<a class="sourceLine" id="cb6-6" data-line-number="6"><span class="st">                                    </span><span class="kw">grepl</span>(<span class="st">&quot;interleukin</span><span class="ch">\\</span><span class="st">b&quot;</span>, TERM, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>) <span class="op">|</span></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="st">                                    </span><span class="kw">grepl</span>(<span class="st">&quot;antigen</span><span class="ch">\\</span><span class="st">b&quot;</span>, TERM, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>) <span class="op">|</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8"><span class="st">                                    </span><span class="kw">grepl</span>(<span class="st">&quot;macrophage</span><span class="ch">\\</span><span class="st">b&quot;</span>, TERM, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>) <span class="op">|</span></a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="st">                                    </span><span class="kw">grepl</span>(<span class="st">&quot;dendritic cell</span><span class="ch">\\</span><span class="st">b&quot;</span>, TERM, <span class="dt">ignore.case=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb6-10" data-line-number="10">immuneRes &lt;-<span class="st"> </span>goBioQCres <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(isImmune) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(FDR)</a>
<a class="sourceLine" id="cb6-11" data-line-number="11"></a>
<a class="sourceLine" id="cb6-12" data-line-number="12">immuneRes <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">write_tsv</span>(<span class="st">&quot;../results/tables/go_enrichment_immune_bulk_single_cell.tsv&quot;</span>)</a>
<a class="sourceLine" id="cb6-13" data-line-number="13"></a>
<a class="sourceLine" id="cb6-14" data-line-number="14">immuneRes <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(FDR) <span class="op">%&gt;%</span><span class="st"> </span>DT<span class="op">::</span><span class="kw">datatable</span>(<span class="dt">options =</span> <span class="kw">list</span>(<span class="dt">scrollX=</span><span class="st">&quot;500px&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span>DT<span class="op">::</span><span class="kw">formatSignif</span>(<span class="dt">columns=</span><span class="kw">c</span>(<span class="st">&quot;BioQC.twosided.pvalue&quot;</span>, <span class="st">&quot;BioQC.Qvalue&quot;</span>, <span class="st">&quot;FDR&quot;</span>, <span class="st">&quot;Bonferroni&quot;</span>))</a></code></pre></div>
<div id="htmlwidget-123dfd4370ea94f6b457" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-123dfd4370ea94f6b457">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57"],["GO:0032740","GO:0045089","GO:0032693","GO:0002218","GO:0050702","GO:0002227","GO:0032743","GO:0032700","GO:0048246","GO:0043011","GO:0070102","GO:0050718","GO:0019221","GO:0032725","GO:0001819","GO:2000484","GO:0002726","GO:0032695","GO:0010759","GO:0019882","GO:0032735","GO:2000483","GO:2000778","GO:0097028","GO:0032722","GO:0032755","GO:0090197","GO:0071347","GO:0002250","GO:0070498","GO:0006955","GO:0032757","GO:0097011","GO:0032703","GO:0010744","GO:0010745","GO:0071345","GO:0050715","GO:0043032","GO:0060760","GO:0045086","GO:0045651","GO:1900016","GO:0045087","GO:0071354","GO:0061844","GO:0045416","GO:0002740","GO:0061760","GO:0050713","GO:0032753","GO:0032715","GO:0070555","GO:2000510","GO:0070098","GO:0034097","GO:0032733"],["BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP","BP"],["positive regulation of interleukin-17 production","positive regulation of innate immune response","negative regulation of interleukin-10 production","activation of innate immune response","interleukin-1 beta secretion","innate immune response in mucosa","positive regulation of interleukin-2 production","negative regulation of interleukin-17 production","macrophage chemotaxis","myeloid dendritic cell differentiation","interleukin-6-mediated signaling pathway","positive regulation of interleukin-1 beta secretion","cytokine-mediated signaling pathway","positive regulation of granulocyte macrophage colony-stimulating factor production","positive regulation of cytokine production","positive regulation of interleukin-8 secretion","positive regulation of T cell cytokine production","negative regulation of interleukin-12 production","positive regulation of macrophage chemotaxis","antigen processing and presentation","positive regulation of interleukin-12 production","negative regulation of interleukin-8 secretion","positive regulation of interleukin-6 secretion","dendritic cell differentiation","positive regulation of chemokine production","positive regulation of interleukin-6 production","positive regulation of chemokine secretion","cellular response to interleukin-1","adaptive immune response","interleukin-1-mediated signaling pathway","immune response","positive regulation of interleukin-8 production","cellular response to granulocyte macrophage colony-stimulating factor stimulus","negative regulation of interleukin-2 production","positive regulation of macrophage derived foam cell differentiation","negative regulation of macrophage derived foam cell differentiation","cellular response to cytokine stimulus","positive regulation of cytokine secretion","positive regulation of macrophage activation","positive regulation of response to cytokine stimulus","positive regulation of interleukin-2 biosynthetic process","positive regulation of macrophage differentiation","negative regulation of cytokine production involved in inflammatory response","innate immune response","cellular response to interleukin-6","antimicrobial humoral immune response mediated by antimicrobial peptide","positive regulation of interleukin-8 biosynthetic process","negative regulation of cytokine secretion involved in immune response","antifungal innate immune response","negative regulation of interleukin-1 beta secretion","positive regulation of interleukin-4 production","negative regulation of interleukin-6 production","response to interleukin-1","positive regulation of dendritic cell chemotaxis","chemokine-mediated signaling pathway","response to cytokine","positive regulation of interleukin-10 production"],[10,8,8,17,5,15,9,6,6,5,7,19,23,9,13,14,6,7,5,12,9,6,15,6,9,16,7,24,5,11,20,25,6,8,7,7,10,15,6,5,5,7,6,58,8,45,8,5,5,7,8,10,17,5,7,19,11],[0.00194358443042518,0.00675603784610174,0.0121909367793501,0.0186772808600602,0.0228823994540547,0.0248128857221808,0.0282191107377223,0.0425846105420883,0.0571458013217533,0.0662601638588125,0.0671384329153276,0.0920157020048618,0.101362838724875,0.106536270651512,0.123905191808594,0.124589319019652,0.125547737883596,0.131529199619524,0.163871121760651,0.164790354747549,0.170489352847342,0.183289715787735,0.184211258777008,0.194566690546936,0.240241683043442,0.247975988080747,0.260862577798304,0.288615371683704,0.322985753764091,0.325136835409782,0.360595932709415,0.369025656613624,0.381848224569384,0.38887641179945,0.415523804607904,0.495270492412008,0.501064311138901,0.532228228481548,0.562745778918914,0.588775013636298,0.602742232862715,0.603462892393252,0.610030270399888,0.619412210321159,0.67119213330294,0.73285488843824,0.748164598001586,0.80206744267758,0.810495200943551,0.813306957277805,0.868021127329586,0.869495513331319,0.906872544830846,0.906723483815201,0.915825608237749,0.924287112378566,0.948899627613372],[2.71139658862092,-2.1703079262551,1.91396292092215,-1.72868635045221,-1.6404984372646,-1.60532272462983,-1.5494566761871,1.37074732024411,-1.24301567305137,1.17874749445934,1.17302879987932,-1.03613805621755,0.994121235286363,0.972502509952328,0.906910495669267,-0.904519187930921,-0.90119110798907,-0.880977822642624,-0.785497573366721,-0.783068211343407,0.768302737750839,-0.736861902206017,-0.734683829756432,0.710931508106084,0.619351638293593,0.605590370580668,0.583588218413693,-0.539680542058778,-0.490816633080942,0.487933825616199,-0.442979176148889,0.432943438344005,0.418109224339927,-0.410188399192535,0.381404091221092,0.30515754596138,-0.300106529261936,0.273902094931845,0.249687753900252,-0.230050628816841,0.219868377377879,0.219349429997984,0.214648614265521,0.208020237754121,-0.173153142291312,0.134982010883409,-0.126002845798749,0.0957891120758813,0.091249552326761,-0.0897455125954177,0.0614697041196008,-0.0607326546508922,0.0424537459787502,-0.0425251360520464,-0.03818721693318,-0.0341931024236304,-0.0227797237993173],[0.0424071612390865,0.088446186888109,0.120835509126196,0.145084033692593,0.163458106444727,0.168967890599704,0.178701919316861,0.220963839982532,0.255705138336205,0.274506680150999,0.276147486192129,0.32128301639816,0.337532359765534,0.342599065014129,0.365807724785424,0.365941192146182,0.367343381214966,0.375377168618355,0.413467775279352,0.413510079656773,0.421349630391866,0.437869383597185,0.439301446427816,0.449713544170457,0.498093842400477,0.504989323282658,0.52002363167667,0.548728478445947,0.584725993602455,0.586526370018748,0.620687918609888,0.627649194563761,0.636228569082515,0.643260548326744,0.660985293165879,0.715876781145685,0.722069238448986,0.748257392390865,0.774526750520305,0.790532411298069,0.795897668869441,0.79622405813367,0.798538467598853,0.801736369404392,0.83344237257292,0.86752372138335,0.872731717933622,0.897738466857244,0.901796251507579,0.903630571834846,0.937151933417569,0.938301564315616,0.958410326590079,0.958410326590079,0.962420618350302,0.966480997869086,0.975751840448364],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>GOID<\/th>\n      <th>ONTOLOGY<\/th>\n      <th>TERM<\/th>\n      <th>GeneCount<\/th>\n      <th>BioQC.twosided.pvalue<\/th>\n      <th>BioQC.Qvalue<\/th>\n      <th>FDR<\/th>\n      <th>Bonferroni<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":"500px","columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nDTWidget.formatSignif(this, row, data, 5, 2, 3, ',', '.');\nDTWidget.formatSignif(this, row, data, 6, 2, 3, ',', '.');\nDTWidget.formatSignif(this, row, data, 7, 2, 3, ',', '.');\nDTWidget.formatSignif(this, row, data, 8, 2, 3, ',', '.');\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>All in all the bulk expression data and aggregated single-cell data are concordant, using a very rough method (non-parameterized gene-set enrichment test with BioQC), we observe that some classes of genes, such as protein ubiquitination/deubiquitination, spindle organization, and plama membrane genes, show tendency to be over- od under-represented in single-cell aggregates than bulk data. The reason of this observation is not clear.</p>
<p>However, we note that, by a limited keyword-search, we found that there is no significant enrichment of gene-ontology terms invovled in immune response that are significantly different between scRNA and bulk data. This further underlines the legitimity of using this particular pair of aggregated single-cell data and paired bulk sequencing data to benchmark different methods.</p>
</div>
</div>
<div id="compare-predicted-fractions" class="section level2">
<h2><span class="header-section-number">3.3</span> Compare predicted fractions</h2>
<p>Most importantly, we ask how consistent the methods’ predictions are on simulated vs. genuine bulk RNA-seq samples. We run all methods on both datasets and compare the results.</p>
<div class="figure"><span id="fig:unnamed-chunk-19"></span>
<img src="_main_files/figure-html/unnamed-chunk-19-1.png" alt="Correlation of the methods' predictions on both simulated and genuine bulk RNA-seq samples" width="1152" />
<p class="caption">
Figure 3.3: Correlation of the methods’ predictions on both simulated and genuine bulk RNA-seq samples
</p>
</div>
<p>The same, but only with the cell types we look at:</p>
<div class="figure"><span id="fig:unnamed-chunk-20"></span>
<img src="_main_files/figure-html/unnamed-chunk-20-1.png" alt="Correlation of the methods' predictions on both simulated and genuine bulk RNA-seq samples on the cell types used in the benchmark only. " width="1152" />
<p class="caption">
Figure 3.4: Correlation of the methods’ predictions on both simulated and genuine bulk RNA-seq samples on the cell types used in the benchmark only.
</p>
</div>
<p>The results show a strong correlation between both datasets and suggest that the approach is valid in general.
The poor overlap of xCell is proabably due to the fact that there is little variance between the samples which xCell
requires to compute a meaningful score (see their <a href="https://github.com/dviraran/xCell/blob/ce4d43121c4a161b1e72a50dc875e43d9cf89b0d/README.Md">README on
GitHub</a>).</p>
<p>To demonstrate that this is the case, we run xCell on the same sample, but this time include 50 immune cell reference samples in the run. By adding additional samples,
we add additional variance which enables xCell to compute a meaningful score.</p>
<div class="figure"><span id="fig:unnamed-chunk-23"></span>
<img src="_main_files/figure-html/unnamed-chunk-23-1.png" alt="Correlation of xCell's prediction on both simulated and genuine bulk RNA-seq samples, including additional samples to increase the variance. " width="768" />
<p class="caption">
Figure 3.5: Correlation of xCell’s prediction on both simulated and genuine bulk RNA-seq samples, including additional samples to increase the variance.
</p>
</div>
</div>
<div id="correlation-with-immune-reference-samples." class="section level2">
<h2><span class="header-section-number">3.4</span> Correlation with immune reference samples.</h2>
<p>Finally, to demonstrate that the simulated bulk samples are also biologically meaningflu, we generate
simulated bulk samples of different immune cell types and correlate them with
reference profiles of pure immune cells.</p>
<div class="figure"><span id="fig:unnamed-chunk-24"></span>
<img src="_main_files/figure-html/unnamed-chunk-24-1.png" alt="Correlation of simulated bulk samples of a certain immune cell type (y-axis) with immune cell reference samples (x-axis)" width="768" />
<p class="caption">
Figure 3.6: Correlation of simulated bulk samples of a certain immune cell type (y-axis) with immune cell reference samples (x-axis)
</p>
</div>
<p>In general, the highest correlation is observed between the expression of the sorted cells and the simulated bulk sample.
However, the simulated Dendritic cells do not correlate well with any of the reference profiles.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="input-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="simulation-benchmark.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
