<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="manuscript in preparation">
<meta name="author" content="Gregor Sturm^{1,2}">
<meta name="author" content="Tatsiana Aneichyk^1">



  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="input-data.html">
<link rel="next" href="simulation-benchmark.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-reference-samples"><i class="fa fa-check"></i><b>2.3</b> Immune cell reference samples</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#pbmc-samples-from-hoek2015"><i class="fa fa-check"></i><b>2.4</b> 8 PBMC samples from <span class="citation">Hoek et al. (<span>2015</span>)</span></a></li>
<li class="chapter" data-level="2.5" data-path="input-data.html"><a href="input-data.html#ovarian-cancer-ascites-samples-from-schelker2017"><i class="fa fa-check"></i><b>2.5</b> 3 ovarian cancer ascites samples from <span class="citation">Schelker et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.6" data-path="input-data.html"><a href="input-data.html#metastatic-melanoma-samples-from-epic2017"><i class="fa fa-check"></i><b>2.6</b> 4 metastatic melanoma samples from <span class="citation">Racle et al. (<span>2017</span>)</span></a></li>
<li class="chapter" data-level="2.7" data-path="input-data.html"><a href="input-data.html#data-sanity-checks"><i class="fa fa-check"></i><b>2.7</b> Data sanity checks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-to-genuine-bulk-samples"><i class="fa fa-check"></i><b>3.1</b> Compare simululated to genuine bulk samples</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-predicted-fractions"><i class="fa fa-check"></i><b>3.2</b> Compare predicted fractions</a></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlation-with-immune-reference-samples."><i class="fa fa-check"></i><b>3.3</b> Correlation with immune reference samples.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html"><i class="fa fa-check"></i><b>4</b> Simulation benchmark</a><ul>
<li class="chapter" data-level="4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>4.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>4.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.3</b> Run the deconvolution</a></li>
<li class="chapter" data-level="4.4" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#results"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#correlation-plots"><i class="fa fa-check"></i><b>4.4.1</b> Correlation plots</a></li>
<li class="chapter" data-level="4.4.2" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>4.4.2</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="4.4.3" data-path="simulation-benchmark.html"><a href="simulation-benchmark.html#deviation-from-slope-xy-for-all-absolute-methods"><i class="fa fa-check"></i><b>4.4.3</b> Deviation from slope (x=y) for all absolute methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>5</b> Validation with real data</a><ul>
<li class="chapter" data-level="5.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between--and-within-sample-comparisons"><i class="fa fa-check"></i><b>5.1</b> Between- and within-sample comparisons</a></li>
<li class="chapter" data-level="5.2" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#within-sample-comparison"><i class="fa fa-check"></i><b>5.2</b> Within-sample comparison</a></li>
<li class="chapter" data-level="5.3" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#between-sample-comparisons"><i class="fa fa-check"></i><b>5.3</b> Between-sample comparisons</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="sensitivity-and-specificity.html"><a href="sensitivity-and-specificity.html"><i class="fa fa-check"></i><b>6</b> Sensitivity and Specificity</a><ul>
<li class="chapter" data-level="6.1" data-path="sensitivity-and-specificity.html"><a href="sensitivity-and-specificity.html#predictions-with-increasing-immune-cell-content"><i class="fa fa-check"></i><b>6.1</b> Predictions with increasing immune cell content</a></li>
<li class="chapter" data-level="6.2" data-path="sensitivity-and-specificity.html"><a href="sensitivity-and-specificity.html#sensitivity"><i class="fa fa-check"></i><b>6.2</b> Sensitivity</a></li>
<li class="chapter" data-level="6.3" data-path="sensitivity-and-specificity.html"><a href="sensitivity-and-specificity.html#specificity"><i class="fa fa-check"></i><b>6.3</b> Specificity</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>7</b> Spillover Analysis</a><ul>
<li class="chapter" data-level="7.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#complete-spillover-matrix"><i class="fa fa-check"></i><b>7.1</b> Complete Spillover Matrix</a></li>
<li class="chapter" data-level="7.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-signal-to-noise-ratio"><i class="fa fa-check"></i><b>7.2</b> Summary figure: Signal to noise ratio</a></li>
<li class="chapter" data-level="7.3" data-path="spillover-analysis.html"><a href="spillover-analysis.html#summary-figure-migration-charts"><i class="fa fa-check"></i><b>7.3</b> Summary figure: migration charts</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html"><i class="fa fa-check"></i><b>8</b> Creating publication ready figures</a><ul>
<li class="chapter" data-level="8.1" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#benchmark-results-sensitivity-specificity-and-absolute-deviation"><i class="fa fa-check"></i><b>8.1</b> Benchmark results, Sensitivity, Specificity and absolute deviation</a></li>
<li class="chapter" data-level="8.2" data-path="creating-publication-ready-figures.html"><a href="creating-publication-ready-figures.html#migration-charts-for-spillover-analysis"><i class="fa fa-check"></i><b>8.2</b> Migration charts for Spillover analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Benchmarking methods for estimating immune cell abundance from bulk RNA-sequencing data (supplementary information)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="validitiy-of-single-cell-rna-seq-as-reference" class="section level1">
<h1><span class="header-section-number">3</span> Validitiy of single cell RNA-seq as reference</h1>
<p>Since we rely on simulated bulk RNA-seq samples generated from scRNA-seq data, we
first investigate if this approach is valid and will give reasonable
estimates of the methods’s performances.</p>
<p>To test, we try:</p>
<ul>
<li>correlation of simulated bulk samples with a corresponding genuine bulk sample.</li>
<li>correlation of the predicted fractions with all methods on bulk vs. simulated samples.</li>
<li>correlation of the simulated bulk samples with a reference immune cell sample of the same cell type.</li>
</ul>
<div id="compare-simululated-to-genuine-bulk-samples" class="section level2">
<h2><span class="header-section-number">3.1</span> Compare simululated to genuine bulk samples</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">bulk_mean =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">colnames</span>(schelker_ovarian<span class="op">$</span>expr_mat), <span class="cf">function</span>(donor) {</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">  ind =<span class="st"> </span><span class="kw">pData</span>(single_cell_schelker<span class="op">$</span>eset)<span class="op">$</span>donor <span class="op">==</span><span class="st"> </span>donor</a>
<a class="sourceLine" id="cb4-3" data-line-number="3">  <span class="kw">apply</span>(<span class="kw">exprs</span>(single_cell_schelker<span class="op">$</span>eset)[,ind], <span class="dv">1</span>, mean)</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">})</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">bulk_mean =<span class="st"> </span><span class="kw">apply</span>(bulk_mean, <span class="dv">2</span>, scale_to_million)</a></code></pre></div>
<div class="figure"><span id="fig:unnamed-chunk-13"></span>
<img src="_main_files/figure-html/unnamed-chunk-13-1.png" alt="Correlation of simulated bulk samples with corresponding genuine bulk RNA-seq samples. " width="1152" />
<p class="caption">
Figure 3.1: Correlation of simulated bulk samples with corresponding genuine bulk RNA-seq samples.
</p>
</div>
</div>
<div id="compare-predicted-fractions" class="section level2">
<h2><span class="header-section-number">3.2</span> Compare predicted fractions</h2>
<p>We run all methods on both datasets and compare the results.</p>
<div class="figure"><span id="fig:unnamed-chunk-15"></span>
<img src="_main_files/figure-html/unnamed-chunk-15-1.png" alt="Correlation of the methods' predictions on both simulated and genuine bulk RNA-seq samples" width="1152" />
<p class="caption">
Figure 3.2: Correlation of the methods’ predictions on both simulated and genuine bulk RNA-seq samples
</p>
</div>
<p>The results show a strong correlation between both datasets and suggest that the approach is valid in general.
The poor overlap of xCell is proabably due to the fact that there is little variance between the samples which xCell
requires to compute a meaningful score (see their <a href="https://github.com/dviraran/xCell/blob/ce4d43121c4a161b1e72a50dc875e43d9cf89b0d/README.Md">README on
GitHub</a>).</p>
<p>To demonstrate that this is the case, we run xCell on the same sample, but this time include 50 immune cell reference samples in the run. By adding additional samples,
we add additional variance which enables xCell to compute a meaningful score.</p>
<div class="figure"><span id="fig:unnamed-chunk-18"></span>
<img src="_main_files/figure-html/unnamed-chunk-18-1.png" alt="Correlation of xCell's prediction on both simulated and genuine bulk RNA-seq samples, including additional samples to increase the variance. " width="768" />
<p class="caption">
Figure 3.3: Correlation of xCell’s prediction on both simulated and genuine bulk RNA-seq samples, including additional samples to increase the variance.
</p>
</div>
</div>
<div id="correlation-with-immune-reference-samples." class="section level2">
<h2><span class="header-section-number">3.3</span> Correlation with immune reference samples.</h2>
<p>To demonstrate that the simulated bulk samples are also biologically meaningflu, we generate
simulated bulk samples of different immune cell types and correlate them with
reference profiles of pure immune cells.</p>
<div class="figure"><span id="fig:unnamed-chunk-19"></span>
<img src="_main_files/figure-html/unnamed-chunk-19-1.png" alt="Correlation of simulated bulk samples of a certain immune cell type (y-axis) with immune cell reference samples (x-axis)" width="768" />
<p class="caption">
Figure 3.4: Correlation of simulated bulk samples of a certain immune cell type (y-axis) with immune cell reference samples (x-axis)
</p>
</div>
<p>In general, the highest correlation is observed between the expression of the sorted cells and the simulated bulk sample.
However, the simulated Dendritic cells do not correlate well with any of the reference profiles.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="input-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="simulation-benchmark.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
