<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Immune Deconvolution Benchmark</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Immune Deconvolution Benchmark" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Immune Deconvolution Benchmark" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm">


<meta name="date" content="2018-05-08">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="input-data.html">
<link rel="next" href="spillover-analysis.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#conceptual-differences-between-methods"><i class="fa fa-check"></i><b>1.1</b> Conceptual differences between methods</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#gene-set-based-methods-allow-comparison-between-samples-but-not-between-cell-types"><i class="fa fa-check"></i><b>1.1.1</b> Gene-set based methods (allow comparison between samples, but not between cell types)</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#deconvolution-based-methods-relative-fractions-allow-comparisons-between-cell-types-but-not-samples"><i class="fa fa-check"></i><b>1.1.2</b> Deconvolution-based methods (relative fractions, allow comparisons between cell types, but not samples)</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#deconvolution-based-methods-absolute-fractions-allow-comparisons-between-cell-types-and-samples"><i class="fa fa-check"></i><b>1.1.3</b> Deconvolution-based methods (absolute fractions, allow comparisons between cell types and samples)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.1</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#validation-data-rnaseq-matched-gold-standard"><i class="fa fa-check"></i><b>2.2</b> Validation data (RNAseq + matched gold standard)</a><ul>
<li class="chapter" data-level="2.2.1" data-path="input-data.html"><a href="input-data.html#hoeck-data"><i class="fa fa-check"></i><b>2.2.1</b> Hoeck data</a></li>
<li class="chapter" data-level="2.2.2" data-path="input-data.html"><a href="input-data.html#schelker-data"><i class="fa fa-check"></i><b>2.2.2</b> Schelker data</a></li>
<li class="chapter" data-level="2.2.3" data-path="input-data.html"><a href="input-data.html#racle-data"><i class="fa fa-check"></i><b>2.2.3</b> Racle data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-single-cell-sequencing-for-counting-immune-cells"><i class="fa fa-check"></i><b>3.1</b> Validity of single cell sequencing for counting immune cells</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data"><i class="fa fa-check"></i><b>3.2</b> Validity of simulated bulk tissues from single cell sequencing data</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-samplese-to-bulk-samples"><i class="fa fa-check"></i><b>3.2.1</b> Compare simululated samplese to bulk samples</a></li>
<li class="chapter" data-level="3.2.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-the-predicted-fractions"><i class="fa fa-check"></i><b>3.2.2</b> Compare the predicted fractions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="spillover-analysis.html"><a href="spillover-analysis.html"><i class="fa fa-check"></i><b>4</b> Spillover analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#run-the-deconvolution"><i class="fa fa-check"></i><b>4.1</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="4.1.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#run-timer"><i class="fa fa-check"></i><b>4.1.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="spillover-analysis.html"><a href="spillover-analysis.html#results"><i class="fa fa-check"></i><b>4.2</b> Results</a><ul>
<li class="chapter" data-level="4.2.1" data-path="spillover-analysis.html"><a href="spillover-analysis.html#calculate-signal-to-noise-ratio-for-each-cell-type"><i class="fa fa-check"></i><b>4.2.1</b> Calculate signal to noise ratio for each cell type</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="mixing-study.html"><a href="mixing-study.html"><i class="fa fa-check"></i><b>5</b> Mixing study</a><ul>
<li class="chapter" data-level="5.1" data-path="mixing-study.html"><a href="mixing-study.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>5.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="5.2" data-path="mixing-study.html"><a href="mixing-study.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>5.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="5.3" data-path="mixing-study.html"><a href="mixing-study.html#run-the-deconvolution-1"><i class="fa fa-check"></i><b>5.3</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="5.3.1" data-path="mixing-study.html"><a href="mixing-study.html#run-timer-1"><i class="fa fa-check"></i><b>5.3.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="mixing-study.html"><a href="mixing-study.html#results-1"><i class="fa fa-check"></i><b>5.4</b> Results</a><ul>
<li class="chapter" data-level="5.4.1" data-path="mixing-study.html"><a href="mixing-study.html#calculate-other-cells-for-all-absolute-methods"><i class="fa fa-check"></i><b>5.4.1</b> Calculate “Other cells” for all absolute methods</a></li>
<li class="chapter" data-level="5.4.2" data-path="mixing-study.html"><a href="mixing-study.html#correlation-plots"><i class="fa fa-check"></i><b>5.4.2</b> Correlation plots</a></li>
<li class="chapter" data-level="5.4.3" data-path="mixing-study.html"><a href="mixing-study.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>5.4.3</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="5.4.4" data-path="mixing-study.html"><a href="mixing-study.html#absolute-error-for-all-qualifying-methods"><i class="fa fa-check"></i><b>5.4.4</b> Absolute error for all qualifying methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>6</b> Validation with real data</a><ul>
<li class="chapter" data-level="6.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#run-the-deconvolution-2"><i class="fa fa-check"></i><b>6.1</b> Run the deconvolution</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>7</b> Conclusions</a></li>
<li class="chapter" data-level="8" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>8</b> Materials and Methods</a><ul>
<li class="chapter" data-level="8.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#data-acquisition"><i class="fa fa-check"></i><b>8.1</b> Data acquisition</a></li>
<li class="chapter" data-level="8.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#validation-datasets-flow-cytometry-scrnaseq-imageing-matched-rna-seq"><i class="fa fa-check"></i><b>8.2</b> Validation datasets (Flow cytometry, scRNAseq, Imageing + matched RNA seq)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>9</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Immune Deconvolution Benchmark</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="validitiy-of-single-cell-rna-seq-as-reference" class="section level1">
<h1><span class="header-section-number">3</span> Validitiy of single cell RNA-seq as reference</h1>
<div id="validity-of-single-cell-sequencing-for-counting-immune-cells" class="section level2">
<h2><span class="header-section-number">3.1</span> Validity of single cell sequencing for counting immune cells</h2>
<ul>
<li>Schelker et al have shown that measurements from single cell sequencing are highly consistent with flow cytometry.</li>
</ul>
</div>
<div id="validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data" class="section level2">
<h2><span class="header-section-number">3.2</span> Validity of simulated bulk tissues from single cell sequencing data</h2>
<p>We use simulated samples from single cell gene expression data. Does this approach make sense after all? We have 3 samples of single cell samples matched to bulk RNA sequencing.</p>
<p>To test, we try: * correlation of the simulated gene expression with the measured bulk gene expression. * correlation of the predicted fractions with all methods on bulk vs. simulated tissue.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ovarian_bulk_replicates =<span class="st"> </span><span class="kw">read_xls</span>(<span class="st">&quot;../data/single_cell_schelker/ascites_bulk_samples.xls&quot;</span>) 

samples =<span class="st"> </span><span class="kw">list</span>(<span class="st">&quot;7873M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7873_1&quot;</span>, <span class="st">&quot;ascites_7873_2&quot;</span>),
               <span class="st">&quot;7882M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7882_1&quot;</span>, <span class="st">&quot;ascites_7882_2&quot;</span>),
               <span class="st">&quot;7892M&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;ascites_7892_1&quot;</span>, <span class="st">&quot;ascites_7892_2&quot;</span>))

ovarian_bulk =<span class="st"> </span><span class="kw">lapply</span>(samples, <span class="cf">function</span>(cols) {
  <span class="kw">apply</span>(ovarian_bulk_replicates[,cols], <span class="dv">1</span>, mean)
}) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">bind_cols</span>() 

ovarian_bulk <span class="op">%&lt;&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">gene_symbol =</span> ovarian_bulk_replicates<span class="op">$</span>Row)

ovarian_bulk_mat =<span class="st"> </span>ovarian_bulk <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;gene_symbol&quot;</span>) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">as.matrix</span>()</code></pre></div>
<p>Load the ‘ground truth’</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">true_cell_count =<span class="st"> </span><span class="kw">pData</span>(single_cell_schelker2) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(source <span class="op">==</span><span class="st"> &quot;ascites&quot;</span>, donor <span class="op">%in%</span><span class="st"> </span><span class="kw">colnames</span>(ovarian_bulk)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(donor, cell_type) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">count</span>() <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">rename</span>(<span class="dt">cell_count=</span>n) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">true_fraction=</span>cell_count<span class="op">/</span><span class="kw">sum</span>(cell_count)) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">arrange</span>(cell_type, donor)

total_cells =<span class="st"> </span>true_cell_count <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">group_by</span>(donor) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">summarise</span>(<span class="kw">sum</span>(cell_count))

total_cells</code></pre></div>
<pre><code>## # A tibble: 3 x 2
##   donor `sum(cell_count)`
##   &lt;chr&gt;             &lt;int&gt;
## 1 7873M               864
## 2 7882M               902
## 3 7892M               773</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">true_cell_count</code></pre></div>
<pre><code>## # A tibble: 25 x 4
## # Groups:   donor [3]
##    donor cell_type                     cell_count true_fraction
##    &lt;chr&gt; &lt;chr&gt;                              &lt;int&gt;         &lt;dbl&gt;
##  1 7873M B cells                                5       0.00579
##  2 7882M B cells                                4       0.00443
##  3 7892M B cells                                5       0.00647
##  4 7873M Cancer associated fibroblasts          9       0.0104 
##  5 7882M Cancer associated fibroblasts         11       0.0122 
##  6 7892M Cancer associated fibroblasts         16       0.0207 
##  7 7873M CD4+ T cells                          29       0.0336 
##  8 7882M CD4+ T cells                         137       0.152  
##  9 7892M CD4+ T cells                         100       0.129  
## 10 7873M CD8+ T cells                           4       0.00463
## # ... with 15 more rows</code></pre>
<div id="compare-simululated-samplese-to-bulk-samples" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Compare simululated samplese to bulk samples</h3>
<p>randomized:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cell_fractions =<span class="st"> </span>true_cell_count <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">select</span>(donor, cell_type, true_fraction) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">complete</span>(donor, <span class="dt">cell_type=</span>available_cell_types) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">true_fraction=</span><span class="kw">ifelse</span>(<span class="kw">is.na</span>(true_fraction), <span class="dv">0</span>, true_fraction)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">spread</span>(cell_type, true_fraction) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;donor&quot;</span>)

bulk_eset_rand =<span class="st"> </span><span class="kw">make_bulk_eset</span>(<span class="dt">eset=</span>single_cell_schelker2, <span class="dt">cell_fractions =</span> cell_fractions, <span class="dt">n_cells=</span><span class="dv">500</span>)
bulk_rand =<span class="st"> </span><span class="kw">eset_to_matrix</span>(bulk_eset_rand, <span class="dt">column =</span> <span class="st">&quot;gene_symbol&quot;</span>)
<span class="kw">colnames</span>(bulk_rand) =<span class="st"> </span><span class="kw">names</span>(samples)</code></pre></div>
<p>summing up all cells:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bulk_sum =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">names</span>(samples), <span class="cf">function</span>(donor) {
  ind =<span class="st"> </span><span class="kw">pData</span>(single_cell_schelker2)<span class="op">$</span>donor <span class="op">==</span><span class="st"> </span>donor
  <span class="co"># apply(exprs(single_cell_schelker2)[,ind], 1, function(x){quantile(x, c(.98))})</span>
  <span class="kw">apply</span>(<span class="kw">exprs</span>(single_cell_schelker2)[,ind], <span class="dv">1</span>, sum)
})</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genes =<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">rownames</span>(bulk_sum), ovarian_bulk<span class="op">$</span>gene_symbol)

expr_all =<span class="st"> </span><span class="kw">bind_rows</span>(
  ovarian_bulk  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;bulk&quot;</span>),
  bulk_rand <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>(<span class="dt">rownames=</span><span class="st">&quot;gene_symbol&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;rand&quot;</span>),
  bulk_sum <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>(<span class="dt">rownames=</span><span class="st">&quot;gene_symbol&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;sum&quot;</span>)
) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(gene_symbol <span class="op">%in%</span><span class="st"> </span>genes) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(sample, tpm, <span class="op">-</span>gene_symbol, <span class="op">-</span>source) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">spread</span>(source, tpm)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">expr_all <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">bulk=</span><span class="kw">log2</span>(bulk<span class="op">+</span><span class="dv">1</span>), <span class="dt">sum=</span><span class="kw">log2</span>(sum<span class="op">+</span><span class="dv">1</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>bulk, <span class="dt">y=</span>sum)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">stat_cor</span>() <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>sample) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">slope=</span><span class="dv">1</span>, <span class="dt">intercept =</span> <span class="dv">0</span>, <span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-15-1.png" width="1536" /></p>
</div>
<div id="compare-the-predicted-fractions" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Compare the predicted fractions</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results_bulk =<span class="st"> </span><span class="kw">lapply</span>(immunedeconv<span class="op">::</span>deconvolution_methods, <span class="cf">function</span>(method) {
  <span class="kw">deconvolute</span>(ovarian_bulk_mat, method) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">method=</span>method) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;bulk&quot;</span>)
})</code></pre></div>
<pre><code>## Warning in EPIC::EPIC(bulk = gene_expression_matrix, ...): mRNA_cell value
## unknown for some cell types: CAFs, Endothelial - using the default value of
## 0.4 for these but this might bias the true cell proportions from all cell
## types.</code></pre>
<pre><code>## 
## Running quanTIseq deconvolution module</code></pre>
<pre><code>## Gene expression normalization and re-annotation (arrays: FALSE)</code></pre>
<pre><code>## Removing 17 noisy genes</code></pre>
<pre><code>## Signature genes found in data set: 152/153 (99.35%)</code></pre>
<pre><code>## Mixture deconvolution (method: lsei)</code></pre>
<pre><code>## Deconvolution sucessful!</code></pre>
<pre><code>## [1] &quot;Num. of genes: 10781&quot;
## Estimating ssGSEA scores for 489 gene sets.
## 
  |                                                                       
  |                                                                 |   0%Using parallel with 4 cores
## 
  |                                                                       
  |======================                                           |  33%
  |                                                                       
  |===========================================                      |  67%
  |                                                                       
  |=================================================================| 100%</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results_simulated =<span class="st"> </span><span class="kw">lapply</span>(immunedeconv<span class="op">::</span>deconvolution_methods, <span class="cf">function</span>(method) {
  <span class="kw">deconvolute</span>(bulk_sum, method) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">method=</span>method) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;sum&quot;</span>)
})</code></pre></div>
<pre><code>## Warning in EPIC::EPIC(bulk = gene_expression_matrix, ...): mRNA_cell value
## unknown for some cell types: CAFs, Endothelial - using the default value of
## 0.4 for these but this might bias the true cell proportions from all cell
## types.</code></pre>
<pre><code>## 
## Running quanTIseq deconvolution module</code></pre>
<pre><code>## Gene expression normalization and re-annotation (arrays: FALSE)</code></pre>
<pre><code>## Removing 17 noisy genes</code></pre>
<pre><code>## Signature genes found in data set: 140/153 (91.5%)</code></pre>
<pre><code>## Mixture deconvolution (method: lsei)</code></pre>
<pre><code>## Deconvolution sucessful!</code></pre>
<pre><code>## [1] &quot;Num. of genes: 10462&quot;</code></pre>
<pre><code>## Warning in .local(expr, gset.idx.list, ...): 1genes with constant
## expression values throuhgout the samples.</code></pre>
<pre><code>## Estimating ssGSEA scores for 489 gene sets.
## 
  |                                                                       
  |                                                                 |   0%Using parallel with 4 cores
## 
  |                                                                       
  |======================                                           |  33%
  |                                                                       
  |===========================================                      |  67%
  |                                                                       
  |=================================================================| 100%</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results =<span class="st"> </span><span class="kw">bind_rows</span>(all_results_bulk, all_results_simulated) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(donor, fraction, <span class="op">-</span>cell_type, <span class="op">-</span>source, <span class="op">-</span>method) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">spread</span>(source, fraction)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> bulk, <span class="dt">y=</span>sum)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">colour=</span>cell_type)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>method, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">stat_cor</span>()</code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-17-1.png" width="1152" /></p>
<!-- ### second dataset -->
<!-- We have another dataset from Chung 2017 et al with single cell and matched bulk.  -->
<!-- We do the same comparison.  -->
<!-- ```{r, cache=TRUE} -->
<!-- single_cell_chung = read_tsv("../data/single_cell_chung/GSE75688_GEO_processed_Breast_Cancer_raw_TPM_matrix.txt") -->
<!-- single_cell_chung2 = single_cell_chung %>% -->
<!--   gather(sample, tpm, -gene_id, -gene_name, -gene_type) %>% -->
<!--   select(gene_name, sample, tpm) %>% -->
<!--   group_by(sample, gene_name) %>% -->
<!--   summarise(tpm=sum(tpm)) %>% -->
<!--   ungroup() %>%  -->
<!--   separate(sample, into=c("donor", "sample"), sep="_") -->
<!-- ``` -->
<!-- ### Overview of bulk samples -->
<!-- ```{r} -->
<!-- single_cell_chung2 %>%  -->
<!--   select(donor, sample) %>% -->
<!--   distinct() %>% -->
<!--   filter(sample %in% c("Pooled", "Tumor")) -->
<!-- ``` -->
<!-- ### Separate bulk and single cells -->
<!-- ```{r, cache=TRUE} -->
<!-- chung_bulk = single_cell_chung2 %>% -->
<!--   filter(sample %in% c("Pooled", "Tumor")) %>% -->
<!--   filter(!(donor == "BC01" & sample == "Tumor")) %>% # remove double bulk reference for sample 1 %>% -->
<!--   mutate(source="bulk") -->
<!-- chung_sum = single_cell_chung2 %>% -->
<!--   filter(!sample %in% c("Pooled", "Tumor")) %>%  -->
<!--   group_by(donor, gene_name) %>% -->
<!--   summarise(tpm = sum(tpm)) %>% -->
<!--   ungroup() %>%  -->
<!--   mutate(source="sum") -->
<!-- genes = intersect(chung_bulk$gene_name %>% unique(), chung_sum$gene_name %>% unique()) -->
<!-- all_results_chung = bind_rows(chung_bulk, chung_sum) %>% -->
<!--   spread(source, tpm) -->
<!-- ``` -->

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="input-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spillover-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
