<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Immune Deconvolution Benchmark</title>
  <meta name="description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods.">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Immune Deconvolution Benchmark" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Immune Deconvolution Benchmark" />
  
  <meta name="twitter:description" content="Systematic Evaluation of state-of-the-art Immune deconvolution methods." />
  

<meta name="author" content="Gregor Sturm">


<meta name="date" content="2018-06-04">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="input-data.html">
<link rel="next" href="sensitivity-analysis.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<script src="libs/viz-0.3/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.0/grViz.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#conceptual-differences-between-methods"><i class="fa fa-check"></i><b>1.1</b> Conceptual differences between methods</a><ul>
<li class="chapter" data-level="1.1.1" data-path="index.html"><a href="index.html#gene-set-based-methods-allow-comparison-between-samples-but-not-between-cell-types"><i class="fa fa-check"></i><b>1.1.1</b> Gene-set based methods (allow comparison between samples, but not between cell types)</a></li>
<li class="chapter" data-level="1.1.2" data-path="index.html"><a href="index.html#deconvolution-based-methods-relative-fractions-allow-comparisons-between-cell-types-but-not-samples"><i class="fa fa-check"></i><b>1.1.2</b> Deconvolution-based methods (relative fractions, allow comparisons between cell types, but not samples)</a></li>
<li class="chapter" data-level="1.1.3" data-path="index.html"><a href="index.html#deconvolution-based-methods-absolute-fractions-allow-comparisons-between-cell-types-and-samples"><i class="fa fa-check"></i><b>1.1.3</b> Deconvolution-based methods (absolute fractions, allow comparisons between cell types and samples)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="input-data.html"><a href="input-data.html"><i class="fa fa-check"></i><b>2</b> Input data</a><ul>
<li class="chapter" data-level="2.1" data-path="input-data.html"><a href="input-data.html#cell-type-hierarchy"><i class="fa fa-check"></i><b>2.1</b> cell type hierarchy</a></li>
<li class="chapter" data-level="2.2" data-path="input-data.html"><a href="input-data.html#single-cell-data-for-simulated-mixtures"><i class="fa fa-check"></i><b>2.2</b> Single cell data for simulated mixtures</a></li>
<li class="chapter" data-level="2.3" data-path="input-data.html"><a href="input-data.html#immune-cell-bulk-reference-samples-quantiseq-training-data"><i class="fa fa-check"></i><b>2.3</b> Immune cell bulk reference samples (quanTIseq training data)</a></li>
<li class="chapter" data-level="2.4" data-path="input-data.html"><a href="input-data.html#validation-data-rnaseq-matched-gold-standard"><i class="fa fa-check"></i><b>2.4</b> Validation data (RNAseq + matched gold standard)</a><ul>
<li class="chapter" data-level="2.4.1" data-path="input-data.html"><a href="input-data.html#hoeck-data"><i class="fa fa-check"></i><b>2.4.1</b> Hoeck data</a></li>
<li class="chapter" data-level="2.4.2" data-path="input-data.html"><a href="input-data.html#schelker-data"><i class="fa fa-check"></i><b>2.4.2</b> Schelker data</a></li>
<li class="chapter" data-level="2.4.3" data-path="input-data.html"><a href="input-data.html#racle-data"><i class="fa fa-check"></i><b>2.4.3</b> Racle data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html"><i class="fa fa-check"></i><b>3</b> Validitiy of single cell RNA-seq as reference</a><ul>
<li class="chapter" data-level="3.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-single-cell-sequencing-for-counting-immune-cells"><i class="fa fa-check"></i><b>3.1</b> Validity of single cell sequencing for counting immune cells</a></li>
<li class="chapter" data-level="3.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data"><i class="fa fa-check"></i><b>3.2</b> Validity of simulated bulk tissues from single cell sequencing data</a><ul>
<li class="chapter" data-level="3.2.1" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-simululated-samples-to-bulk-samples"><i class="fa fa-check"></i><b>3.2.1</b> Compare simululated samples to bulk samples</a></li>
<li class="chapter" data-level="3.2.2" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#compare-the-predicted-fractions"><i class="fa fa-check"></i><b>3.2.2</b> Compare the predicted fractions</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="validitiy-of-single-cell-rna-seq-as-reference.html"><a href="validitiy-of-single-cell-rna-seq-as-reference.html#correlated-simulated-samples-with-bulk-of-the-same-tissue."><i class="fa fa-check"></i><b>3.3</b> Correlated simulated samples with bulk of the same tissue.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sensitivity-analysis.html"><a href="sensitivity-analysis.html"><i class="fa fa-check"></i><b>4</b> Sensitivity analysis</a></li>
<li class="chapter" data-level="5" data-path="specificity-analysis.html"><a href="specificity-analysis.html"><i class="fa fa-check"></i><b>5</b> Specificity Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="specificity-analysis.html"><a href="specificity-analysis.html#create-datasets"><i class="fa fa-check"></i><b>5.1</b> Create datasets</a></li>
<li class="chapter" data-level="5.2" data-path="specificity-analysis.html"><a href="specificity-analysis.html#run-the-deconvolution"><i class="fa fa-check"></i><b>5.2</b> Run the deconvolution</a></li>
<li class="chapter" data-level="5.3" data-path="specificity-analysis.html"><a href="specificity-analysis.html#results"><i class="fa fa-check"></i><b>5.3</b> Results</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="correlation-of-methods-across-tcga-samples-.html"><a href="correlation-of-methods-across-tcga-samples-.html"><i class="fa fa-check"></i><b>6</b> Correlation of methods across TCGA samples.</a></li>
<li class="chapter" data-level="7" data-path="mixing-study.html"><a href="mixing-study.html"><i class="fa fa-check"></i><b>7</b> Mixing study</a><ul>
<li class="chapter" data-level="7.1" data-path="mixing-study.html"><a href="mixing-study.html#average-fraction-of-tumour-cells"><i class="fa fa-check"></i><b>7.1</b> Average fraction of tumour cells</a></li>
<li class="chapter" data-level="7.2" data-path="mixing-study.html"><a href="mixing-study.html#create-simulated-bulk-tissues"><i class="fa fa-check"></i><b>7.2</b> Create simulated bulk tissues</a></li>
<li class="chapter" data-level="7.3" data-path="mixing-study.html"><a href="mixing-study.html#run-the-deconvolution-1"><i class="fa fa-check"></i><b>7.3</b> Run the deconvolution</a><ul>
<li class="chapter" data-level="7.3.1" data-path="mixing-study.html"><a href="mixing-study.html#run-timer"><i class="fa fa-check"></i><b>7.3.1</b> run TIMER</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="mixing-study.html"><a href="mixing-study.html#results-1"><i class="fa fa-check"></i><b>7.4</b> Results</a><ul>
<li class="chapter" data-level="7.4.1" data-path="mixing-study.html"><a href="mixing-study.html#calculate-other-cells-for-all-absolute-methods"><i class="fa fa-check"></i><b>7.4.1</b> Calculate “Other cells” for all absolute methods</a></li>
<li class="chapter" data-level="7.4.2" data-path="mixing-study.html"><a href="mixing-study.html#correlation-plots"><i class="fa fa-check"></i><b>7.4.2</b> Correlation plots</a></li>
<li class="chapter" data-level="7.4.3" data-path="mixing-study.html"><a href="mixing-study.html#calculate-correlations-for-each-method-and-cell-type"><i class="fa fa-check"></i><b>7.4.3</b> Calculate correlations for each method and cell type</a></li>
<li class="chapter" data-level="7.4.4" data-path="mixing-study.html"><a href="mixing-study.html#absolute-error-for-all-qualifying-methods"><i class="fa fa-check"></i><b>7.4.4</b> Absolute error for all qualifying methods</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html"><i class="fa fa-check"></i><b>8</b> Validation with real data</a><ul>
<li class="chapter" data-level="8.1" data-path="validation-with-real-data.html"><a href="validation-with-real-data.html#run-the-deconvolution-2"><i class="fa fa-check"></i><b>8.1</b> Run the deconvolution</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i><b>9</b> Conclusions</a></li>
<li class="chapter" data-level="10" data-path="materials-and-methods.html"><a href="materials-and-methods.html"><i class="fa fa-check"></i><b>10</b> Materials and Methods</a><ul>
<li class="chapter" data-level="10.1" data-path="materials-and-methods.html"><a href="materials-and-methods.html#data-acquisition"><i class="fa fa-check"></i><b>10.1</b> Data acquisition</a></li>
<li class="chapter" data-level="10.2" data-path="materials-and-methods.html"><a href="materials-and-methods.html#validation-datasets-flow-cytometry-scrnaseq-imageing-matched-rna-seq"><i class="fa fa-check"></i><b>10.2</b> Validation datasets (Flow cytometry, scRNAseq, Imageing + matched RNA seq)</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>11</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Immune Deconvolution Benchmark</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="validitiy-of-single-cell-rna-seq-as-reference" class="section level1">
<h1><span class="header-section-number">3</span> Validitiy of single cell RNA-seq as reference</h1>
<div id="validity-of-single-cell-sequencing-for-counting-immune-cells" class="section level2">
<h2><span class="header-section-number">3.1</span> Validity of single cell sequencing for counting immune cells</h2>
<ul>
<li><span class="citation">Schelker et al. (<a href="#ref-Schelker2017">2017</a>)</span> have shown that measurements from single cell sequencing are highly consistent with flow cytometry.</li>
</ul>
</div>
<div id="validity-of-simulated-bulk-tissues-from-single-cell-sequencing-data" class="section level2">
<h2><span class="header-section-number">3.2</span> Validity of simulated bulk tissues from single cell sequencing data</h2>
<p>We use simulated samples from single cell gene expression data. Does this approach make sense after all? We have 3 samples of single cell samples matched to bulk RNA sequencing.</p>
<p>To test, we try:</p>
<ul>
<li>correlation of the simulated gene expression with the measured bulk gene expression.</li>
<li>correlation of the predicted fractions with all methods on bulk vs. simulated tissue.</li>
<li>correlation of the simulated gene expression with bulk gene expression from the same tissue.</li>
</ul>
<div id="compare-simululated-samples-to-bulk-samples" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Compare simululated samples to bulk samples</h3>
<p>Simulate bulk samples by summing up the expression values of all cells.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bulk_sum =<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">colnames</span>(schelker_ovarian<span class="op">$</span>bulk_mat), <span class="cf">function</span>(donor) {
  ind =<span class="st"> </span><span class="kw">pData</span>(single_cell_schelker<span class="op">$</span>eset)<span class="op">$</span>donor <span class="op">==</span><span class="st"> </span>donor
  <span class="kw">apply</span>(<span class="kw">exprs</span>(single_cell_schelker<span class="op">$</span>eset)[,ind], <span class="dv">1</span>, sum)
})</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genes =<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">rownames</span>(bulk_sum), schelker_ovarian<span class="op">$</span>bulk<span class="op">$</span>gene_symbol)

expr_all =<span class="st"> </span><span class="kw">bind_rows</span>(
  schelker_ovarian<span class="op">$</span>bulk  <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;bulk&quot;</span>),
  bulk_sum <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>(<span class="dt">rownames=</span><span class="st">&quot;gene_symbol&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;single_cell_aggregate&quot;</span>)
) <span class="op">%&gt;%</span><span class="st"> </span>
<span class="st">  </span><span class="kw">filter</span>(gene_symbol <span class="op">%in%</span><span class="st"> </span>genes) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(sample, tpm, <span class="op">-</span>gene_symbol, <span class="op">-</span>source) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">spread</span>(source, tpm)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">expr_all <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">mutate</span>(<span class="dt">bulk=</span><span class="kw">log2</span>(bulk<span class="op">+</span><span class="dv">1</span>), <span class="dt">single_cell_aggregate=</span><span class="kw">log2</span>(single_cell_aggregate<span class="op">+</span><span class="dv">1</span>)) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>bulk, <span class="dt">y=</span>single_cell_aggregate)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>() <span class="op">+</span><span class="st"> </span><span class="kw">stat_cor</span>() <span class="op">+</span><span class="st"> </span><span class="kw">facet_wrap</span>(<span class="op">~</span>sample) <span class="op">+</span><span class="st"> </span>
<span class="st">  </span><span class="kw">geom_smooth</span>(<span class="dt">method =</span> lm)</code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-12-1.png" width="1536" /></p>
</div>
<div id="compare-the-predicted-fractions" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Compare the predicted fractions</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results_bulk =<span class="st"> </span><span class="kw">lapply</span>(immunedeconv<span class="op">::</span>deconvolution_methods, <span class="cf">function</span>(method) {
  <span class="kw">deconvolute</span>(schelker_ovarian<span class="op">$</span>bulk_mat, method) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">method=</span>method) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;bulk&quot;</span>)
})</code></pre></div>
<pre><code>## Warning in EPIC::EPIC(bulk = gene_expression_matrix, ...): mRNA_cell value
## unknown for some cell types: CAFs, Endothelial - using the default value of
## 0.4 for these but this might bias the true cell proportions from all cell
## types.</code></pre>
<pre><code>## 
## Running quanTIseq deconvolution module</code></pre>
<pre><code>## Gene expression normalization and re-annotation (arrays: FALSE)</code></pre>
<pre><code>## Removing 17 noisy genes</code></pre>
<pre><code>## Signature genes found in data set: 152/153 (99.35%)</code></pre>
<pre><code>## Mixture deconvolution (method: lsei)</code></pre>
<pre><code>## 
## Attaching package: &#39;limSolve&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ggplot2&#39;:
## 
##     resolution</code></pre>
<pre><code>## Deconvolution sucessful!</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results_simulated =<span class="st"> </span><span class="kw">lapply</span>(immunedeconv<span class="op">::</span>deconvolution_methods, <span class="cf">function</span>(method) {
  <span class="kw">deconvolute</span>(bulk_sum, method) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">method=</span>method) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">mutate</span>(<span class="dt">source=</span><span class="st">&quot;sum&quot;</span>)
})</code></pre></div>
<pre><code>## Warning in EPIC::EPIC(bulk = gene_expression_matrix, ...): mRNA_cell value
## unknown for some cell types: CAFs, Endothelial - using the default value of
## 0.4 for these but this might bias the true cell proportions from all cell
## types.</code></pre>
<pre><code>## 
## Running quanTIseq deconvolution module</code></pre>
<pre><code>## Gene expression normalization and re-annotation (arrays: FALSE)</code></pre>
<pre><code>## Removing 17 noisy genes</code></pre>
<pre><code>## Signature genes found in data set: 140/153 (91.5%)</code></pre>
<pre><code>## Mixture deconvolution (method: lsei)</code></pre>
<pre><code>## Deconvolution sucessful!</code></pre>
<pre><code>## Warning in .local(expr, gset.idx.list, ...): 1genes with constant
## expression values throuhgout the samples.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results =<span class="st"> </span><span class="kw">bind_rows</span>(all_results_bulk, all_results_simulated) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">gather</span>(donor, fraction, <span class="op">-</span>cell_type, <span class="op">-</span>source, <span class="op">-</span>method) <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">spread</span>(source, fraction)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">all_results <span class="op">%&gt;%</span>
<span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> bulk, <span class="dt">y=</span>sum)) <span class="op">+</span>
<span class="st">    </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">colour=</span>cell_type)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>method, <span class="dt">scales=</span><span class="st">&quot;free&quot;</span>) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">stat_cor</span>()</code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-14-1.png" width="1152" /></p>
</div>
</div>
<div id="correlated-simulated-samples-with-bulk-of-the-same-tissue." class="section level2">
<h2><span class="header-section-number">3.3</span> Correlated simulated samples with bulk of the same tissue.</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># immune_cell_reference</span>
<span class="co"># immune_cell_expr_mat</span>

cell_fractions =<span class="st"> </span><span class="kw">diag</span>(<span class="kw">length</span>(cell_types<span class="op">$</span>immune_cells))
cell_fractions =<span class="st"> </span><span class="kw">rbind</span>(cell_fractions, cell_fractions, cell_fractions)
<span class="kw">colnames</span>(cell_fractions) =<span class="st"> </span>cell_types<span class="op">$</span>immune_cells
<span class="kw">rownames</span>(cell_fractions) =<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="cf">function</span>(i) {<span class="kw">paste0</span>(cell_types<span class="op">$</span>immune_cells, <span class="st">&quot;_&quot;</span>, i)}))

simulated_bulk =<span class="st"> </span><span class="kw">make_bulk_eset</span>(<span class="dt">eset=</span>single_cell_schelker<span class="op">$</span>eset, <span class="dt">cell_fractions =</span> cell_fractions, <span class="dt">n_cells=</span><span class="dv">500</span>, <span class="dt">combine=</span>sum) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">exprs</span>() 
simulated_bulk =<span class="st"> </span><span class="kw">log2</span>(simulated_bulk<span class="op">+</span><span class="dv">1</span>)
<span class="kw">colnames</span>(simulated_bulk) =<span class="st"> </span><span class="kw">rownames</span>(cell_fractions)

immune_bulk =<span class="st"> </span><span class="kw">log2</span>(immune_cell_reference<span class="op">$</span>expr_mat<span class="op">+</span><span class="dv">1</span>)
<span class="kw">colnames</span>(immune_bulk) =<span class="st"> </span><span class="kw">paste</span>(immune_cell_reference<span class="op">$</span>sample_description<span class="op">$</span>CellType, immune_cell_reference<span class="op">$</span>sample_description<span class="op">$</span>sample)

common_symbols =<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">rownames</span>(simulated_bulk), <span class="kw">rownames</span>(immune_bulk))


<span class="co"># all_samples = cbind(simulated_bulk[common_symbols,], immune_bulk[common_symbols, ])</span>
<span class="co"># pheatmap(all_samples, cluster_rows = FALSE, cluster_cols = TRUE, clustering_distance_cols=&quot;correlation&quot;)</span>

<span class="kw">cor</span>(simulated_bulk[common_symbols,], immune_bulk[common_symbols,]) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pheatmap</span>()</code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-15-1.png" width="768" /></p>

</div>
</div>
<h3> References</h3>
<div id="refs" class="references">
<div id="ref-Schelker2017">
<p>Schelker, Max, Sonia Feau, Jinyan Du, Nav Ranu, Edda Klipp, Gavin MacBeath, Birgit Schoeberl, and Andreas Raue. 2017. “Estimation of immune cell content in tumour tissue using single-cell RNA-seq data.” <em>Nature Communications</em> 8 (1): 2032. doi:<a href="https://doi.org/10.1038/s41467-017-02289-3">10.1038/s41467-017-02289-3</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="input-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sensitivity-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
