---
title: "Materials and Methods"
output: html_notebook
---

```{r setup, include=FALSE}
if (!require("pacman")) install.packages("pacman")
pacman::p_load(knitr,
               tidyverse,
               cowplot,
               Biobase,
               pheatmap,
               magrittr,
               assertthat,
               testthat,
               BioQC,
               readxl,
               ggpubr,
               stringr)

devtools::load_all("/storage/home/sturm/projects/immune_deconvolution_methods")
set_cibersort_binary("../lib/CIBERSORT/CIBERSORT.R")
set_cibersort_mat("../lib/CIBERSORT/LM22.txt")

color_file = "../tables/color_scales.xlsx"
color_scales = sapply(excel_sheets(color_file), function(sheet) {
  tbl = read_excel(color_file, sheet=sheet)
  colors = tbl$color
  names(colors) = tbl$value
  colors
})
```


# Input data

## Single cell data for simulated mixtures

```{r}
load("../data/single_cell_schelker/single_cell_schelker.rda")
```


```{r, include=FALSE}
valid_cells = which(pData(single_cell_schelker)$cell_type!="Unknown")

single_cell_schelker2 = single_cell_schelker[,valid_cells]

assert_that(!any(pData(single_cell_schelker2)$cell_type == "Unknown"))

available_cell_types = pData(single_cell_schelker2) %>% select(cell_type) %>% distinct() %>% pull(cell_type)

cancer_cells = c("Melanoma cells", "Ovarian carcinoma cells")
other_cells = c("Cancer associated fibroblasts", "Endothelial cells")
immune_cells = available_cell_types[!available_cell_types %in% c(cancer_cells, other_cells)]
```

In this study, we make use of the single cell dataset curated by [@Schelker2017]. They aggregated single cell sequencing data from different sources resulting in a set of ~12,000 single cells. They classified the cells using at set of 45 marker genes into `r length(available_cell_types)` categories:
`r length(cancer_cells)` cancer types (`r str_c(cancer_cells, collapse=", ")`),
`r length(immune_cells)` immune cells (`r str_c(immune_cells, collapse=", ")`),
`r length(other_cells)` other cells (`r str_c(other_cells, collapse=", ")`) and 
*Unknown cells* which could not have been classified unambiguously. 

Unknown cells are excluded from the downstream analysis. 

Table: the ~12,000 samples by cell type
```{r, echo=FALSE}
pData(single_cell_schelker) %>% group_by(cell_type) %>% count() %>% knitr::kable()
```


```{r, echo=FALSE, fig.width=10, fig.height=10, fig.cap="tSNE-clustering of the ~12000 single cells from [@Schelker2017]. "}
pData(single_cell_schelker) %>% 
  ggplot(aes(x=tsneX1, y=tsneX2, colour=cell_type)) +
           geom_point(size=1) + 
           theme(legend.position="bottom")
```

## Validation data (RNAseq + matched gold standard)
e.g. IHC, Flow cytometry, single cell

```{r}
datasets = read_excel("../tables/validation-datasets.xlsx")
kable(datasets)

```

### Hoeck data
8 PBMC, RNAseq+flow cytometry
```{r}
load("../data/hoeck/HoekPBMC_mixture.RData")
hoeck_mixture = mix.mat
load("../data/hoeck/HoekPBMC_gtruth.RData")
hoeck_ref = RefData
```


### Schelker data
Load the data and merge replicates
```{r}
ovarian_bulk_replicates = read_xls("../data/single_cell_schelker/ascites_bulk_samples.xls") 

samples = list("7873M"=c("ascites_7873_1", "ascites_7873_2"),
               "7882M"=c("ascites_7882_1", "ascites_7882_2"),
               "7892M"=c("ascites_7892_1", "ascites_7892_2"))

ovarian_bulk = lapply(samples, function(cols) {
  apply(ovarian_bulk_replicates[,cols], 1, mean)
}) %>% bind_cols() 

ovarian_bulk %<>% 
  mutate(gene_symbol = ovarian_bulk_replicates$Row)

ovarian_bulk_mat = ovarian_bulk %>%
  as.data.frame() %>%
  column_to_rownames("gene_symbol") %>%
  as.matrix()
```

Load the 'ground truth'
```{r}
true_cell_count = pData(single_cell_schelker2) %>% 
  filter(source == "ascites", donor %in% colnames(ovarian_bulk)) %>% 
  group_by(donor, cell_type) %>% 
  count() %>% 
  rename(cell_count=n) %>% 
  group_by(donor) %>%
  mutate(true_fraction=cell_count/sum(cell_count)) %>% 
  arrange(cell_type, donor)

total_cells = true_cell_count %>% 
  group_by(donor) %>%
  summarise(sum(cell_count))

total_cells
true_cell_count

schelker_ref = true_cell_count %>% select(sample=donor, cell_type, true_fraction)

```


### Racle data
```{r}
ensg_hgnc = read_tsv("../data/ensemble_hgnc.txt") %>% 
  rename(gene_id = `Gene stable ID`, hgnc_symbol=`HGNC symbol`)
racle_cyto = read_excel("../data/racle/racle2017_flow_cytometry.xlsx")
racle_files = list.files("../data/racle/GSE93722_RAW/", full.names = TRUE)
racle_identifiers = str_extract(racle_files, "LAU\\d+")
racle_expression_raw = lapply(list.files("../data/racle/GSE93722_RAW/", full.names = TRUE), read_tsv) 
gene_ids = racle_expression_raw[[1]]$gene_id
racle_tpm = sapply(racle_expression_raw, function(df) {pull(df, "TPM")}) %>% as.tibble()
colnames(racle_tpm) = racle_identifiers
racle_tpm$gene_id = gene_ids
racle_expression = racle_tpm %>%
  inner_join(ensg_hgnc) %>%
  filter(!is.na(hgnc_symbol)) %>%
  select(-gene_id) %>%
  group_by(hgnc_symbol) %>%
  summarise_all(sum)
```

