# Validitiy of single cell RNA-seq as reference

@Schelker2017 have shown that measurements from single cell sequencing are highly consistent with flow cytometry. 

## Validity of simulated bulk tissues from single cell sequencing data
We use simulated samples from single cell gene expression data. Does this approach make sense after all? 
We have 3 samples of single cell samples matched to bulk RNA sequencing. 

To test, we try:

* correlation of the simulated gene expression with the measured bulk gene expression. 
* correlation of the predicted fractions with all methods on bulk vs. simulated tissue.
* correlation of the simulated gene expression with bulk gene expression from the same tissue. 


### Compare simululated samples to bulk samples
```{r, include=FALSE}
res_methods_validity = new.env()
```

Simulate bulk samples by summing up the TPM gene expression values of all cells. Theoretically, it should not matter whether to use
`sum` or `mean` as they only differ by a constant scaling factor. However, we ran into numerical issues with CIBERSORT 
and quanTIseq using mean, due to the fact that some values may be really small due to the zero inflation of single cell data. 
```{r, cache=TRUE}
bulk_sum = sapply(colnames(schelker_ovarian$expr_mat), function(donor) {
  ind = pData(single_cell_schelker$eset)$donor == donor
  apply(exprs(single_cell_schelker$eset)[,ind], 1, sum)
})
```

```{r, cache=TRUE, echo=FALSE}
genes = intersect(rownames(bulk_sum), schelker_ovarian$bulk$gene_symbol)

expr_all = bind_rows(
  schelker_ovarian$bulk  %>% mutate(source="bulk"),
  bulk_sum %>% as_tibble(rownames="gene_symbol") %>% mutate(source="single_cell_aggregate")
) %>% 
  filter(gene_symbol %in% genes) %>%
  gather(sample, tpm, -gene_symbol, -source) %>%
  spread(source, tpm)
```


```{r, fig.width=16, fig.height=8, echo=FALSE, fig.cap="Correlation of bulkRNAseq data vs. simulated bulk data." }
expr_all %>%
  mutate(bulk=log2(bulk+1), single_cell_aggregate=log2(single_cell_aggregate+1)) %>%
  ggplot(aes(x=bulk, y=single_cell_aggregate)) + geom_point(alpha=.1) + stat_cor() + facet_wrap(~sample) + 
  geom_smooth(method = lm) + 
  xlab("log2(TPM + 1) of bulk RNA-seq sample") + 
  ylab("log2(sum(TPM) + 1)) of single cell RNA-seq samples ")

```


We observe a good correlation, although one can argue that it is very noisy and far from perfect. 
We do not draw any conclusions from this plot. 

### Additional tests for validity 
```{r}
ascites_mat = exprs(single_cell_schelker$eset)[,pData(single_cell_schelker$eset)$source == "ascites"]

qplot(ascites_mat[,1], ascites_mat[,3])

qplot()

```


### Compare the predicted fractions
We run all methods on both datasets and compare the results. 
```{r, cache=TRUE, message=FALSE, echo=FALSE, warning=FALSE, results='hide'}
timer_indications = rep("OV", ncol(schelker_ovarian$expr_mat)) 
all_results_bulk = lapply(immunedeconv::deconvolution_methods, function(method) {
  deconvolute(schelker_ovarian$expr_mat, method, indications=timer_indications) %>%
    mutate(method=method) %>%
    mutate(source="bulk")
})

all_results_simulated = lapply(immunedeconv::deconvolution_methods, function(method) {
  deconvolute(bulk_sum, method, indications=timer_indications) %>%
    mutate(method=method) %>%
    mutate(source="sum")
})

all_results = bind_rows(all_results_bulk, all_results_simulated) %>%
  gather(donor, fraction, -cell_type, -source, -method) %>%
  spread(source, fraction)

res_methods_validity$all_results = all_results
```

```{r, fig.width=12, fig.height=10, echo=FALSE}
all_results %>%
  ggplot(aes(x = bulk, y=sum)) +
    geom_point(aes(colour=cell_type)) + 
    facet_wrap(~method, scales="free") + 
    stat_cor() 
```

The results suggest a reasonable correlation between the estimates on bulkRNAseq and simulated bulkRNAseq. The results depend on the 
method, however. The very good agreement of EPIC and MCPCounter suggest, that the methodology itself is valid; just some methods appear to be more dependent on noise than others. xCell does not pick up a signal here as is "does not work on homogeneous samples", as stated on their github page. 



## Correlated simulated samples with bulk of the same tissue. 
We generate simulated bulk tissues of various cell types and correlate them with
bulk RNA seq samples from sorted cells. 
```{r, fig.width=8, fig.height=8, echo=FALSE}
# immune_cell_reference
# immune_cell_expr_mat

cell_fractions = diag(length(cell_types$immune_cells))
cell_fractions = rbind(cell_fractions, cell_fractions, cell_fractions)
colnames(cell_fractions) = cell_types$immune_cells
rownames(cell_fractions) = unlist(lapply(1:3, function(i) {paste0(cell_types$immune_cells, "_", i)}))

simulated_bulk = make_bulk_eset(eset=single_cell_schelker$eset, cell_fractions = cell_fractions, n_cells=500, combine=sum) %>% exprs() 
simulated_bulk = log2(simulated_bulk+1)
colnames(simulated_bulk) = rownames(cell_fractions)

immune_bulk = log2(immune_cell_reference$expr_mat + 1)
colnames(immune_bulk) = paste(immune_cell_reference$sample_description$cell_type, immune_cell_reference$sample_description$sample)

common_symbols = intersect(rownames(simulated_bulk), rownames(immune_bulk))


# all_samples = cbind(simulated_bulk[common_symbols,], immune_bulk[common_symbols, ])
# pheatmap(all_samples, cluster_rows = FALSE, cluster_cols = TRUE, clustering_distance_cols="correlation")

cor(simulated_bulk[common_symbols,], immune_bulk[common_symbols,]) %>% pheatmap()

```


In general, the highest correlation is observed between the expression of the sorted cells and the simulated bulk sample. 
However, the simulated Dendritic cells do not correlate well with any of the reference profiles. 