---
title: "Untitled"
output: html_document
---


# Validitiy of single cell RNA-seq as reference

## Validity of single cell sequencing for counting immune cells
* Schelker et al have shown that measurements from single cell sequencing are highly consistent with flow cytometry. 

## Validity of simulated bulk tissues from single cell sequencing data
We use simulated samples from single cell gene expression data. Does this approach make sense after all? 
We have 3 samples of single cell samples matched to bulk RNA sequencing. 

To test, we try:

* correlation of the simulated gene expression with the measured bulk gene expression. 
* correlation of the predicted fractions with all methods on bulk vs. simulated tissue.
* correlation of the simulated gene expression with bulk gene expression from the same tissue. 



```{r, cache=TRUE}
ovarian_bulk_replicates = read_xls("../data/single_cell_schelker/ascites_bulk_samples.xls") 

samples = list("7873M"=c("ascites_7873_1", "ascites_7873_2"),
               "7882M"=c("ascites_7882_1", "ascites_7882_2"),
               "7892M"=c("ascites_7892_1", "ascites_7892_2"))

ovarian_bulk = lapply(samples, function(cols) {
  apply(ovarian_bulk_replicates[,cols], 1, mean)
}) %>% bind_cols() 

ovarian_bulk %<>% 
  mutate(gene_symbol = ovarian_bulk_replicates$Row)

ovarian_bulk_mat = ovarian_bulk %>%
  as.data.frame() %>%
  column_to_rownames("gene_symbol") %>%
  as.matrix()
```

Load the 'ground truth'
```{r, cache=TRUE}
true_cell_count = pData(single_cell_schelker2) %>% 
  filter(source == "ascites", donor %in% colnames(ovarian_bulk)) %>% 
  group_by(donor, cell_type) %>% 
  count() %>% 
  rename(cell_count=n) %>% 
  group_by(donor) %>%
  mutate(true_fraction=cell_count/sum(cell_count)) %>% 
  arrange(cell_type, donor)

total_cells = true_cell_count %>% 
  group_by(donor) %>%
  summarise(sum(cell_count))

total_cells
true_cell_count

```

### Compare simululated samplese to bulk samples

randomized: 

```{r, cache=TRUE}
cell_fractions = true_cell_count %>%
  select(donor, cell_type, true_fraction) %>%
  complete(donor, cell_type=available_cell_types) %>%
  mutate(true_fraction=ifelse(is.na(true_fraction), 0, true_fraction)) %>%
  spread(cell_type, true_fraction) %>%
  as.data.frame() %>%
  column_to_rownames("donor")

bulk_eset_rand = make_bulk_eset(eset=single_cell_schelker2, cell_fractions = cell_fractions, n_cells=500)
bulk_rand = eset_to_matrix(bulk_eset_rand, column = "gene_symbol")
colnames(bulk_rand) = names(samples)

```

summing up all cells: 
```{r, cache=TRUE}
bulk_sum = sapply(names(samples), function(donor) {
  ind = pData(single_cell_schelker2)$donor == donor
  # apply(exprs(single_cell_schelker2)[,ind], 1, function(x){quantile(x, c(.98))})
  apply(exprs(single_cell_schelker2)[,ind], 1, sum)
})
```


```{r, cache=TRUE}
genes = intersect(rownames(bulk_sum), ovarian_bulk$gene_symbol)

expr_all = bind_rows(
  ovarian_bulk  %>% mutate(source="bulk"),
  bulk_rand %>% as_tibble(rownames="gene_symbol") %>% mutate(source="rand"),
  bulk_sum %>% as_tibble(rownames="gene_symbol") %>% mutate(source="sum")
) %>% 
  filter(gene_symbol %in% genes) %>%
  gather(sample, tpm, -gene_symbol, -source) %>%
  spread(source, tpm)
```


```{r, fig.width=16, fig.height=8}
expr_all %>%
  mutate(bulk=log2(bulk+1), sum=log2(sum+1)) %>%
  ggplot(aes(x=bulk, y=sum)) + geom_point() + stat_cor() + facet_wrap(~sample) + 
  geom_abline(slope=1, intercept = 0, col="red")

```

### Compare the predicted fractions
```{r, cache=TRUE}
all_results_bulk = lapply(immunedeconv::deconvolution_methods, function(method) {
  deconvolute(ovarian_bulk_mat, method) %>%
    mutate(method=method) %>%
    mutate(source="bulk")
})

all_results_simulated = lapply(immunedeconv::deconvolution_methods, function(method) {
  deconvolute(bulk_sum, method) %>%
    mutate(method=method) %>%
    mutate(source="sum")
})

all_results = bind_rows(all_results_bulk, all_results_simulated) %>%
  gather(donor, fraction, -cell_type, -source, -method) %>%
  spread(source, fraction)

```

```{r, fig.width=12, fig.height=8}
all_results %>%
  ggplot(aes(x = bulk, y=sum)) +
    geom_point(aes(colour=cell_type)) + 
    facet_wrap(~method, scales="free") + 
    stat_cor()

```


## Correlated simulated samples with bulk of the same tissue. 
```{r, fig.width=8, fig.height=8}
# immune_cell_reference
# immune_cell_expr_mat

cell_fractions = diag(length(cell_types$immune_cells))
cell_fractions = rbind(cell_fractions, cell_fractions, cell_fractions)
colnames(cell_fractions) = cell_types$immune_cells
rownames(cell_fractions) = unlist(lapply(1:3, function(i) {paste0(cell_types$immune_cells, "_", i)}))

simulated_bulk = make_bulk_eset(eset=single_cell_schelker$expr, cell_fractions = cell_fractions, n_cells=500) %>% exprs() 
simulated_bulk = log2(simulated_bulk+1)
colnames(simulated_bulk) = rownames(cell_fractions)

immune_bulk = log2(immune_cell_reference$expr_mat+1)
colnames(immune_bulk) = paste(immune_cell_reference$sample_description$CellType, immune_cell_reference$sample_description$sample)

common_symbols = intersect(rownames(simulated_bulk), rownames(immune_bulk))


# all_samples = cbind(simulated_bulk[common_symbols,], immune_bulk[common_symbols, ])
# pheatmap(all_samples, cluster_rows = FALSE, cluster_cols = TRUE, clustering_distance_cols="correlation")

cor(simulated_bulk[common_symbols,], immune_bulk[common_symbols,]) %>% pheatmap()

```


<!-- ### second dataset -->
<!-- We have another dataset from Chung 2017 et al with single cell and matched bulk.  -->
<!-- We do the same comparison.  -->


<!-- ```{r, cache=TRUE} -->
<!-- single_cell_chung = read_tsv("../data/single_cell_chung/GSE75688_GEO_processed_Breast_Cancer_raw_TPM_matrix.txt") -->

<!-- single_cell_chung2 = single_cell_chung %>% -->
<!--   gather(sample, tpm, -gene_id, -gene_name, -gene_type) %>% -->
<!--   select(gene_name, sample, tpm) %>% -->
<!--   group_by(sample, gene_name) %>% -->
<!--   summarise(tpm=sum(tpm)) %>% -->
<!--   ungroup() %>%  -->
<!--   separate(sample, into=c("donor", "sample"), sep="_") -->
<!-- ``` -->

<!-- ### Overview of bulk samples -->
<!-- ```{r} -->
<!-- single_cell_chung2 %>%  -->
<!--   select(donor, sample) %>% -->
<!--   distinct() %>% -->
<!--   filter(sample %in% c("Pooled", "Tumor")) -->
<!-- ``` -->


<!-- ### Separate bulk and single cells -->
<!-- ```{r, cache=TRUE} -->
<!-- chung_bulk = single_cell_chung2 %>% -->
<!--   filter(sample %in% c("Pooled", "Tumor")) %>% -->
<!--   filter(!(donor == "BC01" & sample == "Tumor")) %>% # remove double bulk reference for sample 1 %>% -->
<!--   mutate(source="bulk") -->

<!-- chung_sum = single_cell_chung2 %>% -->
<!--   filter(!sample %in% c("Pooled", "Tumor")) %>%  -->
<!--   group_by(donor, gene_name) %>% -->
<!--   summarise(tpm = sum(tpm)) %>% -->
<!--   ungroup() %>%  -->
<!--   mutate(source="sum") -->

<!-- genes = intersect(chung_bulk$gene_name %>% unique(), chung_sum$gene_name %>% unique()) -->

<!-- all_results_chung = bind_rows(chung_bulk, chung_sum) %>% -->
<!--   spread(source, tpm) -->
<!-- ``` -->