```{r setup, include=FALSE}
library(knitr)
library(ggplot2)
library(dplyr)
library(foreach)
library(tibble)
library(tidyr)
library(magrittr)
library(readr)
library(BioQC)
library(doMC)
library(ggpubr)
library(ggbeeswarm)
opts_knit$set(self.contained=TRUE)
devtools::load_all("../immunedeconv")
source("../lib/make_cell_fraction_vector.R")
source("config.R")

```

# Investigate marker genes
```{r}
load("../data/schelker/single_cell_schelker.rda")
mat = cbind(pData(single_cell_schelker), t(exprs(single_cell_schelker))) %>% as_tibble() 
mat = mat %>% filter(source != "pbmc")
```


## Show B cell and DC marker across the t-SNE plot
```{r, fig.cap="B cell and D cell clusters in the single cell dataset. ", echo=FALSE}
mat %>% 
  mutate(cell_type = ifelse(!cell_type %in% c("Natural killer cells", "CD4+ T cells"), "other", cell_type)) %>% 
  ggplot(aes(x=tsneX1, y=tsneX2, colour=cell_type)) +
           geom_point(size=1) + 
           theme(legend.position="right") +
#           scale_color_manual(values=color_scales$cell_types) +
           guides(colour=guide_legend(ncol = 1, override.aes= list(size=4))) +
           xlab("tSNE-1") +
           ylab("tSNE-2") +
           labs(color="cell type")
```

### Marker gene expression:
```{r, fig.width=5, fig.height=4, echo=FALSE, message=FALSE}
foreach(gene = c("MS4A1", "CD19", "ITGAM", "ITGAX", "CLEC4C", "IL3RA", "TCL1A"),
        cell_type = c("B cell", "B cell", "myleoid", "myleoid", "plasmacytoid DCs", "plasmacytoid DCs", "??")) %do% {
          mat %>% ggplot(aes(x=tsneX1, y=tsneX2)) + 
            geom_point(aes_string(colour=gene), size=.4) +
#            scale_colour_distiller(palette="RdBu") +
            ggtitle(paste(gene, "-", cell_type))
        }
```

**Conclusion**: The B cell and DC cluster separate well, the marker genes do not overlap. 
The DC cluster are plasmacytoid DCs, mDCs are somewhere hidden in the Macrophage/Monocyte cluster
and cannot be distinguished. 

```{r, cache=TRUE, include=FALSE}
load("../results/cache/sensitivity_analysis_dataset.rda")
```


## Check expression of markers in "detection limit" simulation dataset. 
```{r, include=FALSE}
show_cell_types = c("B cell", "Dendritic cell", "Macrophage/Monocyte",
                    "NK cell", "T cell CD4+", "T cell CD8+")
```

```{r, echo=FALSE}
n_immune_cells = rep(c(seq(0, 50, 5)
                     , seq(60, 100, 10)
                     , seq(150, 500, 50)
                     , seq(600, 1000, 100)
                     ), 5)
```


```{r, cache=TRUE, include=FALSE}
alldata = foreach(celltype = names(sensitivity_analysis_dataset),
        .combine=bind_rows) %do% {
          sensitivity_analysis_dataset[[celltype]] %>% as.tibble(rownames="gene") %>% mutate(celltype=celltype)
  
        }
colname2ncell = data.frame(sample=colnames(sensitivity_analysis_dataset$`B cell`), ncell=n_immune_cells)
alldata2 = alldata %>% gather(sample, TPM, -gene, -celltype)
alldata2 %<>% inner_join(colname2ncell)
```


```{r}
markers = list(
  "B cell" = c("MS4A1", #=CD20
               "CD19", "CD22"
  ), 
  "myleoid" = c("ITGAM", "ITGAX"), #=CD11B,CD11C
  "MHC class II" = c("HLA-DRB1", "HLA-DRA"),
  "plasmacytoid DC" = c("CLEC4C", "IL3RA"))
```

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Correlation of marker genes with increase of the amount of a certain cell type. Along x-axis: cell type used for simulation. Along y-axis: groups of marker genes. ", fig.height=12, fig.width=12}
markers_df = foreach(cell_type = names(markers), genes=markers, .combine=bind_rows) %do% {
  data.frame(gene=genes) %>% mutate(markers_for=cell_type)
}
allmarkers = unlist(markers)
alldata2 %>% filter(gene %in% allmarkers) %>%
  inner_join(markers_df) %>%
  ggplot(aes(x=ncell, y=TPM)) +
    geom_point(aes(colour=gene)) +
    facet_grid(gene~celltype, scales = "free_y") + 
    scale_color_brewer(palette="Set3") + stat_cor()
```

```{r, echo=FALSE, fig.width=3, fig.height=3, fig.cap="Even though the number of dendritic cells does correlate with MS4A1, a B cell marker, the absolute expression is really low. This effect can be driven by few misclassified cells but cannot explain the spillover effects we observe. "}

alldata2 %>% filter(celltype == "Dendritic cell", gene == "MS4A1") %>%
  ggplot(aes(x=ncell, y=TPM)) +
  geom_point()+
  stat_smooth(method="lm") +
  stat_cor()
```

### Derive marker genes from quanTIseq/EPIC signature matrices. 
We use gini-index (Zhang et al. (2017), BMC genomics) to filter for highly discriminating genes
within the matrix. We select all genes with a gini index > 0.7. We assign all selected genes as marker gene
to the cell type in which it is expressed the most. 
In the end, we receive a list of signature genes for each cell type. 

```{r, message=FALSE, warning=FALSE}
til10 = read_tsv("../immunedeconv/inst/extdata/quantiseq/TIL10_signature.txt") %>% as.data.frame() %>%
  column_to_rownames("ID")
gini_index = apply(til10, 1, gini)
til10_gini = til10[gini_index > .7,]
til10_gini_rk = apply(til10_gini, 1, function(x) {rank(-x)}) %>% t()

quantiseq_markers = sapply(colnames(til10_gini_rk), function(cell_type) {
   rownames(til10_gini_rk)[til10_gini_rk[,cell_type] == 1]
}, simplify = FALSE, USE.NAMES = TRUE)
```

The same procedure is applied to EPIC and CIBERSORT. 
```{r, include=FALSE}
tref = EPIC::TRef$refProfiles[EPIC::TRef$sigGenes,]
gini_index = apply(tref, 1, gini)
tref_gini = tref[gini_index > .7,]
tref_gini_rk = apply(tref, 1, function(x) {rank(-x)}) %>% t()

epic_markers = sapply(colnames(tref_gini_rk), function(cell_type) {
  rownames(tref_gini_rk)[tref_gini_rk[,cell_type] == 1]
}, simplify = FALSE, USE.NAMES = TRUE)
```

```{r, include=FALSE}
lm22 = read_tsv("../lib/CIBERSORT/LM22.txt") %>% as.data.frame() %>% column_to_rownames("Gene symbol")
gini_index = apply(tref, 1, gini)
lm22_gini = tref[gini_index > .7,]
lm22_gini_rk = apply(lm22, 1, function(x) {rank(-x)}) %>% t()

cibersort_markers = sapply(colnames(lm22_gini_rk), function(cell_type) {
  rownames(lm22_gini_rk)[lm22_gini_rk[,cell_type] == 1]
}, simplify = FALSE, USE.NAMES = TRUE)
```


```{r, echo=FALSE, fig.width=10, fig.height=3, warning=FALSE, message=FALSE, fig.cap="Expression across cell types. Along x-axis: cell types used for simulation. Along y-axis: Signature genes from EPIC/quanTIseq for the cell types. "}
marker_list = list(quantiseq=quantiseq_markers, epic=epic_markers)
foreach(method = names(marker_list)) %do% {
  markers = marker_list[[method]]
  markers_df = foreach(cell_type = names(markers), genes=markers, .combine=bind_rows) %do% {
    data.frame(gene=genes) %>% mutate(markers_for=cell_type)
  }
  allmarkers = unlist(markers)
  alldata3 = alldata2 %>% filter(gene %in% allmarkers) %>%
    filter(ncell > 600) %>%
    inner_join(markers_df) %>%
    filter(markers_for %in% c("T.cells.CD8", "CD8_Tcells"))
 
  alldata3 %>%
    ggplot(aes(x=gene, y=TPM)) +
      geom_boxplot() + 
      coord_flip() + 
      facet_grid(markers_for~celltype, space="free") + 
      scale_color_brewer(palette="Set1") + 
      theme(strip.text.y = element_text(angle=0)) + 
      ggtitle(method) # + scale_y_log10() 
  # ggsave(paste0("../results/figures/marker_gene_expression_", method, ".pdf"), width = 15, height = ifelse(method=="cibersort", 75, 15), limitsize=FALSE)
}

```

See figures:
* [EPIC](figures/marker_gene_expression_epic.pdf)
* [quanTIseq](figures/marker_gene_expression_quantiseq.pdf)
* [CIBERSORT](figures/marker_gene_expression_cibersort.pdf)

The figures show the expression of marker genes across cell types. Along x-axis: cell types used for simulation. Along y-axis: signature genes for the cell types derived from the signature matrices of the deconvolution methods. 


## Deconvolution results before and after
```{r, include=FALSE}
dc_expression = foreach(i=1:10,
        .combine = bind_cols,
        .final = as.matrix) %do% {
          cell_n = make_cell_fraction_vector("Natural killer cells", k=500, background=NULL)
          cell_frac = cell_n/sum(cell_n)
          make_random_bulk(single_cell_schelker, cell_frac, n_cells=500)
        }
rownames(dc_expression) = rownames(exprs(single_cell_schelker))

use_methods = c("epic", "quantiseq", "cibersort")
use_methods = immunedeconv::deconvolution_methods
show_cell_types = c("B cell", "Dendritic cell", "Macrophage/Monocyte",
                    "NK cell", "T cell CD4+", "T cell CD8+")
timer_indications = rep("OV", ncol(dc_expression))

res_before = foreach(method=use_methods, 
                     .combine=bind_rows) %do% {
                       deconvolute(dc_expression, method=method, indications=timer_indications) %>% 
                         map_result_to_celltypes(show_cell_types, method) %>% 
                         as_tibble(rownames = "cell_type") %>% 
                         mutate(method=method, dataset="before")
                     }

res_after = foreach(method=use_methods, 
                     .combine=bind_rows) %do% {
                       deconvolute(dc_expression, method=method, indications=timer_indications, rmgenes=c("IL7R", "TRAT1")) %>% 
                         map_result_to_celltypes(show_cell_types, method) %>% 
                         as_tibble(rownames = "cell_type") %>% 
                         mutate(method=method, dataset="after")
                     }

res_all = bind_rows(res_before, res_after) %>%
  gather(sample, predicted_fraction, -cell_type, -method, -dataset) %>% 
  mutate(dataset = factor(dataset, levels=c("before", "after")))

```

```{r, fig.width=12, fig.cap="Predicted", echo=FALSE, message=FALSE, fig.cap="Predictions on 10 simulated DC samples before and after removal of the five genes. ", warning=FALSE}
res_all %>% 
  ggplot(aes(x=dataset, y=predicted_fraction, colour=dataset)) +
    geom_quasirandom() + 
    stat_summary(fun.y=mean, geom="crossbar", fun.ymin=mean, fun.ymax=mean, width=.5, color="black") + 
   facet_grid(method~cell_type, drop = TRUE, scales = "free") + 
  stat_compare_means(paired = TRUE, method = "t.test")

```
